<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Pipeliner is an open-source and scalable solution for analyzing  next-generation sequencing data. Pipeliner provides access to  the same best-practices NGS pipelines developed and benchmarked  by experts at CCBR and NCBR. This repository is the main source  of documentation for users and developers working with or  contributing to Pipeliner."><meta name=author content="Skyler Kuhn"><link rel=icon href=../assets/favicon/favicon.ico><meta name=generator content="mkdocs-1.4.2, mkdocs-material-8.3.0"><title>Changelog - Pipeliner Documentation</title><link rel=stylesheet href=../assets/stylesheets/main.fd896c8a.min.css><link rel=stylesheet href=../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><script>__md_scope=new URL("..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#v402 class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=.. title="Pipeliner Documentation" class="md-header__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../assets/icons/doc-book.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Pipeliner Documentation </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Changelog </span> </div> </div> </div> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=.. title="Pipeliner Documentation" class="md-nav__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../assets/icons/doc-book.svg alt=logo> </a> Pipeliner Documentation </label> <div class=md-nav__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=.. class=md-nav__link> Home </a> </li> <li class=md-nav__item> <a href=../about/ class=md-nav__link> About </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3> <label class=md-nav__link for=__nav_3> Getting started <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Getting started" data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Getting started </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1> <label class=md-nav__link for=__nav_3_1> Users <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Users data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Users </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../dev/coming-soon/ class=md-nav__link> Prerequisites </a> </li> <li class=md-nav__item> <a href=../dev/coming-soon/ class=md-nav__link> Installation </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <label class=md-nav__link for=__nav_3_2> Developers <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Developers data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Developers </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../dev/coming-soon/ class=md-nav__link> Installation </a> </li> <li class=md-nav__item> <a href=../dev/coming-soon/ class=md-nav__link> Customization </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4> <label class=md-nav__link for=__nav_4> Pipelines <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Pipelines data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Pipelines </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <label class=md-nav__link for=__nav_4_1> ChIP-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=ChIP-seq data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> ChIP-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../ChIP-seq/ChIP-seq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../ChIP-seq/ChIP-seq-output-files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../ChIP-seq/ChIP-seq-tools/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2> <label class=md-nav__link for=__nav_4_2> RNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=RNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> RNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../RNA-seq/Theory-and-practical-guide-for-RNA-seq/ class=md-nav__link> Theory </a> </li> <li class=md-nav__item> <a href=../RNA-seq/Gene-and-isoform-expression-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../RNA-seq/TLDR-RNA-seq/ class=md-nav__link> Quick Start </a> </li> <li class=md-nav__item> <a href=../RNA-seq/Gene-and-isoform-expression-guided-tutorial/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../RNA-seq/Differential-expression-pipeline-tools-and-versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <label class=md-nav__link for=__nav_4_3> miRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=miRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> miRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../miRNA-seq/overview-miRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../miRNA-seq/TLDR-miRNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../miRNA-seq/miRSeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../miRNA-seq/miRSeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <label class=md-nav__link for=__nav_4_4> scRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=scRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> scRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../scRNA-seq/overview-scRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../scRNA-seq/TLDR-Single-Cell-RNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../scRNA-seq/scRNASeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../scRNA-seq/scRNASeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <label class=md-nav__link for=__nav_4_5> WES/WGS <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=WES/WGS data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> WES/WGS </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../WEG-seq/Whole-exome-genome-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../WEG-seq/TLDR-WEG-seq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../WEG-seq/Whole-exome-genome-output/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../WEG-seq/Whole-exome-genome-tools-and-versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_5 type=checkbox id=__nav_5> <label class=md-nav__link for=__nav_5> FAQ <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=FAQ data-md-level=1> <label class=md-nav__title for=__nav_5> <span class="md-nav__icon md-icon"></span> FAQ </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../troubleshooting/ class=md-nav__link> Troubleshooting </a> </li> <li class=md-nav__item> <a href=../dev/coming-soon/ class=md-nav__link> Data Privacy </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../community-and-contribution/ class=md-nav__link> Contribute </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> Changelog <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> Changelog </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#v402 class=md-nav__link> v4.0.2 </a> </li> <li class=md-nav__item> <a href=#v401 class=md-nav__link> v4.0.1 </a> </li> <li class=md-nav__item> <a href=#v400 class=md-nav__link> v4.0.0 </a> </li> <li class=md-nav__item> <a href=#v306 class=md-nav__link> v3.0.6 </a> </li> <li class=md-nav__item> <a href=#v305 class=md-nav__link> v3.0.5 </a> </li> <li class=md-nav__item> <a href=#v304 class=md-nav__link> v3.0.4 </a> </li> <li class=md-nav__item> <a href=#v303 class=md-nav__link> v3.0.3 </a> </li> <li class=md-nav__item> <a href=#v302 class=md-nav__link> v3.0.2 </a> </li> <li class=md-nav__item> <a href=#v301 class=md-nav__link> v3.0.1 </a> </li> <li class=md-nav__item> <a href=#v300 class=md-nav__link> v3.0.0 </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../contact-us/ class=md-nav__link> Contact </a> </li> <li class=md-nav__item> <a href=../license/ class=md-nav__link> License </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#v402 class=md-nav__link> v4.0.2 </a> </li> <li class=md-nav__item> <a href=#v401 class=md-nav__link> v4.0.1 </a> </li> <li class=md-nav__item> <a href=#v400 class=md-nav__link> v4.0.0 </a> </li> <li class=md-nav__item> <a href=#v306 class=md-nav__link> v3.0.6 </a> </li> <li class=md-nav__item> <a href=#v305 class=md-nav__link> v3.0.5 </a> </li> <li class=md-nav__item> <a href=#v304 class=md-nav__link> v3.0.4 </a> </li> <li class=md-nav__item> <a href=#v303 class=md-nav__link> v3.0.3 </a> </li> <li class=md-nav__item> <a href=#v302 class=md-nav__link> v3.0.2 </a> </li> <li class=md-nav__item> <a href=#v301 class=md-nav__link> v3.0.1 </a> </li> <li class=md-nav__item> <a href=#v300 class=md-nav__link> v3.0.0 </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/CCBR/pipeliner-docs/edit/master/docs/changelog.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1>Changelog</h1> <h2 id=v402>v4.0.2<a class=headerlink href=#v402 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/c69ab80a2794114b7b3fd1508e24960aea93e15b>Increasing picard memory allocation and num of threads for garbage collection</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2082431d04ffd3b17eedd8f727680b4eb801c0cd>ChIP-seq updates: estimated fragment length and more (#450)</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8701b49d5bcdb64726ededb66353899a2d97acab>Merge pull request #449 from wong-nw/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9810ee09ea2c1ef2342845ffcfcbba0946c58a72>Bug Fix in integrateBatches.R URD call</a></li> </ul> <h2 id=v401>v4.0.1<a class=headerlink href=#v401 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/63a5ce9ac8992b5160a2fa266462a86c5dbbe1b8>removed AVIA</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e6ac6c176c7855feef35380fce3d19abc8a46623>Increasing resources for fusion inspector</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6becb18e6da109428253b6688b33b0981eba13a9>Switch to python script</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f82a692aef74e29b0078bb54504f101ad0691701>Python implementation of 'reformat_bed.pl'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/95c32405aa1266939e1b13857d0f13523b2c075c>Adding memory for fusion inspector rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d9378e8cfd8907614cab07752b259c06160f11b6>mm10 error</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4142003616e706729177208c16f9b123aae8b8bd>Run only for 'hg19' or 'hg38'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ab608406aa2743d84bce68fbc3339c8c551b1566>Remove ancient tag</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/693d84a776848316d06aaee122499621e18c143e>Fixing STAR fusion perl issue</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ead71c061550f08b7b794d0f42326786162a0792>Explicitly load Perl</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/baf82250a44a790a66c4abe5b88d0ebbf024b8be>scRNASeq changes Feb 2020 (#438)</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/749f808efee6aef5479e6e47d8b2d1c13035ca75>Changing git version: module git/2.15.1 is no longer available</a></li> </ul> <h2 id=v400>v4.0.0<a class=headerlink href=#v400 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/57ca868f7cbb96eb805d37afbd07e56659958e8e>Merging activeDev for v4.0.1 release (#444)</a></li> </ul> <h2 id=v306>v3.0.6<a class=headerlink href=#v306 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/e92a6a357403789fae073a5c27253da46ffd18dc>Setting activeDev branch to master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d7ef5c52cb5646820e13a28fe67c0bf49b77d316>Adding updated viral integrated genome hg38_30+HPV16</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e03e5eb17f822dd233910710d40ae9ec54411f04>Archiving older viral integrated hg38_28+HPV16 references json</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2ab3f8378c678e6c618f4325b3a6412e9f98cee1>Explicitly loading perl to aviod: 'Perl module GD::Graph::bars not installed, skipping charts' error</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/223f71749c75237c8bca39ff82f2ea2e83a54083>Replacing hg19_KSHV with hg38_30_KSHV</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e0a28aad0858898203d5f191c5accf5f4c4e2e5f>Adding "References/" as a cache for old reference files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d2d175b92905f224989a716caabd93b31379bd50>Updating targets file text, now changes based on reference genome</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/13178d6504fb8925f90478093533720d1bc052cc>Merge pull request #432 from mtandon09/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f473d5864fed36c2d6565eff60fe78b8f16758f6>Specify R version (3.5) because default (3.6) does not have 'magick' installed</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9b17bdaf757e0510c55b1cdc1cd878a2273e4e50>Merge pull request #431 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a5bdb62813411f2e1269d11c02159ca41c6a06a4>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/115a593f66964894c8fe60287493914dcea63f56>fixed gzipped VCF issue with germline calling</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6dccc34be8ebe8d9bc95c6878ec2a52fb2078d0e>Merge pull request #430 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b922e977a7fa9a61d37008c38152da416f89bc36>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8ea10fddba86f82418d977bb62f5467a7bee50cd>updated hg38 PON</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/963b73f1e3798440f507eaa1ccb4597305654fb1>Updating rule pca: prevents race condition due to severe filesystem latency</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4dcf8a59a24a1617c89894feb43f90c9355ca689>Updating rule pca: severe filesystem latency can create race condition when copying over files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0925e5e4d538a83339d4e9faff87a0294d804fc6>Updating pcacall usage to aviod filesystem latency issues</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ba16a03a67866fefdba601e1c4c4c46e0f0b4145>RNA-seq Initial QC: Contrasts file not longer required</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1931a807af9c69040e82574b527b72f5ca271a39>ChIP-seq GUI: Adding "contrasts" button to the GUI</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c65b2e79fc7eba8a0cefd1b9428cd8f9b4886aa0>Updating Initial QC RNA-seq Interface: only ask for group information</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/77acdb2ee8ed4891e52990810fdba22c810bbd99>Update dryrun file-checking for DEG pipelines only</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/392c2663583b3a03f8759bf52569e68f25252ff0>Updating DEG Reports for mm10_M21</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/734b16eec9c544be587d59dd0ca164931d1803e2>Updating karyploter's usage to include mm10_M21</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ef24dd86adb7e9ed544b607b3716dc7ba6c9d824>Adding "mm10_M21" to drop down reference genome</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1077ced925beb59abf18580c07dcb723e5bbcd69>Adding "mm10_M21" Reference Genome: Built using version M21 (Ensembl 96) annotation</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/45841878a9b7222952f4738caf7fe0ed62306491>Adding <strong>pycache</strong> to gitignore</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/dc0d10e503fceea160c98639e711abf12de4fc45>Merge pull request #429 from kopardev/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7849530e04a23b2265461421dd35563ae46c7469>tidyverse added to genecounts.R along with writetogz function</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/090c8f02985f605ebdeabf0bfe12bffb6806ef81>trim_se is now temp too</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2d8f342de64484bc6066ed5030cb70d9228332f5>tmp is actually temp</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1fd2ddca3d4db26e56e332148148956cf361dde4>tmp is actually temp</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6bd98fcd0dbc4768fb78ddf077d304fb5df9657a>make trim files temp</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1ae01455a2391f1b3a6cbd1072decc1bc2755fc3>not usin g deeptools anymore for bam2bw ... not doing any smoothing for bigwig files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cc68b8c5393471e922706ffa4690c122ee129478>bringing back bw creation</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e69aec24f837ba1eaa7ec17e2a0ddaf85d5a400c>Merge remote-tracking branch 'upstream/activeDev' into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d4159d6e7748b364d78222d9f1b44e1489613743>Updating STAR from 2.5.2b to 2.7.0f for all RNA-seq references genomes</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e3b3407af7e32271f41cd6c66ab748b090aeafa1>changing PcaReport.Rmd genecounts.R rsemcounts.R to account for the scenario that we have only 1 group</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/37c34bc4279572af183dbaa16e1306d7866f311f>Adding compatibility for "ref_vAnnotation" formatted reference genomes</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c974e2a33005b677654d4d2a930268c6aad2d97b>Removing Strand-specific BigWigs as a target file</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f26a19941ac251bae08235322c03678ac107d751>tabs converted to 4 spaces</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a6d8bda3be02d5d13dbf4f526d177c3504f75ee1>bam2bw bc fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/43b1cec7cc7d486206dde3036744db2ff6c69dc3>renaming bam2bw in rnaseq to bam2bw_rnaseq... and increasing its ram to 120g and cpus to 56</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2b367ba59cefd0cba9b988acd7ff7c7b50f1f87f>Adding hg38_30 to pipeliner's user interface</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/10edf55929f51cf5239938368220268eeea37fdc>Adding support for hg38_30 to karyoploter.R</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9b348301c105f360ea8151365f541a1814a36b6d>Updating DEG Reports for hg38_30</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/14c5cb078f5c9817f36494e771643b006c0edec2>Updating PATH to KARYOPLOTER</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5f11b91f5e614d3c1f234da885e2144e54e942c9>Updating the version of STAR from 2.5.2b to 2.7.0f</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/22d67c65e020fa870c0cab98114dcd33d2cf5d4b>Adding hg38_30 references using GENCODE version 30 annotation of the human genome (GRCh38)</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6ac7383d3b3add034484794cd9c156b15d07a08a>Merge pull request #428 from kopardev/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cbc0fec56d829a3ed223016eb70f33b5732c0176>added missing tail -n1 bam2bw rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4173ccedf9adeebda1c8be7f2047588eebf9e690>Merge pull request #427 from kopardev/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/81836701167bf417cf30a8266855f1912b612051>D1.0 type fixed to 1.0 in bam2bw rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/eb80395dee354d01c3e1255668d9bfbfb60815e1>Merge pull request #426 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e656192da80a1f0f15ff3b87a88cadbcb4c6ba48>updated STARfusion and FusionInspector, added merged final fusion outputs across 2 callers</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3a53d0f2169de7af6b054a4b2acc5ef9f00ff5b6>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0d4e54513ceb4e0f9824b54ba81c59c809ca741e>updated STARfusion, fusioncatcher, and FusionInspector versions</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fa276d46a87c23994e30ace1d1558703144c1093>cluster.json ... partition fixed back to norm only</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a3340aed78b478d6c9eb8f51db51a799a2d85f03>Merge pull request #425 from kopardev/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a48f458766c455e5707a480b3fefae141d3c266e>Deeptools rule now outputs bigwig files to bams folder</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4161c6e165c6f9cef8f007c43a4676305c3d00f0>Deeptools rule all wildcard fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a1283f5eb745b2cb31c8a5f8e8c258685641d4c5>Deeptools rule all indentation error fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d667ab51d54b4e8eaaa2fd53ca7daed85fd2dec1>rule all edited to add bam2bw outputs</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7234355c5da237a6daed4175013de28a7fd8a231>Merge pull request #424 from kopardev/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7337e36025d2df0fea5ce442228c916c42c90f1d>rpsd fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3d077d47a6bd2a5342077d696ad22f48ec793333>forward-prob fix for rsem</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2cf3eaa799bd3f0ef7bda37f1d3e8bc3ab5f94e0>file2 defined for se rsem rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c6472ed1fad86ba9f56da18c771188709ca8c32a>adding strand specificity to rsem counts command; adding rule to convert bams to forward and reverse bigwigs for easy visualization</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4334dd59b04fb2743a39e6ea913168ee875eb994>Merge pull request #423 from tovahmarkowitz/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/71a2562f5c2ad24224e2f22a391f44479bbef9db>fixed bug in gem rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5f327211324f3052b23b3d3c8d5653040aa6fc68>DiffBind,uropa,homer,manorm</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4e0fabbed9ff550e0dd3f6b422077cb2674f400d>homer,uropa,manorm,diffbind,jaccard</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/525dc930eb6fa1a9ca31765cfb140efb02e8e6ae>added BCFtools variant QC</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a2ce46bed52f584bc1e2f2950d2a038e12c1ff2e>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/45ede8c129d5db282ee53a29336375b981cfa911>homer,manorm,uropa,protcoding</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2075e2c87f4f5743d7711a250009abe56afe7ecb>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/acc9d4d7c157fb2ae1b9062086dd91e262acf2ea>added new fusion rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/90b1534dc3947b2fde7d3bad666ea283975eb7ea>removed target selection from polished germline WES variants; added BCFtools stats QC; substituted Agilent SS v7 target BED file for default targets</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/dfb9016188c95b5bb5d07113d704a6d952dbef39>update to mm9/mm10 json files... blacklist bwa index now includes chr_rDNA as well</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bd051b1c36b59aea203941ba78cba5c151598e5b>merging with main branch nggerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/38d74eee130ebc0bfd7f43ee9b9a8ab4c0520ac8>Changing effective genome size and adding differential binding analysis</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7600f40cb5f5879770b537e208d7c1d0f1393bb9>Added warning message to "karyoploter.R"</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7b33da1cf71ff51ec3d0a6b4857d6094c81dc749>Adding DEG Type prefix to output karyplot to prevent race conditions</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4090cbae8b05a38f918356f7dfbd84d674e58703>Adding last reference files for "karoploter.R"</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/88a7e65ace5f7f92b97a33ec4e582105f806472c>Adding gene expression karyoploter script</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4b01ce4ca9c4ab0373f1e75619740f6645f213ce>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e2bb8bf8190f62d4166bc38fa098e347c364a443>Marked Qualimap Counts to be moved to DEG</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/eb766fcb8192549001bd0637db1c2a8771e1c6fd>Updating GUI title</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7d5c0778ea2b288de457ab3786efa0b20a04d7da>Updating version to "4.0" for activeDev pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/23340faed56873eaaed6413367443af24d4cfa76>Updating "activeDev" Pipeliner branch version to 4.0</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e730d1d84ec3b66e2c6b23e728a293ab8f7661e4>Merge pull request #422 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/75f154fef4145d4bbe57af452b7546fc5e4fcecb>Adding "karyotxt" reference file to DESeq2, edgeR, limma calls</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/38a834184f968020d4d422a20c79d0ba05f4964b>Updating LimmaReport Rmarkdown</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f85070fb6a0a1798dc574b50956adaeeedf01439>Updating EdgerReport Rmarkdown</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1ffa5b36fd4c15657905a41b4c166a2575d12f7f>Updating DESeq2 Rmarkdown</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/20e76b0184e740fcf6d9c7beffa8c84990898a61>Adding karyoploter reference file to organism json files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/16cb9170f4372924b725cdf0cf1b54dcf1ac7e0d>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8df9f5da77f6416729cdfecb2a6fc98d9b5ebff7>expanded MAF annotation to merged somatic variants</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7336554cd30aeae834d00b5ae053765d5c6b45cd>Updating EBSeq.py Script</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/012965f38de51287bed206f370801470dfa5f275>Updating Ebseq rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/16ecfca6c1a9fdf09ae48cca8c697326fd9baeec>Adding 'hg38_HPV16' to GUI</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7584175ac3bd26d2e273996b132792369e73ea04>Adding resources for hg38_HPV16: hg38_HPV16.json</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/baf646f5c928c1c857320978f448510af611566c>Merge pull request #421 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9cc7fae271e12d9920ff908835bcdae8b4ad7986>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1cdea7d0cf45c472578825e0e42ad4128e2ec6d1>reduced verbosity of VarDictt to decrease slurm log file size</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e68d788799aa008709578d7268b88012cf37744f>Adding hg19_KSHV reference files and updating GUI</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7058e92867f12dcbf497fd182a1cb459eab796c0>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1a42a00bb4cc380667d78d263de36bb2bf1a3969>Updating QualiMap commands</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5babfabbbf2317add2bac77aecf02e650a4f3bc4>Merge pull request #420 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/21d48ac7518dc87e1b8434768d4fe52bb825ee17>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0cb02628f9a008ca8d700325ba68df2c2f21ffd6>added multi-caller somatic variant detection to tumor-normal pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0abb32e102745e7125a10b239b81cedfabc32aba>Adding QualiMap bamQC and QualiMap counts</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/eef42b4f4d6f68f8efccbf2de1428c47f39df967>Updating path to RNA-seq MultiQC config file</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a6355f9d1adc664ccbb5124a30067a59f4b5c9b1>Adding adding qualimap sources to json files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5679c242cc7039337d25a35daa5460d7f195a8be>Adding support for edge-case: group CPM and MINSAMPLE filtering 0 0 (i.e. no filtering)</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/dbf4d753725290e3d42dc76db492bb2ca515c0fe>completed multi-caller additions to tumor-only somatic pipeline</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4c2ee26aa2afa598e48edec7d97bb10698d15e36>Adding rule 'pca' to cluster.json</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f00d7cbc6a0f54ae49f67fd53fa7916c4138e3a8>fixed bugs in expanded WES tumor only pipeline; changed admixture outputs to all be printed in admixture_out; incorporated vardict somatic calling</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9b15248742981172b2efd37511ff5e3c6c038fc5>added Vardict tumor only somatic caller, added joint calling with 3 callers for tumor only calling</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c52fc9527d530328cda4d682d75ebfd53e817643>improving HOMER/UROPA speeds</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/db164af37cb9be8de0c9516087a7ceba6b2fa687>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6349c7ae176349f271e0a1fc89b52d13f9da6f64>Merge pull request #417 from tovahmarkowitz/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/66d2cb021876430f8c79da58cbafb2256d28461e>adding spearman correlation heatmap to multiQC</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cfd6a329cc6271d5c756884c0fc3d5036b9c4b05>adding to the multiQC report</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2a32b661418a812d98aead9855c974638e25c71e>removed mtDNA from germline genotyping</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b5c7acb406c2b331b5cc99de3fcabc6c5922f0f5>Updating 'rule QCstats' to grab the fragment Length</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6df704d62cb46b7180f32579c12105589384ddaf>Updating to 'filterMetrics' to accept fragLen as an argument</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/51e94d63fc22cf268b9ded8db205e63d4262435b>improving multiqc report, deeptools runtime, and other details</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4863a7b8db6b46b308f8c5f67e0a10cdce258d1f>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/50fed72562e1fead35a95ac3b47afbd2e6a1fb24>fixed bug in admixture plot generation</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/282b2580693946a2169f767cc01be883f820fad5>Resolves #416: use the project directory's 'cluster.json' file</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/290b7c7a3b717343034ef13ce00b45765a10e418>Merge pull request #408 from tovahmarkowitz/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/647a0879bae97bab4a025dd9532dfda2ecfd6845>begin tracking R scripts for generating sample network and admixture PNGs</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/be01e5ad46f687f946b3e97f48cf150625f6703f>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5198cc89540602bf5e63d750db76dd5990d66018>added conversion of sample network to PNG and embedding it in multiQC report</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f6b1ec16299ea00986f7fca23192f881c8ba21bb>updates to gui by Vishal</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7a8cec3bfab0bf69bc3539bec7a6903275f692f8>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f3deb2fd1893482b70d01c2fc03e6dbc0c6db08c>Resolves #415, added canFam3 and Mmul_8.0.1 to gui</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3de4652a64310d8867e3629017e63a987b7b9f83>added summary figure generation and P-value annotation for FREEC CNVs</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/224aeb34466454e13c387d78a4a79b601c46dafe>Adding Macaca mulatta (Mmul_8.0.1) and Canis_lupus_familiaris (canFam3) reference files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/02191eabadbd266ea907b92a7fcbaeaa8796ba5d>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6673df4a85ca412cf9343dafb5e486036beec5e2>adding vardict calling and multi-callier merging</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/23604f65f40208f0036ff1a28dd65eba24f68f3b>e for your changes. Lines starting</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5ebe1264651920aa768df855304d6e2718c8cb6b>rsemcount.R but fix... columns were getting duplicate names</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0b8070775e141ebeeaddcfdf2bfc280decf2507b>Updating STAR-p1 arguement: "clip3pAdapterSeq" to "--clip3pAdapterSeq"</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9afe4bb5189267a874c02120cef10976b47e3374>contrasts.tab now accepts 2, 3 or 4 columns, accepts numbers or fraction as column 4, fraction as to be &gt;=0.5 and &lt;1, numbers have to be integers; both numbers and fractions now denote number of samples or fraction of samples PER GROUP where GROUP is one of the groups in the contrast. Filtering of genes in DEG_ALL is changed from 0.5 CPM cutoff of atleast 2 samples out of all the samples to 0.5,0.5 setting ie., 50% of the samples PER GROUP should have atleast 0.5 CPM.</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2b56fcfa6ea173bd26600c893d6aaba95c508087>more memory for ngsqc</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e4158dc4ab073f8c7f777ff2658ec86e5dac1f42>multiqc update and uropa update</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d0575c89982732fff473b10f3516ee4ad19d567f>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/74827afc65d0108a4e131adda1828a9510919658>Merge pull request #406 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a21564907418c681c8b639f67bc2538aa2b7dcdc>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7129d464a120b69b3e25579424f0afb9744d2cdb>fixed bug in prep_mafs.pl script</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9530af736039744d145ed78f40a9b77e0b727c13>Updating ambiguous Subread genes output filename</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e7a7772ed9c5b67751c809cf5d17a8cedcfc768c>uropa fixes and cluster.json update</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/603bf223532cc60cc38b57f2f468711022ca7c2c>filtersamples.R: changed .. 4<sup>th</sup> column in contrasts.tab now denotes the number of samples PER GROUP that are required to meet the CPM threshold denoted in column 3 of the same file. If number in column 4 is greater than number of samples in the group, then MINCOUNT is reset to the number of samples in the group.</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4c5bb9858152b173504183cea8d609e79a3a018b>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b0d993306b0e5ccff00287d299c9f8e9c9f2ebaf>Resolves #405, 'preseq lc_extrap' approximation</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/35a7acf743eb015af1f6c6bc8d85f22d1b764c75>uropa temporary workaround</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/347cb5697def53d35036795cd20a0825594f35b6>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/df316c998d788a8711ef2c907a32924b1dd5c617>Merge pull request #404 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8a9a669d46c6d74f2db7e893ae07f4193b4a1383>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e534349afbe4fbc660304d082bed4597d50fd77d>improved somatic variant fitlering and maftools summaries</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e6dc1422b8d8822ea4e6e9d820d9bd4cb652968d>merging branches</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bb74f19297498dc0e382c65f627348ae8be72e8d>adding uropa and other changes to chip-seq pipeline</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1039f14977df2a4b8cb516b2be80cb8723a1f3dc>PhantomPeakQualTool: Bigwig Generation from Estimated Fragment Length, resolves #403</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b1589c51bc9c8d97c94a6db254534a482ab64cb6>Updated Heatmap Height and PCA report generation (when &gt; 100 samples)</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/80788a3195e0727d3d311044ef604ba9a9d97a0f>fixing issues caused by lack of input or PE sequencing data</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/712a68035e1876288e5aede53ed17e63e4ca691b>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1d20ea1f73e818903ea2024772745b3f22083120>Merge pull request #401 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ae0d41fc5b5f2124322dd6af428c6b6b63dbb104>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2b698e6f7c1a23612912797880252306cbbbcf33>fixed bug in WES BED file generation where empty files are written when an incorrect path to target bed is given</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cd72f4a3a9b25e41157abc928398a277604d0678>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cb2fedc760e7e35efc01974b65f1584f201c799d>Merge pull request #400 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c3f95fdae0fb33a82c5a4287aa1625b1c85dccfb>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a8e9aa83c088456309e45b523902433e01847930>moved sort out of STAR and into samtools rule to deal with memory bug</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/dbd5052c02fe061076995cdaf8517860cd866770>PcaReport Rmarkdown Fix: Before Normalization/DESeq2 #399</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0e56f041dd3a4613dc19f1c8a499e7d880b98642>reworked ChIPseq.snakefile with improvements</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fc7345e95b9a14cc607f1d31b62493c4e90a54ce>Merge pull request #394 from tovahmarkowitz/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cba940382b4856594f3936d7e19d260a148418f1>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/48e868587ec2726672b0c4441f97a76806794264>Merge pull request #398 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8ca55b3ad564c5c88f511ac443c6b7c93c99f985>added memory limit to STAR 2-pass in rnaseqvarcalling</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a5e7c58d5bd4ab1ad8401f793834c91baa13719e>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/effa650c5cf51405dfe8da46d071d5ef5ba7dea7>fixed bug in 2<sup>nd</sup> pass of freec</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8b837a25da3580b1b016205d5be69420eccff263>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ae68b5f1a1a928c686d12fd7150cf4e938b14d5c>Merge pull request #397 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/138c52f91818154b4934f06002388d4f25684d34>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e26e107bdeddeb978f935c8785e1fd61cb122fff>fixed bug tracking FREEC 2<sup>nd</sup> pass output</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6de40056823fb1733793fa8aa5604c439763ba76>gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/00d2f3cb32b6c646cd5f47f8d51fbe2830bc956a>Merge pull request #396 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9683091965c0a9c757fb76fcdef75ca436c7201f>completed new CNV workflow for WES and WGS</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8ce26e8a2c65c35b6edeac5ea23cf97c37796293>implementing 2-pass FREEC and Sequenza to WES and WGS pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d3ecf3c198af8ce7b74218aa7601ff1c30543241>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d01d82ce8b64119f6b03220fe8a6b1225a364834>completed sequenza rule for WES and WGS, and completed second-pass rules for 2-pass freec</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a5135e4949034d415e33ae9baa339186cc01fc0b>added multipass freec CNV analysis to use ploidy and purity prior from sequenza</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/da54b3c49f2e85288e6b8ecf44e77b95f7871a6e>tracking new files and scripts to run FREEC and sequenza</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f95d3f3966f6e1c85dc160a785028ef7884fcd95>added FREEC to tumor only pipeline, and sequenza and freec to tumor-normal pipeline</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/56eb3a22ec4254f558e2308a213fa33fc3616869>updated Canvas germline to most recent version, and fixed bugs</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cc0950bdc6ab164bbb80765965e2498bce5f08ef>updated Canvas and fixed bugs in Canvas somatic CNV calling</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/abb6466c0155829d5ef51dea90079e2008af4336>fixed error with ngsqc_plot rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/097280f65cbb8ff3adf6405e60932ff862d939be>adding ngsqc plot feature</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f76552758659b5932d8c49994fb4956f97549e8b>adding macs2 dedup functionality</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f27d54d61a70acf89cc7f5787397957d221e2bfe>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/de7b4d2ab0347f33b7ce8ac438edf5cdd796cb9d>added full germline calling to wgs tumor-only pipeline</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bedbd58bd3a8f88cde594bde92b8be94ad465bc4>added full germline genotyping to all somatic pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/063d63828e3b3e88fc5d88dd5132def2ac4ad6f8>Merge pull request #389 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a18b3d566586ef9d4631b76e45acbe0d9916fc0c>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/287069e9067fa40445a85a89465c9ddadf6e79a4>updated GATK version to 3.8 in combine somatic GVCFs, and switched to filtering all MAF files with perl script rather than in maftools</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/65cdb13ffdce911f0f52965a162be2b1f6657814>Merge pull request #388 from tovahmarkowitz/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/81b654d61614be1ada9148f961b180ab9aa77b82>fixed fingerprint plot and multiQC</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e1795c0010ecf0b0424b2ab87001811f0729e95f>Fix: WES Slurm Master Job Failure</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a4e2cb1f64b889e81198d77f381f5c0f143cd043>WGS/WES Reference Genome Selection Warning Message #377</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/59531998a77c483159eb98c8b219a4398fe16d19>Merge pull request #376 from tovahmarkowitz/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7666d4247cf18ed25b8a6bac069c99ebfe2afce4>Update cluster.json</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9309efae58873bb259f711a65fcc2ce9c26634a0>Initial ChIP-seq QC DAG SVG Generation, Updated Peaking Calling Rules, Generated MEMEchip Reference Files #375</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4a33bd74406bf1e7a846cddc037c68f4c03abaf2>fixed ppqt</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4c6c5fb844a29ecbbb87feb93c70be212cd5093f>bak files removed; cutadapt updated to 1.18 on rnaseq</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3ab0d0e8be537628e214c51d485ab5256e943468>fixing .gitignore</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/85da38684a879382f8794a23670c5dc1e4b67b5e>Fixed ChIP-seq Python Package ImportError #374</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/477aecebac322aae9ed8fabee361baa2961a183f>Fixed ChIP-seq Python Package ImportError #374</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/21d2437a10f87bc68d3272d7e074a5e53c4b8b3b>Merge pull request #1 from vnkoparde/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1e4ead78963514acd67bc5a4475cbcf74ac6d04a>changing cutadapt in initial chipseq qc snakefile from 1.16 to 1.18 and adding a ppqt_pe rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/63d4f46f09c32a8e0197b784545c43901f0c18b2>actual purge and adding .DS_Store to the list</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/601e537fe96068d3f6884aa028342effcdf24636>adding .gitignore to ignore .* *.bak and *~ files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2d6487a2deb77a8defd1ce24cc0b594af2accfaa>standard-bin.json changed to change version of cutadapt from 1.16 to 1.18</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ece9a57576710bcb6f628cde0cc1b1d977aa057c>Rolling back InitialChIP Seq snakefile to commit ac47e92a20af8de3fe1ddd9344c613dc38f9cf63</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ef995f94a14b712c8c0d8f30f97fb6f99e8f0f30>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/461027f4cdb0d88194b26e94d0b51b8d6600ae8d>In PE mode readlength based filters to be applied to any (not both) of the the reads</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ea998d9f4c53c9a4a3a443df76be5e47997c2484>Merge pull request #373 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/44e3d03c7b505c96aa834051b95395339c56c395>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b31dc57bc811ec0d650169087732f10828512777>added automated filtering to prep_mafs.pl script</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e9e1d6780ae14ff29fb66ff7ba64c3a618b0207c>Python Package Import Incomaptiblity Fix: pysam</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/62f008ad32ea8bb5ab1493845233bc812d00ee39>chipseq:cutadapt to v 1.18, remove files ending in tilda, added .gitignore, added separate ppqt rules for se and pe</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/362e3e8d3abcfec55e358ef0b56deda3a767cfbc>Labels Testing: unbound variable fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ac47e92a20af8de3fe1ddd9344c613dc38f9cf63>updated genome fasta for rnaseqvarcalling for mm10</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0ba2b674f05608e82f710a087e0bc4fbfba3ece6>Labels file edge case: peak call pop-up box fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/efa6a25ba16431fb9b6d9195b8798dc6fd85c84a>Copy peakcall.tab if labels.txt in rawdata directory</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5549023baf23f5250cc6d00d19d4c4bce91bd2a6>merge with main pipeline gerge th otherbranch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c394c4edca32714dcf599391a8df3b24231f6300>ltered somatic variant filtering to be based on alt and ref counts in tumor sample, instead of allele frequency</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/dc977d6ceb9395498ade7b88cb4551eea60e8d28>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/606964e162874ef7ada04b36fec25160a28444d8>Explicitly Loading Snakemake Module #371</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/34064433a4903f8ffdc55e26b4f037b97b94ae09>Explicitly Loading Snakemake Module #371</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/29b8db54469f4aa770335131364063b6bc36f0ed>fixed somatic directory creation bug due to snakemake version change</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/be1bbe28f97573c990368edc5da03ef72be85268>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0039502a264de3c2cbfdf0eacd1fc661ad285485>fixed bug in loading VEP and vcf2maf for MAF generation rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/51b1e823bcf13e3d403262c31113b6d445c93104>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev combining pipelines :gibincomninge enter a commit message to explain why this merge is necessary,</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/dff8bae3f44b179f68664a31145b94d3834d2d1d>separating fingerprint plots; merging inputnorm heatmaps</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2f266a670c4ebb1524d45570a18eb6a7fbcf5d4e>Update README.md</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/33be5dbf0e4ec9cc99b64c38e4562d6f956c7246>Update README.md</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/def151e7c92b284a6694899905d19f1c9a2a7118>readding geneinfo to chipseq</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e51d90ac647119adb7d7a47b4fc78ac037c62393>fixing merge issue</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/16219046e827ba998ab6beba3e015c483a269716>merging with CCBR/activeDev gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7e072714e59e5607d2d2987d660e393346ce079d>Dynamically Resolve CCR BUY-IN NODES ISSUE #358</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b1953058609d3758162916c036dc780891a1f8bb>Dynamically Resolve BUY-IN NODE Issue #358</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/576bb9a442ea36374beeaec0872011ea1dd441dd>RSEM CPM filtering Issue #357</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f2a9b3ae891659e3fd51177385c96f1c737ca211>json merge issue</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/68d45f104192a96196db1e1b39671d899b926642>merging json files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/687ceba6b378603cbee39728d5897b024a45f9b7>Removed an include statement</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/872c2a92f39c5c748999bba9b793ba96743bf462>Updated GEM GEMONECHRS</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d3291219d322487ed5f53ffb9c5c0e8f0bd6c9f0>Merge pull request #356 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/076782d08c6c052b860c16d8dc3ece9820d74aa8>updated splitNcigar GATK version to 3.8</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9cbdb92f6e024864fe67e54be095d8da5f0d3da8>Re-validated cluster.json file</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e14d71a7c76bdd672f771648ed6b3f77f6aad6c1>fix error with read extension</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2d83f280c2ab038938a1a8d0f51f793f8aa65186>gerge with Justin's changes branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/86941fc5dff331b18b8c7ac2800219c44675b307>Changing Partition from 'ccr,norm' to 'norm'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/22b95b3b14bb4e1a3c931e36f7b3a3aae86e8a97>Merge pull request #355 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/df80c7381117fe05a6eb23d685771172e60f095e>reduced wall times for rules that do not need max wall time</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/effb13bfc3331e534b6979b9a9f8d193097b33f9>changed default partition to norm</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c75005ea5ca75694bc636a8ca6ec8169d7afe475>renaming bws</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e4bc04612dc0078dc72c1ef3b21cc3e73a531927>changing bw naming</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/411a99e8d7ca60f0274d8f25e244d05d24024b18>Update Deeptools PE Target Rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4e30326b1b7b9b1e1a2348bf5414f8ca7313aafe>Resolving Deeptools Filenames</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/75ee4a1bc08f2d3a1f1799ed697c8f4a5871382a>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8eeb32685dacf72839cd05424c83e62301025b54>added indexing rule to all WGS and WES pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0f845a9efe010a0f8e8d8230d8ab60848ce486b7>playing with dependencies</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/add0d9c24db8481734d130cd7bb55ea7c958af65>merge with other pipelines branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a508b77f8fa8d23b0af5b86b90afb47d4f384dda>Merge pull request #351 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/94fa796e8b7cf398a23f413d70126788482b8a2c>repaired wgs_tumor_only bug in maftools</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4bb47b673cf187dbe5b8b6c4eef343accb6204a4>merging ChIP-seq with other pipelines gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fa07df2342641518e268e1c56286b12279ebb0e1>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/24e800643530139f2d915d2304dbc0d414f4bb14>CPM Filtering Fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8f0f3cc5cff240260eb01184a4a0706adbcf951a>Use Updated Sampletable for PCA</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/62e3420d7caf3545b4cd65dd43d766962a22cd29>Export RSTUDIO_PANDOC</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1072a13ff9a11087f59e2702f9155be0110bcaa3>merging with Paul's function gerge branch 'activeDev' of https://github.com/tovahmarkowitz/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/267383c87718b62c3b3bbafec4ce03b4891e2501>merging with Justin's changes gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3819b9e1ff8b098dfc2d29ab06672f74174ece83>Inputnorm rule integration</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3878a8a170a4ab8a8d8e5862b3d0e0dfd2039064>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bdf2115b917fbac160d2cebdf84bfed9b9ee41b7>increased master job wall time to 10 days</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c22bf7c0c28e43f243331172c16b03fc34327a7a>Merge pull request #350 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fe124f6216d1113b4db33b95536f22dc4232a905>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ea70d094b86d27a50186312ae688438c6b8bf711>made theta2 outputs dynamic to allow pipeline to complete under conditions where the theta2 rule fails</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3c5c76609c738466830f4cd3168567fc10706037>read extension for bigwigs and dynamic deeptools rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f94657f4dfff19344f816122eedb02e57d4d7812>merge with Justin's changes branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cc04a483a8af4fd711d19be735c8e8da5eb326f7>Merge pull request #349 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/682850331d0d66164b898a7388c3c1b88c53ecde>added chr prefix to hg38 chromosome names in haplotype caller rules split by chromosome</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0c5b04437ec286481a1fc864eb175738cab00666>merging with contrasts.tab changes Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8a9987e7cd72637e7bb8fe9326f85ed8b4fac96e>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c96c899585e17a725b52614613fb86142f4d9581>added filtering for MAF files, and added additional summary plot and analysis of MAF files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/419bc3364f902ef9ad05e484fced5a1ae4648a83>Safe-guards to ensure referential integrity during differential expression analysis</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/11e049bd5fb037c28740ffe60f1aa0595d3ae178>Updating rule pca</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1bfadf14bb213f3c2406154889f72b0a2639344f>adding heatmaps and profile plots</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c05f54c39fd023ecc482681e2c4547349d987c35>merging changes with ChIP-seq and RNA-seq pipelinesMerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b97bf163dbb65306b536ed83f7651bda5a193416>Pop-up box warning if 'groups.tab' or 'contrast.tab' are not in RNA-seq working directory before dry-running</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d908ec2ade22ffb9ae87fcc62dd6c84814b12fe7>pcacall.R revert</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1a6c66e27018f7193e190f2c6a80d5bd183108df>removed some pointers</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bc7de0614a3bc0ad0816947841acaa3dd68757b2>changed conditions for trim_pe and trim_se</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6967f367dd7689f79c0ed152a2dac0e892018fd9>combining structural annotation information and chip-seq file pointers Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f47775321a5672d58ec15f95e0e12c56ab55d595>Merge pull request #348 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/be80367e63d96d36a88efbe031a54532c409924f>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/50fe1e87836a07110a8afbd0f4d6b567a040f1e0>split rnaseqvarcalling by chromosome</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d3a1e5c44a34970ed9e7a441b8f8f7aad4c61797>Adjusting different rules in IniitialChIPseq.snakefile merge branch 'activeDev' of https://github.com/tovahmarkowitz/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0b029e8d67dafe96d0f1cf3d316d0dfb8dfd6ae0>changes made to both exome-seq and chip-seq portions of hg19.json Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cb3b65a13d6b3aab09461955260539c07c3a46a0>Merge pull request #347 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ce7629176ef5f981a583944be80a7773a4100325>fixed bug in location of fusion STAR libraries</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fb15dd2423b4ad059204633ef9dcf3d895735691>add peakcall.tab parsing and rule inputnorm (not implemented)</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/825ac64a2ffe0a4ff3dc38773429e573c6f2727c>new heatmap rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/120ab94bc8f7ec1d5227df346d70c75310065cd2>adding GENEINFO to ChIP-seq</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/17243949e4ad167fcd735f5a586829fcca27aee0>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bdb8a22396742481b96345416a6a9a7719056246>fixing thread usage</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fa5b8b50ad581df0232b506e51ca144676337d1d>Merge pull request #345 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a4a0f20d24eea0858cddda6c8b3d2a00ad811eba>'peakcall.tab' is copied over if it's in the data directory</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1fd121f67aad4e8a4b2fdee5ece6dde882bffea0>bug fixes with deeptools_QC</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/56aeedd76daff7500849d3c75e8596551caf423f>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9c9b9e7d4d909bd16fb723c92fd0589310340b05>increased the number of jobs submitted at a time to 500</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7c6c0f2a2832d26acc9f74c1b48ea489e9f2a0b5>bug fixes</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0afe04bc1d30b908dfd70b0767863d319e470611>adding fingerprint plot</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4d8dd35dc9daec3ef7fd9bd9ac12b9c76d40f9df>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev changes to RNA-seq merged with changes to ChIP-seq:</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3dfbefb8a76ce2f9888f27fd0678930a38126d79>Merge branch 'kopardev-activeDev' into activeDev 344 pr updates</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/33fc78ad5705cef0e27e8b3fbb707cf4cf606102>344 pr updates</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/39f6ca37203c01be0f9d50006981da57ad6157c7>split haplotype caller by chromosome for all germline pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/3ac58f5c781686c27d5f6b85625142b9d0ee4b0c>removed Delly from all pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ebde723c4359cac3461b180aa962901bc4a30ce7>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/350550c915300901e89d9159d22d9567e7bbc20f>added knwn ploidy config file to mm10.json for running Canvas</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cee89f8014fd96eadb0e07a36d0128377ee2e64d>Rename symlinked bams to .recal.bam</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/21273c1057449b45c0cdab3c6169013595956cf2>Genome-seq/Exome-seq symlink update to work with labels.txt file</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/02d518d38a92ae5a6e3c89339162a9dc0503ad54>fixed bug</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7e6e9778e6b31660b000ffa3f0dd1d6291bd70e6>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f0b040278bce738c6bdce5fd38637fd39bb4268a>GUI Changes: peakcall.tab now required for InitialChIPseqQC</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/816c2d558321821a5e17ff23d091e6da79aa7183>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b5e81e38df6c273ab76b9b6acff63aa3ec4f4574>Symlink BAM and VCF files for Exome-seq/Genome-seq</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d81826d78cc707a5488307dd1fa3e41aac6d1314>fixed rule connections</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b53e16ca36bb016b407c39d4d2e2ce9a082eb767>fixed indentation error</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/895e0001b2d1fc1449cd77e4e5b3f874d1c42210>organized rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/82af20667f56d5d262e2d8a27d9c2f399aca9108>cleaning script for clarity</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d96a6524883deb3e9edc62b44754f0bf31f4c718>consolidating rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d9d07c5c883c5a66ab0f3c2061f644a74ee4d51f>updated MallDoneerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bf2950de67f085c99ac011aee006d93d4dfbb704>repaired starfusion bug and hg38 exome somatic only bug</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/385073abfd085ef76e01db6858ef894839aa877c>cleaned up code</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2e38f57d96a07d53b0d4150c33ad95252c7cd526>Merge pull request #343 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5dfa18233205ce3d716efebe93b7506e704f174f>updated versions for STARfusion and Canvas, and updated annotation resources for STARfusion to the most recent. Also, added multithreading to STARfusion</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/267e4a4d2126ac84095cc48b77a749e22a70284e>fixed bug in all rule for wgs somatic tumor-only</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/aba6142612056b7c7a31429feaaca9a0abeb6cb5>stepped back to v0.9.1 for cnvkit due to python bug</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fe333e21ed99400792da35e5cee976fe101bbe9a>RNA-seq DEG changes</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/20bf985143d5028ec019f8e17abd754201da74a3>Syntax Error: removing extra parentheses</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/41304a9a036abc47bbba777c7fb3741f3212dfff>Undo reading in files in binary mode</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1bf7a8ccf4bdfcc5886693f37b89aad5e8f20b3b>Low Abundance Gene Thresholds option removed from the DEG gui frame ... this data is now entered in the contrasts.tab</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/e6653a4f8a299abd70f1edf387e60101d8db61f9>rRNA list fixed for mm9</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c4211e897e62c5f1e8f607a5e65a252ec5a1dd10>Merge pull request #342 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2a7e1cbdcfa0cd3e20693ac2a9b77f2f1b1337c8>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/037939d6f5f5032dc383df44e78b1f596a5e5a35>updated snpeff genome version for mouse</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1b52c7dcc7dcf7b39cb1eee11abbb8435a9d14d6>Merge pull request #341 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1e341699edebcb0bc3d8d9365d1c1ebeb43af1a9>changed hg38 filter ExAC VCF</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1a13002ef70a5c7df881a78c370ad56f87636328>adding warnings for selecting the wrong genome ... for eg. GRCh38 will only work with scRNASeq pipeline ... pop up warning if it is selected for other pipeline</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a7d19c09413ca94ca0a4196de39729b73f29a3e2>adding decoy genomes for RNASeq/ChIPSeq pipelines ... updating annotations for other genomes</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/888b362cd3aa429375fce135fd225e03d5d3406e>new mouse RNASeq and ChIPSeq indexes included in json files</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4f7740403c4797636cc5fc40e373a91252920f23>Fixed indention inconsistency</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b468c3546e325a909a683a413b518029a6f477cf>QCTable NReads Fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1f3f8b38b688bb58c8d479b9b66169d3fd529e5d>add sample info to initialrnaseqqc gui; added mincount column to groups.tab</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9f8f08ed481f8463ad57564ca1877460d2b621cb>Merge pull request #339 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/977123b0f6998f7a50c5045cbd320b28f68deb1b>updated hatplotype caller steps to GATK 3.8-0, fixed jdk inflater bug, and added index BAMs step to Dragen entry point</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/091b13f96f2feeee6e8d0e7f8deacc75a18a5f8c>Adding peakcaller GEM</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/477715accec22eb3398a80bdc69da8a780693e22>ChIP-seq QC Table</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8711aeb536f667377d5f2b7618742a1e3179261d>Merge pull request #338 from jlac/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1da27b402e2b9166470879647fac08562becfb18>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev updating</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/98b2e8b755e90b7a20d2e448aa3df7a97f3f840e>updated Canvas version from 1.31 to 1.35</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/dc3340fc10f7c278367ec908980740a6c2596099>ChIP-seq Multiqc: Samtools Flagstat and Fastq Screen</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/328b92c9d68e3383755a23e8e2379101e7c524c4>Delete PipelinerVer1.0_documentation.pdf</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/23524d2e3872e65a5b2b6f6ad9a93ba1dd9e1024>Delete PipelinerVer1.0_documentation.docx</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2bf2ab1142f29ff2ada7eec701b80d6e7c3d6825>Delete PipelinerVer1.0_documentation.pdf</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d73e9c66f8d778e6d2cfa73a2971c75b23592280>Delete PipelinerVer1.0_documentation.docx</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/16423fb408024a300649a77144a9ae2059981a84>Filename bug fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/929c83c8218e651b19a54ea675d652136de283f5>Snakemake DAG generation fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/865540fd1d5deb029d74db0794d780d5ce649323>filtersample scripts</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cc758c1b3dfdf7c2191131f7629191ef4e83e496>DEG_ALL folder ... extra column for contrasts.tab to have different cutoffs</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/2ae45460ea2d74c90a0cf38c65300cf274409417>increased memory for qualimap rules</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4672f7027fe0c7f2dc7760999ed57db6173341f5>Label Name Fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/d4a80ff4f438561f57860b8c20c05680e0a22c3b>Merge pull request #332 from CCBR/revert-331-activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1195565af206a7a3c6ce400c615490b6a38fd4fa>Revert "test commit"</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/879b5b4a525d4464d4bebc41c4729432e03b67d3>Merge pull request #331 from pajailwala/activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6ce187a3e17b3044e179d47f7b7dc26e99c6681b>test commit</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/882aa1c0634f53e65307fc8428edf41037dbaaf3>Venn Diagram Additions</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0067f30df7c99b9edc01a568f77c35cdd2948fd2>deleting test</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/963bc331ac818776dca6fd8ed05e6ebfbff9f131>Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1572a568139da88a7701f13a0fb5aaca726f4e3f>Update README.md</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/311f923777f75ceba8d2f2d06370732037ab02fe>avia curl request fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/995d727f6522206d367634aa5ef345d3b5d7cebf>test2</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/50741c9078b46128dff383517b8f6ad9e518638a>testing</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fc7b6f548b2a5dc75a612c47246d42890a0b18e4>Venn Diagram showing overlap of DE genes</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7bce1390aa0733ff1db53c5351bd74be8ef580dc>Reformatted Limma, DESeq2, EdgeR Output: RNASeq</a></li> </ul> <h2 id=v305>v3.0.5<a class=headerlink href=#v305 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/bfa24e22f518cdba5c527783ebb9476e2e284893>ChIP-seq bug fix and gui additions</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a68a919703fb547be3fcf0a92bcc3443dd65675f>Merge pull request #322 from nikhilbg/centosOS7</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f6290f124eefcb99d980b712e554d1b431c81328>minor bugfix for annotationdbi usage in scrnaseq</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a05b313e0e5fd86a3180af4d73d589cda2cc4e3f>cell cycle regression option in scrnaseq</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/5bacec9ffacbf2624d539f1c327e4fb7f12e72c6>Nihkil changes for Single Cell</a></li> </ul> <h2 id=v304>v3.0.4<a class=headerlink href=#v304 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/9bea53ed92bc793d65afb2c7cad3b2b47d8caf6f>centOS7 ChIP-seq compatibility and requested GUI changes</a></li> </ul> <h2 id=v303>v3.0.3<a class=headerlink href=#v303 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/6516d590f56807e9be011a7792e7057fc7ad03c4>WGS and WES changes for centos7</a></li> </ul> <h2 id=v302>v3.0.2<a class=headerlink href=#v302 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/24784bef0b5eeb3eeed9ea72426312de825298b9>Skylers changes for centos7 / RNASeq</a></li> </ul> <h2 id=v301>v3.0.1<a class=headerlink href=#v301 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/3fa25a97851510c7c174b5bde3743824e140bc9c>bringing branch to the version on ccbrpipeliner (v3.0.2)</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6a081055bf028d3845a6ae34d83feb7878071a31>bringing upto v3.0.2</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b2863296fb2c4eb2e897125d08bcc147e4c91196>v3.0.2</a></li> </ul> <h2 id=v300>v3.0.0<a class=headerlink href=#v300 title="Permanent link">&para;</a></h2> <ul> <li><a href=https://github.com/CCBR/Pipeliner/commit/997b26d11be674d3f7c3d173fc2606d215dcfa49>multiqc is now not running in DEG part</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/61972bafcc918c8259af83e4f3a18436267c6b1c>Merge pull request #316 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c58223a3eb123bdbaf4f0e737e54775b13ca059d>updated versions for cnvkit and manta</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f31de25c8785f6c1516e56b708e0d001008754d6>Merge pull request #315 from abdallahamr/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cf80d51fffa5022c9cedebf8092ede7b1d0c1b18>This is for the cpm filtering as it was not doing cpm twice, I only commented out one line per file</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/796b75389a6efee6cf0e7202f7c33299194d84e8>Merge pull request #314 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/227067f6f2051079c279aca9695776f475a45578>removed realign and recal from WES somatic and germline calling stages of WES pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/cc75e682b610d99c4042cd0655d008341443f0ec>Merge pull request #313 from CCBR/v3.0branch</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ba1622cc66aa305f9a3bbb9d5f95c5356f3a7b92>data/work folder is copy paste on gui... workflow error fix</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c22456bd304e298ebb6d8c79039ba52378da92e8>Merge pull request #312 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a2556171ea5dd438f394446d248b05b4f35a85e8>Merge remote branch 'upstream/master'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7214b76ad4ccf70f6459d81bff4a2064eb404905>updates version of multiqc to v1.3</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/0234876f789e9cd214b9504fd34f7a2d35075584>Merge pull request #311 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/49d4c11e11a6c2be969839f2162d5bbd9d08d5c7>fixed bugs in fusion expression cleanup</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f8889fef7023427e0bcae50ed990d26f4d227432>Merge pull request #310 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/4694b4e3bb9f1cc4dea3a09f5cb69c915602139b>Merge remote branch 'upstream/master'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/9f080d856f31e4b449262fbdbd391c110f69dcd4>changed path to Oncofuse jar file</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/01a8fe866279d1bc7fadf9a2b5ba3e8d3783f3ff>Merge pull request #309 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/86b20b4056851ef76ddd3f77a3ed3ad5e86a6020>removed subread components from gene expression rules for fusion pipeline</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/a7ac920c79a0e6acf10ffc7ae3af11e950c71643>Merge remote branch 'upstream/master'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/47319f6f12935dcb0c3e7fe8ac7f361cf18c71d4>changed filtering criteria for MAF filtering</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ef31f08b9176d8c5d77b1692af3c7a0a15756c6a>fixed bug in processing tumor-only maftools rule</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/1706cd0d1ea9aafa641172d88b5407f3306a7efb>Merge pull request #308 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/99045db3f413b9f1286cfe2e6f823c1e12662973>Merge remote branch 'upstream/master'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/f6614e8c3a6599f4083f80ca36699437703ecd55>added filtering to somatic mutation calls; fixed jsons to repair missing resources in fusion and rnaseqvar pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/fc70f228034685fc8d9345729cd3f811fe21eec4>Merge pull request #307 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/7df59ea61cce96dedcd89e4839aa870c212437ad>fixed Canvas version module</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b13808eb1a8436bf5e39f0e90f7df60b4d727d79>Merge pull request #306 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c25067a04304f21c66913bcb404a23effcbf034e>repaired missing reference resources for fusion and ranseqvar pipelines</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b93793dc84b38fdc0afd2fd58ce2ff9c3dc10638>fixed bug in counting sample for mutsigCV</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6b1c9888433b2c6dd1c304d7b89f37a36837a453>Merge pull request #305 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/c2f0cd759a64251e807c2c48bbab770b91a70771>Merge remote branch 'upstream/master'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/ebf060eaf7ad065f1bcf1c99e4c883940aedec3b>added population allele frequencies to MAF filtering, and set mutsig a threshold of 10 samples or tumor/normal pairs for mutsig to run</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/35e747f773fd4fa2e52f7a6e374854b6b506d052>Merge pull request #304 from jlac/master</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/6492220d870c29fc978c500f747eb5fefaec88a7>Merge remote branch 'upstream/master'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/b45c833d909be723ba7ce379bf8ce56684ce994a>fixed X11 graphics bug in qualimap</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/8818d0e58ec1c9b068d90dfc38eb7acb0a01ea80>Merge remote branch 'upstream/master'</a></li> <li><a href=https://github.com/CCBR/Pipeliner/commit/bf6ceb7742650662d128db7928c339ea1fb0a7d5>removed extraneous file</a></li> </ul> <hr> <div class=md-source-file> <small> Last update: 2022-11-04 </small> </div> </article> </div> </div> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../community-and-contribution/ class="md-footer__link md-footer__link--prev" aria-label="Previous: Contribute" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> Contribute </div> </div> </a> <a href=../contact-us/ class="md-footer__link md-footer__link--next" aria-label="Next: Contact" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Contact </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2020 CCBR </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=http://bioinformatics.cancer.gov target=_blank rel=noopener title=bioinformatics.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M319.9 320c57.41 0 103.1-46.56 103.1-104s-46.54-104-103.1-104c-57.41 0-103.1 46.56-103.1 104-.9 57.4 45.7 104 103.1 104zm50 32h-99.8C191.6 352 128 411.7 128 485.3c0 14.8 12.7 26.7 28.4 26.7h327.2c15.7 0 28.4-11.9 28.4-26.7 0-73.6-63.6-133.3-142.1-133.3zM512 160c44.18 0 80-35.82 80-80S556.2 0 512 0c-44.18 0-80 35.82-80 80s35.8 80 80 80zm-328.1 56c0-5.449.982-10.63 1.609-15.91C174.6 194.1 162.6 192 149.9 192H88.08C39.44 192 0 233.8 0 285.3 0 295.6 7.887 304 17.62 304h199.5c-20.42-23.8-33.22-54.3-33.22-88zM128 160c44.18 0 80-35.82 80-80S172.2 0 128 0C83.82 0 48 35.82 48 80s35.82 80 80 80zm423.9 32h-61.84c-12.8 0-24.88 3.037-35.86 8.24.6 5.26 1.6 10.36 1.6 15.76 0 33.71-12.78 64.21-33.16 88h199.7c9.76 0 17.66-8.4 17.66-18.7 0-51.5-39.4-93.3-88.1-93.3z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://hub.docker.com/orgs/nciccbr/repositories target=_blank rel=noopener title=hub.docker.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.b028fd86.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../assets/javascripts/bundle.0f659f14.min.js></script> </body> </html>