<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Pipeliner is an open-source and scalable solution for analyzing  next-generation sequencing data. Pipeliner provides access to  the same best-practices NGS pipelines developed and benchmarked  by experts at CCBR and NCBR. This repository is the main source  of documentation for users and developers working with or  contributing to Pipeliner."><meta name=author content="Skyler Kuhn"><link rel=icon href=../../assets/favicon/favicon.ico><meta name=generator content="mkdocs-1.4.2, mkdocs-material-8.3.0"><title>User Tutorial - Pipeliner Documentation</title><link rel=stylesheet href=../../assets/stylesheets/main.fd896c8a.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#getting-started-with-the-quality-control-pipeline class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="Pipeliner Documentation" class="md-header__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Pipeliner Documentation </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> User Tutorial </span> </div> </div> </div> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="Pipeliner Documentation" class="md-nav__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> Pipeliner Documentation </label> <div class=md-nav__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> Home </a> </li> <li class=md-nav__item> <a href=../../about/ class=md-nav__link> About </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3> <label class=md-nav__link for=__nav_3> Getting started <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Getting started" data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Getting started </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1> <label class=md-nav__link for=__nav_3_1> Users <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Users data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Users </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Prerequisites </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Installation </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <label class=md-nav__link for=__nav_3_2> Developers <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Developers data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Developers </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Installation </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Customization </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4 checked> <label class=md-nav__link for=__nav_4> Pipelines <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Pipelines data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Pipelines </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1 checked> <label class=md-nav__link for=__nav_4_1> ChIP-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=ChIP-seq data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> ChIP-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> User Tutorial <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> User Tutorial </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#getting-started-with-the-quality-control-pipeline class=md-nav__link> Getting started with the Quality-control pipeline </a> </li> <li class=md-nav__item> <a href=#about-the-demo-dataset class=md-nav__link> About the Demo Dataset </a> </li> <li class=md-nav__item> <a href=#phase-1-initial-qc class=md-nav__link> Phase 1: Initial QC </a> <nav class=md-nav aria-label="Phase 1: Initial QC"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#step-0-fill-out-the-project-information-section class=md-nav__link> Step 0. Fill out the Project Information section </a> </li> <li class=md-nav__item> <a href=#step-1-setting-pipeline class=md-nav__link> Step 1. Setting Pipeline </a> </li> <li class=md-nav__item> <a href=#step-2-select-genome class=md-nav__link> Step 2. Select Genome </a> </li> <li class=md-nav__item> <a href=#step-3-select-your-data-directory class=md-nav__link> Step 3. Select your Data Directory </a> </li> <li class=md-nav__item> <a href=#step-4-select-your-working-directory class=md-nav__link> Step 4. Select your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-5-initialize-your-working-directory class=md-nav__link> Step 5. Initialize your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-6-set-peak-information class=md-nav__link> Step 6. Set Peak information </a> </li> <li class=md-nav__item> <a href=#step-7-perform-dry-run class=md-nav__link> Step 7. Perform Dry Run </a> </li> <li class=md-nav__item> <a href=#step-8-run class=md-nav__link> Step 8. Run </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#check-progress class=md-nav__link> Check progress </a> </li> <li class=md-nav__item> <a href=#confirm-successful-completion-of-phase-1 class=md-nav__link> Confirm successful completion of Phase 1. </a> </li> <li class=md-nav__item> <a href=#phase-2-peak-calling-differential-peak-calling-annotations-and-motif-searches class=md-nav__link> Phase 2: Peak calling, differential peak calling, annotations, and motif searches </a> <nav class=md-nav aria-label="Phase 2: Peak calling, differential peak calling, annotations, and motif searches"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#step-1-initial-setup class=md-nav__link> Step 1. Initial Setup </a> </li> <li class=md-nav__item> <a href=#step-2-select-pipeline-options class=md-nav__link> Step 2. Select Pipeline Options </a> </li> <li class=md-nav__item> <a href=#step-3-set-contrasts-optional class=md-nav__link> Step 3. Set contrasts (Optional) </a> </li> <li class=md-nav__item> <a href=#step-4-perform-dry-run class=md-nav__link> Step 4. Perform Dry Run </a> </li> <li class=md-nav__item> <a href=#step-5-run class=md-nav__link> Step 5. Run </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#check-progress_1 class=md-nav__link> Check progress </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../ChIP-seq-output-files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../ChIP-seq-tools/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2> <label class=md-nav__link for=__nav_4_2> RNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=RNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> RNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../RNA-seq/Theory-and-practical-guide-for-RNA-seq/ class=md-nav__link> Theory </a> </li> <li class=md-nav__item> <a href=../../RNA-seq/Gene-and-isoform-expression-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../RNA-seq/TLDR-RNA-seq/ class=md-nav__link> Quick Start </a> </li> <li class=md-nav__item> <a href=../../RNA-seq/Gene-and-isoform-expression-guided-tutorial/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../RNA-seq/Differential-expression-pipeline-tools-and-versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <label class=md-nav__link for=__nav_4_3> miRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=miRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> miRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../miRNA-seq/overview-miRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/TLDR-miRNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/miRSeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/miRSeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <label class=md-nav__link for=__nav_4_4> scRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=scRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> scRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../scRNA-seq/overview-scRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/TLDR-Single-Cell-RNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/scRNASeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/scRNASeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <label class=md-nav__link for=__nav_4_5> WES/WGS <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=WES/WGS data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> WES/WGS </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/TLDR-WEG-seq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-output/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-tools-and-versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_5 type=checkbox id=__nav_5> <label class=md-nav__link for=__nav_5> FAQ <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=FAQ data-md-level=1> <label class=md-nav__title for=__nav_5> <span class="md-nav__icon md-icon"></span> FAQ </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../troubleshooting/ class=md-nav__link> Troubleshooting </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Data Privacy </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../community-and-contribution/ class=md-nav__link> Contribute </a> </li> <li class=md-nav__item> <a href=../../changelog/ class=md-nav__link> Changelog </a> </li> <li class=md-nav__item> <a href=../../contact-us/ class=md-nav__link> Contact </a> </li> <li class=md-nav__item> <a href=../../license/ class=md-nav__link> License </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#getting-started-with-the-quality-control-pipeline class=md-nav__link> Getting started with the Quality-control pipeline </a> </li> <li class=md-nav__item> <a href=#about-the-demo-dataset class=md-nav__link> About the Demo Dataset </a> </li> <li class=md-nav__item> <a href=#phase-1-initial-qc class=md-nav__link> Phase 1: Initial QC </a> <nav class=md-nav aria-label="Phase 1: Initial QC"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#step-0-fill-out-the-project-information-section class=md-nav__link> Step 0. Fill out the Project Information section </a> </li> <li class=md-nav__item> <a href=#step-1-setting-pipeline class=md-nav__link> Step 1. Setting Pipeline </a> </li> <li class=md-nav__item> <a href=#step-2-select-genome class=md-nav__link> Step 2. Select Genome </a> </li> <li class=md-nav__item> <a href=#step-3-select-your-data-directory class=md-nav__link> Step 3. Select your Data Directory </a> </li> <li class=md-nav__item> <a href=#step-4-select-your-working-directory class=md-nav__link> Step 4. Select your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-5-initialize-your-working-directory class=md-nav__link> Step 5. Initialize your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-6-set-peak-information class=md-nav__link> Step 6. Set Peak information </a> </li> <li class=md-nav__item> <a href=#step-7-perform-dry-run class=md-nav__link> Step 7. Perform Dry Run </a> </li> <li class=md-nav__item> <a href=#step-8-run class=md-nav__link> Step 8. Run </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#check-progress class=md-nav__link> Check progress </a> </li> <li class=md-nav__item> <a href=#confirm-successful-completion-of-phase-1 class=md-nav__link> Confirm successful completion of Phase 1. </a> </li> <li class=md-nav__item> <a href=#phase-2-peak-calling-differential-peak-calling-annotations-and-motif-searches class=md-nav__link> Phase 2: Peak calling, differential peak calling, annotations, and motif searches </a> <nav class=md-nav aria-label="Phase 2: Peak calling, differential peak calling, annotations, and motif searches"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#step-1-initial-setup class=md-nav__link> Step 1. Initial Setup </a> </li> <li class=md-nav__item> <a href=#step-2-select-pipeline-options class=md-nav__link> Step 2. Select Pipeline Options </a> </li> <li class=md-nav__item> <a href=#step-3-set-contrasts-optional class=md-nav__link> Step 3. Set contrasts (Optional) </a> </li> <li class=md-nav__item> <a href=#step-4-perform-dry-run class=md-nav__link> Step 4. Perform Dry Run </a> </li> <li class=md-nav__item> <a href=#step-5-run class=md-nav__link> Step 5. Run </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#check-progress_1 class=md-nav__link> Check progress </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/CCBR/pipeliner-docs/edit/master/docs/ChIP-seq/ChIP-seq.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1>User Tutorial</h1> <p>This page is the main source of documentation for users that are getting started with the ChIP-seq pipeline. Please check out our <a href=https://github.com/CCBR/Pipeliner/wiki/ChIP-seq-pipeline-tools-and-versions>pipeline resources</a> page for a comprehensive listing of supported reference genomes and different tools the (with version information) pipeline employs.</p> <p><strong>If you are a new user</strong>, we recommend following our <a href=https://github.com/CCBR/Pipeliner/wiki/ChIP-seq#tutorial>Tutorial</a> with the provided test data set on Biowulf. If you are a new user and you would like to skip our guided tutorial, please see our <a href=https://github.com/CCBR/Pipeliner/wiki/TLDR-ChIP-seq#quick-start>quick start section</a>. </p> <h3 id=getting-started-with-the-quality-control-pipeline>Getting started with the Quality-control pipeline<a class=headerlink href=#getting-started-with-the-quality-control-pipeline title="Permanent link">&para;</a></h3> <p>This section provides step-by-step instructions for setting up the first pipeline (Quality-control pipeline) and a brief description of the pipeline's output files and directories. We assume that you have already successfully launched the GUI. If not, please see <a href=https://github.com/CCBR/Pipeliner/wiki/TLDR-RNA-seq#quick-start>Launch Pipeliner</a>. Also, in order to use CCBR_Pipeliner on <a href=https://hpc.nih.gov>biowulf</a>, you will need an active biowulf account which can be requested <a href=https://hpc.nih.gov/docs/accounts.html>here</a>.</p> <h3 id=about-the-demo-dataset><strong>About the Demo Dataset</strong><a class=headerlink href=#about-the-demo-dataset title="Permanent link">&para;</a></h3> <p>For the purposes of this demo we have chosen a subset of a publicly available dataset from GEO (<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE76478">GSE76478</a>). The origin datasets contains CTCF and H3K27ac ChIP-seq experimental data at early and late stages in iPS cells. Here we will focus on the following samples:</p> <table> <thead> <tr> <th>GSM number</th> <th>SRR number</th> <th>Sample</th> <th>Cell type</th> </tr> </thead> <tbody> <tr> <td>GSM2026234</td> <td>SRR3081748</td> <td>CTCF_ChIP_macrophage_p20_1</td> <td>macrophage derived induced pluripotent stem cells after 20 passages</td> </tr> <tr> <td>GSM2026235</td> <td>SRR3081749</td> <td>CTCF_ChIP_macrophage_p20_2</td> <td>macrophage derived induced pluripotent stem cells after 20 passages</td> </tr> <tr> <td>GSM2026236</td> <td>SRR3081750</td> <td>CTCF_ChIP_macrophage_p3_1</td> <td>macrophage derived induced pluripotent stem cells after 3 passages</td> </tr> <tr> <td>GSM2026237</td> <td>SRR3081751</td> <td>CTCF_ChIP_macrophage_p3_2</td> <td>macrophage derived induced pluripotent stem cells after 3 passages</td> </tr> <tr> <td>GSM2026238</td> <td>SRR3081752</td> <td>CTCF_ChIP_MEF_p20_1</td> <td>mouse embryonic fibroblasts derived induced pluripotent stem cells after 20 passages</td> </tr> <tr> <td>GSM2026239</td> <td>SRR3081753</td> <td>CTCF_ChIP_MEF_p20_2</td> <td>mouse embryonic fibroblasts derived induced pluripotent stem cells after 20 passages</td> </tr> <tr> <td>GSM2026258</td> <td>SRR3081772</td> <td>WCE_p3</td> <td>whole cell extract after 3 passages</td> </tr> <tr> <td>GSM2026259</td> <td>SRR3081773</td> <td>WCE_p20</td> <td>whole cell extract after 20 passages</td> </tr> </tbody> </table> <p>The raw fastq files were downloaded from SRA using the <a href=https://hpc.nih.gov/apps/sratoolkit.html>sra toolkit</a></p> <div class=highlight><pre><span></span><code>&gt; ls /data/CCBR_Pipeliner/testdata/chipseq/*fastq.gz
/data/CCBR_Pipeliner/testdata/chipseq/SRR3081749_1.fastq.gz
/data/CCBR_Pipeliner/testdata/chipseq/SRR3081750_1.fastq.gz
/data/CCBR_Pipeliner/testdata/chipseq/SRR3081751_1.fastq.gz
/data/CCBR_Pipeliner/testdata/chipseq/SRR3081752_1.fastq.gz
/data/CCBR_Pipeliner/testdata/chipseq/SRR3081753_1.fastq.gz
/data/CCBR_Pipeliner/testdata/chipseq/SRR3081772_1.fastq.gz
/data/CCBR_Pipeliner/testdata/chipseq/SRR3081773_1.fastq.gz
</code></pre></div> <p>This is single-ended sequencing dataset, hence we have one fastq file per sample. Single-ended (SE) sequencing is the preferred sequencing choice for ChIP-seq data, but CCBR_Pipeliner does accept paired-end (PE) data. In case of PE data, each sample will have two fastq files of raw data, namely, a <em>Read1</em> and a <em>Read2</em>, generally denoted by <em>R1</em> and <em>R2</em>. Please note that, <em>R1</em> and <em>R2</em> in the context of PE data, do not represent two different replicates as this is the most common source of confusion for most novice NGS data analysts.</p> <p>The SRA filenames do not comply with our <a href>file naming convention</a>. There are three possible ways to fix this:</p> <ol> <li>Rename the files with CCBR_Pipeliner compliant names</li> <li>Create symbolic links at a different location and rename the symbolic links with CCBR_Pipeliner compliant filenames</li> <li>Add a tab-delimited file called <strong>labels.txt</strong> in the same folder as the raw data. This file has 2 columns:</li> <li>First column denotes the current name of the file, and</li> <li>Second column is the what the corresponding CCBR_Pipeliner compliant name should be.</li> </ol> <p>Here, we are going to use option 3. The <strong>labels.txt</strong> file looks something like this:</p> <div class=highlight><pre><span></span><code>SRR3081748_1.fastq.gz   CTCF_ChIP_macrophage_p20_1.R1.fastq.gz
SRR3081749_1.fastq.gz   CTCF_ChIP_macrophage_p20_2.R1.fastq.gz
SRR3081750_1.fastq.gz   CTCF_ChIP_macrophage_p3_1.R1.fastq.gz
SRR3081751_1.fastq.gz   CTCF_ChIP_macrophage_p3_2.R1.fastq.gz
SRR3081752_1.fastq.gz   CTCF_ChIP_MEF_p20_1.R1.fastq.gz
SRR3081753_1.fastq.gz   CTCF_ChIP_MEF_p20_2.R1.fastq.gz
SRR3081772_1.fastq.gz   WCE_p3_input.R1.fastq.gz
SRR3081773_1.fastq.gz   WCE_p20_input.R1.fastq.gz
</code></pre></div> <h3 id=phase-1-initial-qc><strong>Phase 1: Initial QC</strong><a class=headerlink href=#phase-1-initial-qc title="Permanent link">&para;</a></h3> <p>Before you can run the QC portion of the ChIP-seq pipeline, there are a few key pieces of information that must be provided.</p> <h4 id=step-0-fill-out-the-project-information-section><strong>Step 0. Fill out the <code>Project Information</code> section</strong><a class=headerlink href=#step-0-fill-out-the-project-information-section title="Permanent link">&para;</a></h4> <p>This section contains three fields: <code>Project Id</code>, <code>Email address</code>, <code>Flow Cell ID</code>. For this demo, you can set <em>Project Id</em> to <code>project</code>. This will be the name of the pipeline's master job. Please enter your email address in the <code>Email address</code> field. You may leave <code>Flow cell ID</code> blank for now.</p> <h4 id=step-1-setting-pipeline><strong>Step 1. Setting <code>Pipeline</code></strong><a class=headerlink href=#step-1-setting-pipeline title="Permanent link">&para;</a></h4> <p>Under the <code>Pipeline Family</code> select <code>ChIPSeq</code>. Once <code>ChIPSeq</code> pipeline is selected a new tab should appear next to <code>Project Description</code>.</p> <p><img alt="Select Pipeline" src=../images/pfamily.png></p> <h4 id=step-2-select-genome><strong>Step 2. Select <code>Genome</code></strong><a class=headerlink href=#step-2-select-genome title="Permanent link">&para;</a></h4> <p>Next select the genome to choose <code>mm10</code> as this is <em>Mus musculus</em> data. Although many genome options show up in the <code>Genome</code> drop down only the following four are supported by the ChIP-seq pipeline.</p> <table> <thead> <tr> <th>Genome</th> <th>Organism (assembly)</th> </tr> </thead> <tbody> <tr> <td>hg19</td> <td>Human (GRCh37)</td> </tr> <tr> <td>hg38</td> <td>Human (GRCh38)</td> </tr> <tr> <td>mm9</td> <td>Mouse (GRCm37)</td> </tr> <tr> <td>mm10</td> <td>Mouse (GRCm38)</td> </tr> </tbody> </table> <p>If any other genome is selected from the drop-down, then a pop-up will give a warning and will prevent you from proceeding with your analysis.</p> <p><img alt="Select Pipeline" src=../images/genome.png></p> <blockquote> <p><em><strong>Please Note:</strong></em> If you select a reference genome that is not supported by the pipeline, a pop-up box will notify you.</p> </blockquote> <h4 id=step-3-select-your-data-directory><strong>Step 3. Select your Data Directory</strong><a class=headerlink href=#step-3-select-your-data-directory title="Permanent link">&para;</a></h4> <p>If you are following along with this tutorial, you can use the included demo dataset. Please select the <strong><code>Open Directory</code></strong> button and navigate to the following path on biowulf's filesystem in the pop-up box: </p> <div class=highlight><pre><span></span><code>/data/CCBR_Pipeliner/testdata/chipseq/
</code></pre></div> <p><img alt="Select Pipeline" src=../images/opendatadirectory.png></p> <blockquote> <p><em><strong>Please Note:</strong></em> Input files will be read from and output files will be written to the biowulf filesystem. If your rawdata does not exist on biowulf, then you may need to upload it there first. Output files can be downloaded from biowulf to your local computer for downstream analysis.</p> </blockquote> <p>Once you have pointed to or navigated to the directory above, select <strong><code>OK</code></strong>. You should receive a message that 8 single end files were found:</p> <p><img alt="Select Pipeline" src=../images/filesfound.png></p> <p>Click OK.</p> <h4 id=step-4-select-your-working-directory><strong>Step 4. Select your Working Directory</strong><a class=headerlink href=#step-4-select-your-working-directory title="Permanent link">&para;</a></h4> <p>You are now ready to select your working directory on biowulf's filesystem. This is where all of the pipeline's output files will be created. In this tutorial, I set the working directory to <code>/scratch/demo/chipseq</code> by clicking the <code>Open Directory</code> button in front of <code>Working Directory</code> and then navigating the my directory of choice. But, you could set it to another location like <code>/scratch/&lt;yourUserName&gt;/chipseq_demo</code>. The only pre-requisites are that</p> <ul> <li>this directory should <strong>not</strong> exist.</li> <li>this directory should <strong>not</strong> be a sub-directory of CCBR pipeliner <em>working dir</em> from a previous run.</li> </ul> <p>The process very similar to the previous step except, you will point to a directory that does not exist on the biowulf filesystem by typing in the directory name, as it will be created in the next step. </p> <p><img alt="Select Pipeline" src=../images/workdir.png></p> <blockquote> <p><em><strong>Please Note:</strong></em> Input files will be read from and output files will be written to the biowulf filesystem. If you rawdata does not exist on biowulf, then you may need to upload it there first. Output files can be downloaded from biowulf to your local computer for downstream analysis.</p> </blockquote> <h4 id=step-5-initialize-your-working-directory><strong>Step 5. Initialize your Working Directory</strong><a class=headerlink href=#step-5-initialize-your-working-directory title="Permanent link">&para;</a></h4> <p>In this step, we will initialize or create the Working Directory defined in the step above. Please select the <strong><code>Initialize Directory</code></strong> button. During this step, the required resources to run the pipeline are copied into the working directory. After a few moments, you should receive a notification stating the directory was successfully initialized:</p> <p><img alt="Select Pipeline" src=../images/workdir_initialized.png></p> <p>Click Ok.</p> <p>If the directory already exists or you do not have write permissions, then you will get a warning like this:</p> <p><img alt=Warning src=../images/workdir_not_initialized.png></p> <h4 id=step-6-set-peak-information><strong>Step 6. Set Peak information</strong><a class=headerlink href=#step-6-set-peak-information title="Permanent link">&para;</a></h4> <p>Under <code>Options</code>, <code>InitialChIPseqQC</code>is selected by default. Click the <code>Set Peak Information</code> button, to pop-up another window, where you can specify which samples are <em>ChIP</em> samples and which samples should be used as their corresponding <em>Input</em> samples:</p> <p><img alt="Create peakcalls.tab" src=../images/set_peakcalltab.png></p> <ul> <li>Select all the ChIP samples in the <em>ChIP Names</em> column. </li> <li>For each ChIP sample, select the matching control sample (Input, IgG control, mock, no tag control, or WCE) in the <em>Input Names</em> column. Inputs can be paired to more than one ChIP sample. Leave as N/A if there is no control match for a ChIP sample. </li> <li>Type a <em>Group Name</em> for each ChIP sample or select from the pull-down list in the rightmost column. Each ChIP sample must have a Group Name. Replicates should be given the same Group Name. Group names should only contain alphanumeric characters (No spaces, hyphens, dashes, etc.), but underscores are okay. </li> <li>Click <code>Save</code> button once done and close the popup. This will save this information as <em>peakcall.tab</em> file in the <em>work directory</em>. Here we created three groups, namely <strong>mac_p20</strong>, <strong>map_p3</strong> and <strong>mef_p20</strong>. Each of these groups have 2 replicates each denoted by the <code>_1</code> and <code>_2</code> in the file basenames.</li> <li>You can also save the three columns, ie, <strong>ChIP Names</strong>, <strong>Input Names</strong> and <strong>Group Names</strong> as a tab-delimited file in the <em>work directory</em>, name it <em>peakcall.tab</em> and load the predefined file using the <code>Load</code> button.</li> </ul> <h4 id=step-7-perform-dry-run><strong>Step 7. Perform Dry Run</strong><a class=headerlink href=#step-7-perform-dry-run title="Permanent link">&para;</a></h4> <p>Click on the <code>Dry Run</code> button to check if the initial QC phase tasks and the pipeline is configured properly. A successful dry run will display all tasks to be run and will have a job counts table at the top and bottom of the popup. This step take about 10-20 seconds to run, so please be patient until another window pops up with DAG information:</p> <p><img alt="Dry run" src=../images/dag.png></p> <p>If there is no error in the <code>Dry Run</code>, then you are ready to proceed.</p> <h4 id=step-8-run><strong>Step 8. Run</strong><a class=headerlink href=#step-8-run title="Permanent link">&para;</a></h4> <p>Close the <code>Dry Run</code> popup and click the <code>Run</code> button. In a few seconds, you will see another popup:</p> <p><img alt="Submit to Slurm" src=../images/slurm.png></p> <p>Once you click <code>OK</code>button on the popup, the job is submitted to the <a href=https://hpc.nih.gov/docs/userguide.html>slurm queue</a> on biowulf.</p> <h3 id=check-progress>Check progress<a class=headerlink href=#check-progress title="Permanent link">&para;</a></h3> <p>You can check status of the jobs submitted using the HPC <a href=https://hpc.nih.gov/dashboard/ >dashboard</a>. If you are comfortable using the command line, you can use <code>squeue</code> or <code>sjobs</code> command to monitor jobs submitted to <em>slurm</em>. Please refer <em>slurm</em> <a href=https://hpc.nih.gov/docs/userguide.html>documentation</a> for details.</p> <h3 id=confirm-successful-completion-of-phase-1>Confirm successful completion of Phase 1.<a class=headerlink href=#confirm-successful-completion-of-phase-1 title="Permanent link">&para;</a></h3> <p>In the <em>work directory</em>, look for a file name <em>HPC_usage_table.txt</em>, which lists details about all submitted jobs. This file doesn't appear until the job is finished. Here are columns in this tab-delimited file:</p> <div class=highlight><pre><span></span><code><span class=m>1</span> JobName
<span class=m>2</span> Jobid
<span class=m>3</span> Partition
<span class=m>4</span> State
<span class=m>5</span> Nodes
<span class=m>6</span> CPUs
<span class=m>7</span> Walltime
<span class=m>8</span> Runtime
<span class=m>9</span> MemReq
<span class=m>10</span> MemUsed
<span class=m>11</span> Nodelist
<span class=m>12</span> MaxCPUUsed
<span class=m>13</span> Queuetime
<span class=m>14</span> CPUHours
<span class=m>15</span> Account
<span class=m>16</span> Username
</code></pre></div> <p>This file can be open in Microsoft Excel for easy viewing. Examine the values under the column <strong>Status</strong> . The values can be <strong>COMPLETED</strong>. or <strong>FAILED</strong>. If no <strong>FAILED</strong> values are found and all values are <strong>COMPLETED</strong>, then Phase 1 has finished running successfully and you are ready to move to phase 2. </p> <h3 id=phase-2-peak-calling-differential-peak-calling-annotations-and-motif-searches><strong>Phase 2: Peak calling, differential peak calling, annotations, and motif searches</strong><a class=headerlink href=#phase-2-peak-calling-differential-peak-calling-annotations-and-motif-searches title="Permanent link">&para;</a></h3> <p>Phase 1 will take a few hours to run. It is important to know that the second phase of the ChIP-seq Pipeliner can only be run after successful completion of initialQC phase. This workflow is to be executed in the same working directory (e.g.<code>/scratch/&lt;yourUserName&gt;/chipseq_demo</code>) as that specified in the InitialQC phase, so that the pipeline can identify the output files from the InitialQC run, that are being used as input files to the differential expression workflow. </p> <h4 id=step-1-initial-setup>Step 1. Initial Setup<a class=headerlink href=#step-1-initial-setup title="Permanent link">&para;</a></h4> <p>Repeat steps 0 through 3 from Phase 1. Also enter the <em>work directory</em> to the same folder selected in Step 4 of Phase 1, the only difference is that the <em>work directory</em> should already exist with Phase 1 results in it.</p> <h4 id=step-2-select-pipeline-options>Step 2. Select Pipeline Options<a class=headerlink href=#step-2-select-pipeline-options title="Permanent link">&para;</a></h4> <p>From <code>Options</code> select <code>ChIPseq</code> <code>Pipeline</code> from the dropdown.</p> <p><img alt="Select Pipeline" src=../images/phase2.png></p> <h4 id=step-3-set-contrasts-optional>Step 3. Set contrasts (Optional)<a class=headerlink href=#step-3-set-contrasts-optional title="Permanent link">&para;</a></h4> <p>If you wish to compare peaks between groups after peakcalling, then you can specify the contrasts that need to be performed here. Click <code>Set Groups to compare peaks</code></p> <p><img alt="Set groups to compare peaks" src=../images/setgroups.png></p> <p>This will popup another window where you can select the groups to compare. Here, each row denoted a different comparison, ie, an independent instance of differential peak analysis using <em>DiffBind</em>.</p> <p><img alt="Set contrasts" src=../images/setcontrasts.png></p> <p>Clicking <code>Save</code> button creates a <em>contrast.tab</em> file in the <em>work directory</em>. If you are familiar with creating and editing file at the command line, then you can also create a tab-delimited file with the name <em>contrast.tab</em> in the <em>work directory</em> in order to set the contrasts without using the GUI interface. The <em>contrast.tab</em> looks like this:</p> <div class=highlight><pre><span></span><code>Macrophage_p3   Macrophage_p20
MEF_p20 Macrophage_p20
</code></pre></div> <p>Here, we have specified two contrasts trying to compare the peaks in the groups <em>Macrophage_p3</em> and <em>MEF_p20</em> with respect to the peaks in group <em>Macrophage_p20</em>.</p> <h4 id=step-4-perform-dry-run>Step 4. Perform Dry Run<a class=headerlink href=#step-4-perform-dry-run title="Permanent link">&para;</a></h4> <p>Close the previous popup and then, click on the <code>Dry Run</code> button to check if the phase2 tasks and the pipeline is configured properly. A successful dry run will display all tasks to be run and will have a job counts table at the top and bottom of the popup. This step take about 10-20 seconds to run, so please be patient until another window pops up with DAG information:</p> <p><img alt="Dry Run" src=../images/dryrun.png></p> <p>If there is no error in the <code>Dry Run</code> , as seen in the truncated snapshot above, then you are ready to proceed.</p> <h4 id=step-5-run><strong>Step 5. Run</strong><a class=headerlink href=#step-5-run title="Permanent link">&para;</a></h4> <p>Close the <code>Dry Run</code> popup and click the <code>Run</code> button on the main GUI. In a few seconds, you will see another popup:</p> <p><img alt="Submit to Slurm" src=../images/slurm2.png></p> <p>Once you click <code>OK</code>button on the popup, the job is submitted to the <a href=https://hpc.nih.gov/docs/userguide.html>slurm queue</a> on biowulf.</p> <h3 id=check-progress_1>Check progress<a class=headerlink href=#check-progress_1 title="Permanent link">&para;</a></h3> <p>You can check status of the jobs submitted using the HPC <a href=https://hpc.nih.gov/dashboard/ >dashboard</a>. If you are comfortable using the command line, you can use <code>squeue</code> or <code>sjobs</code> command to monitor jobs submitted to <em>slurm</em>. Please refer <em>slurm</em> <a href=https://hpc.nih.gov/docs/userguide.html>documentation</a> for details. You can also use the <em>HPC_usage_table.txt</em> as we did in Phase 1.</p> <hr> <div class=md-source-file> <small> Last update: 2022-11-04 </small> </div> </article> </div> </div> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../../dev/coming-soon/ class="md-footer__link md-footer__link--prev" aria-label="Previous: Customization" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> Customization </div> </div> </a> <a href=../ChIP-seq-output-files/ class="md-footer__link md-footer__link--next" aria-label="Next: Output Files" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Output Files </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2020 CCBR </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=http://bioinformatics.cancer.gov target=_blank rel=noopener title=bioinformatics.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M319.9 320c57.41 0 103.1-46.56 103.1-104s-46.54-104-103.1-104c-57.41 0-103.1 46.56-103.1 104-.9 57.4 45.7 104 103.1 104zm50 32h-99.8C191.6 352 128 411.7 128 485.3c0 14.8 12.7 26.7 28.4 26.7h327.2c15.7 0 28.4-11.9 28.4-26.7 0-73.6-63.6-133.3-142.1-133.3zM512 160c44.18 0 80-35.82 80-80S556.2 0 512 0c-44.18 0-80 35.82-80 80s35.8 80 80 80zm-328.1 56c0-5.449.982-10.63 1.609-15.91C174.6 194.1 162.6 192 149.9 192H88.08C39.44 192 0 233.8 0 285.3 0 295.6 7.887 304 17.62 304h199.5c-20.42-23.8-33.22-54.3-33.22-88zM128 160c44.18 0 80-35.82 80-80S172.2 0 128 0C83.82 0 48 35.82 48 80s35.82 80 80 80zm423.9 32h-61.84c-12.8 0-24.88 3.037-35.86 8.24.6 5.26 1.6 10.36 1.6 15.76 0 33.71-12.78 64.21-33.16 88h199.7c9.76 0 17.66-8.4 17.66-18.7 0-51.5-39.4-93.3-88.1-93.3z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://hub.docker.com/orgs/nciccbr/repositories target=_blank rel=noopener title=hub.docker.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.b028fd86.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../assets/javascripts/bundle.0f659f14.min.js></script> </body> </html>