<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Pipeliner is an open-source and scalable solution for analyzing  next-generation sequencing data. Pipeliner provides access to  the same best-practices NGS pipelines developed and benchmarked  by experts at CCBR and NCBR. This repository is the main source  of documentation for users and developers working with or  contributing to Pipeliner."><meta name=author content="Skyler Kuhn"><link rel=icon href=../../assets/favicon/favicon.ico><meta name=generator content="mkdocs-1.4.2, mkdocs-material-8.3.0"><title>User Tutorial - Pipeliner Documentation</title><link rel=stylesheet href=../../assets/stylesheets/main.fd896c8a.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#user-tutorial class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="Pipeliner Documentation" class="md-header__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Pipeliner Documentation </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> User Tutorial </span> </div> </div> </div> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="Pipeliner Documentation" class="md-nav__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> Pipeliner Documentation </label> <div class=md-nav__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> Home </a> </li> <li class=md-nav__item> <a href=../../about/ class=md-nav__link> About </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3> <label class=md-nav__link for=__nav_3> Getting started <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Getting started" data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Getting started </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1> <label class=md-nav__link for=__nav_3_1> Users <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Users data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Users </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Prerequisites </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Installation </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <label class=md-nav__link for=__nav_3_2> Developers <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Developers data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Developers </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Installation </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Customization </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4 checked> <label class=md-nav__link for=__nav_4> Pipelines <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Pipelines data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Pipelines </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <label class=md-nav__link for=__nav_4_1> ChIP-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=ChIP-seq data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> ChIP-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../ChIP-seq/ChIP-seq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../ChIP-seq/ChIP-seq-output-files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../ChIP-seq/ChIP-seq-tools/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2 checked> <label class=md-nav__link for=__nav_4_2> RNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=RNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> RNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../Theory-and-practical-guide-for-RNA-seq/ class=md-nav__link> Theory </a> </li> <li class=md-nav__item> <a href=../Gene-and-isoform-expression-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../TLDR-RNA-seq/ class=md-nav__link> Quick Start </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> User Tutorial <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> User Tutorial </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#quantification-and-quality-control-pipeline class=md-nav__link> Quantification and quality-control pipeline </a> <nav class=md-nav aria-label="Quantification and quality-control pipeline"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#about-the-demo-dataset class=md-nav__link> About the Demo Dataset </a> </li> <li class=md-nav__item> <a href=#step-0-fill-out-the-project-information-section class=md-nav__link> Step 0. Fill out the Project Information section </a> </li> <li class=md-nav__item> <a href=#step-1-select-the-correct-pipeline-family class=md-nav__link> Step 1. Select the correct Pipeline Family </a> </li> <li class=md-nav__item> <a href=#step-2-select-the-reference-genome class=md-nav__link> Step 2. Select the Reference Genome </a> </li> <li class=md-nav__item> <a href=#step-3-select-your-data-directory class=md-nav__link> Step 3. Select your Data Directory </a> </li> <li class=md-nav__item> <a href=#step-4-select-your-working-directory class=md-nav__link> Step 4. Select your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-5-initialize-your-working-directory class=md-nav__link> Step 5. Initialize your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-6-select-the-pipeline class=md-nav__link> Step 6. Select the Pipeline </a> </li> <li class=md-nav__item> <a href=#step-7-load-group-information class=md-nav__link> Step 7. Load Group Information </a> </li> <li class=md-nav__item> <a href=#step-8-dry-run-the-pipeline class=md-nav__link> Step 8. Dry-run the pipeline </a> </li> <li class=md-nav__item> <a href=#step-9-run-the-pipeline class=md-nav__link> Step 9. Run the pipeline </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#check-progress class=md-nav__link> Check Progress </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../Differential-expression-pipeline-tools-and-versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <label class=md-nav__link for=__nav_4_3> miRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=miRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> miRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../miRNA-seq/overview-miRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/TLDR-miRNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/miRSeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/miRSeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <label class=md-nav__link for=__nav_4_4> scRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=scRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> scRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../scRNA-seq/overview-scRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/TLDR-Single-Cell-RNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/scRNASeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/scRNASeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <label class=md-nav__link for=__nav_4_5> WES/WGS <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=WES/WGS data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> WES/WGS </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/TLDR-WEG-seq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-output/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-tools-and-versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_5 type=checkbox id=__nav_5> <label class=md-nav__link for=__nav_5> FAQ <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=FAQ data-md-level=1> <label class=md-nav__title for=__nav_5> <span class="md-nav__icon md-icon"></span> FAQ </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../troubleshooting/ class=md-nav__link> Troubleshooting </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Data Privacy </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../community-and-contribution/ class=md-nav__link> Contribute </a> </li> <li class=md-nav__item> <a href=../../changelog/ class=md-nav__link> Changelog </a> </li> <li class=md-nav__item> <a href=../../contact-us/ class=md-nav__link> Contact </a> </li> <li class=md-nav__item> <a href=../../license/ class=md-nav__link> License </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#quantification-and-quality-control-pipeline class=md-nav__link> Quantification and quality-control pipeline </a> <nav class=md-nav aria-label="Quantification and quality-control pipeline"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#about-the-demo-dataset class=md-nav__link> About the Demo Dataset </a> </li> <li class=md-nav__item> <a href=#step-0-fill-out-the-project-information-section class=md-nav__link> Step 0. Fill out the Project Information section </a> </li> <li class=md-nav__item> <a href=#step-1-select-the-correct-pipeline-family class=md-nav__link> Step 1. Select the correct Pipeline Family </a> </li> <li class=md-nav__item> <a href=#step-2-select-the-reference-genome class=md-nav__link> Step 2. Select the Reference Genome </a> </li> <li class=md-nav__item> <a href=#step-3-select-your-data-directory class=md-nav__link> Step 3. Select your Data Directory </a> </li> <li class=md-nav__item> <a href=#step-4-select-your-working-directory class=md-nav__link> Step 4. Select your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-5-initialize-your-working-directory class=md-nav__link> Step 5. Initialize your Working Directory </a> </li> <li class=md-nav__item> <a href=#step-6-select-the-pipeline class=md-nav__link> Step 6. Select the Pipeline </a> </li> <li class=md-nav__item> <a href=#step-7-load-group-information class=md-nav__link> Step 7. Load Group Information </a> </li> <li class=md-nav__item> <a href=#step-8-dry-run-the-pipeline class=md-nav__link> Step 8. Dry-run the pipeline </a> </li> <li class=md-nav__item> <a href=#step-9-run-the-pipeline class=md-nav__link> Step 9. Run the pipeline </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#check-progress class=md-nav__link> Check Progress </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/CCBR/pipeliner-docs/edit/master/docs/RNA-seq/Gene-and-isoform-expression-guided-tutorial.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1 id=user-tutorial>User Tutorial<a class=headerlink href=#user-tutorial title="Permanent link">&para;</a></h1> <p><strong>Estimated completion time:</strong> <code>15 mins</code></p> <p>This section offers a guided tutorial that you can follow along on Biowulf. We have provided a test data set so you can learn how to set-up and run this pipeline.</p> <blockquote> <p><em><strong>Please note:</strong></em> To follow along with this tutorial, you must have a <a href=https://hpc.nih.gov/docs/accounts.html>Biowulf account</a>.</p> </blockquote> <h2 id=quantification-and-quality-control-pipeline>Quantification and quality-control pipeline<a class=headerlink href=#quantification-and-quality-control-pipeline title="Permanent link">&para;</a></h2> <p>This section provides step-by-step instructions for setting up the first pipeline (Quantification and Quality-control pipeline) and a brief description of the pipeline's output files and directories. We assume that you have already successfully launched the GUI. If not, please see <a href=https://github.com/CCBR/Pipeliner/wiki/TLDR-RNA-seq#quick-start>Launch Pipeliner</a>.</p> <h3 id=about-the-demo-dataset>About the Demo Dataset<a class=headerlink href=#about-the-demo-dataset title="Permanent link">&para;</a></h3> <p>We have provided a test dataset that you can use to follow along with this tutorial. The raw data for this demo resides in <code>/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/</code>. This demo dataset consists of 24 paired-end FastQ files for 12 samples, originating from a human cell-line. Libraries were constructed with a poly-A selection library preparation kit. There are three groups each with 4 replicates: <code>Cntrl</code>, <code>TgA</code>, and <code>TgB</code>. The <code>Cntrl</code> samples represent a baseline biological state while <code>TgA</code> and <code>TgB</code> samples represent the same cell-line after the treatment of two drugs: <code>A</code> and <code>B</code>.<br> Here is a listing of all the samples in the demo dataset: <div class=highlight><pre><span></span><code>$ ls /data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/*.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S62.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S62.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S63.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S63.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S64.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S64.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S65.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/Cntrl_S65.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S66.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S66.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S67.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S67.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S68.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S68.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S69.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentA_S69.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S70.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S70.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S71.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S71.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S72.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S72.R2.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S73.R1.fastq.gz
/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/TreatmentB_S73.R2.fastq.gz
</code></pre></div></p> <blockquote> <p><em><strong>Please Note:</strong></em> that "R1" and "R2" do not represent the replicate number. They denote forward and reverse reads, respectively.</p> </blockquote> <p>Pipeliner expects input files to follow a specific naming convention: <code>&lt;sampleName&gt;.R1.fastq.gz</code>, <code>&lt;sampleName&gt;.R2.fastq.gz</code>. FastQ files should end with the following extensions: <code>.R1.fastq.gz</code> and <code>.R2.fastq.gz</code>. Within <code>&lt;sampleName&gt;</code>, there are additional conventions you must follow. Please carefully read through <a href=https://github.com/CCBR/Pipeliner/wiki/TLDR-RNA-seq#rules-for-naming-raw-data-files>all the rules</a> for creating filenames.</p> <p>Before you can run the expression pipeline, there are a few key pieces of information that must be provided.</p> <h3 id=step-0-fill-out-the-project-information-section>Step 0. Fill out the <code>Project Information</code> section<a class=headerlink href=#step-0-fill-out-the-project-information-section title="Permanent link">&para;</a></h3> <p>This section contains three fields: <code>Project Id</code>, <code>Email address</code>, <code>Flow Cell ID</code>. For this demo, you can set <em>Project Id</em> to <code>project</code>. This will be the name of the pipeline's master job. Please enter your email address in the <code>Email address</code> field.</p> <h3 id=step-1-select-the-correct-pipeline-family>Step 1. Select the correct Pipeline Family<a class=headerlink href=#step-1-select-the-correct-pipeline-family title="Permanent link">&para;</a></h3> <p>Please select <strong><code>rnaseq</code></strong> from the pipeline family drop-down menu.</p> <p align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/pfamily.png width=50% height=50%> </p> <blockquote> <p><em><strong>Please note:</strong></em> The <code>scrnaseq</code> option is for single-cell data. This tutorial and pipeline are for bulk RNA-seq. To run the single-cell RNA-seq pipeline, please see its documentation.</p> </blockquote> <h3 id=step-2-select-the-reference-genome>Step 2. Select the Reference Genome<a class=headerlink href=#step-2-select-the-reference-genome title="Permanent link">&para;</a></h3> <p>As mentioned above, these samples originate from a human cell-line. Please select <code>hg38</code> from the <code>Genome</code> drop-down menu. <p align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/refgenome.png width=50% height=50%> </p></p> <blockquote> <p><em><strong>Please Note:</strong></em> If you select a reference genome that is not supported by the pipeline, a pop-up box will notify you.</p> </blockquote> <h3 id=step-3-select-your-data-directory>Step 3. Select your Data Directory<a class=headerlink href=#step-3-select-your-data-directory title="Permanent link">&para;</a></h3> <p>If you are following along with this tutorial, you can use the included demo dataset. Please select the <strong><code>Open Directory</code></strong> button and navigate to the following path in the pop-up box:<br> <div class=highlight><pre><span></span><code>/data/CCBR_Pipeliner/testdata/rnaseq/expression_demo/
</code></pre></div> Once you have pointed-to or navigated to the directory above, select <strong><code>OK</code></strong>.<br> <p float=left align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/RNA_select_wd.png width=40% height=40%> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/found_data.png width=40% height=40%> </p></p> <p>You should receive a message that 24 files were found. Click <strong><code>OK</code></strong>.</p> <h3 id=step-4-select-your-working-directory>Step 4. Select your Working Directory<a class=headerlink href=#step-4-select-your-working-directory title="Permanent link">&para;</a></h3> <p>You are now ready to select your working directory. This is where all of the pipeline's output files will be created. In this tutorial, I set the working directory to <code>/scratch/demo/rnaseq</code>, but you could set it to another location like <code>/scratch/demo/yourUserName/</code>. The only pre-requisite is that this directory should not exist. The process very similar to the step above except, you will point to a directory that does not exist on the filesystem. <p align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/RNA_find_data_dir.png width=50% height=50%> </p></p> <blockquote> <p><em><strong>Please Note:</strong></em> Input and output files will be read from and written to the Biowulf's filesystem. If your raw data does not exist on Biowulf, you will need to upload it there first. Output files can be downloaded from Biowulf to your local computer for downstream analysis. </p> </blockquote> <h3 id=step-5-initialize-your-working-directory>Step 5. Initialize your Working Directory<a class=headerlink href=#step-5-initialize-your-working-directory title="Permanent link">&para;</a></h3> <p>In this step, we will initialize or create the Working Directory defined in the step above. Please select the <strong><code>Initialize Directory</code></strong> button. During this step, the required resources to run the pipeline are copied into the working directory. After a few moments, you should receive a notification stating the directory was successfully initialized: <p align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/select_init.png width=50% height=50%> </p></p> <h3 id=step-6-select-the-pipeline>Step 6. Select the Pipeline<a class=headerlink href=#step-6-select-the-pipeline title="Permanent link">&para;</a></h3> <p>You are now ready to select the Quantification and Quality-control Pipeline. In the options section, please select <strong><code>Quality Control Analysis</code></strong> from the <code>Pipeline</code> field.</p> <h3 id=step-7-load-group-information>Step 7. Load Group Information<a class=headerlink href=#step-7-load-group-information title="Permanent link">&para;</a></h3> <p>In the options, you will need to provide group information for each sample. Select <strong><code>Set Groups</code></strong> from the <code>Sample Information</code> section. Paste the group information below into the pop-up box and click <strong><code>save</code></strong>. </p> <p>Example <code>groups.tab</code> file <div class=highlight><pre><span></span><code>Cntrl_S62   Cntrl   Cntrl_1
Cntrl_S63   Cntrl   Cntrl_2
Cntrl_S64   Cntrl   Cntrl_3
Cntrl_S65   Cntrl   Cntrl_4
TreatmentA_S66  TgA TgA_1
TreatmentA_S67  TgA TgA_2
TreatmentA_S68  TgA TgA_3
TreatmentA_S69  TgA TgA_4
TreatmentB_S70  TgB TgB_1
TreatmentB_S71  TgB TgB_2
TreatmentB_S72  TgB TgB_3
TreatmentB_S73  TgB TgB_4
</code></pre></div></p> <h3 id=step-8-dry-run-the-pipeline>Step 8. Dry-run the pipeline<a class=headerlink href=#step-8-dry-run-the-pipeline title="Permanent link">&para;</a></h3> <p>Please select the <strong><code>Dry-run</code></strong> button. This will generate the pipeline's <em>Snakefile</em>, <em>run.json</em>, and it will dry run the pipeline. This step take about 10-20 seconds to run, so please be patient until another window pops up with DAG information:</p> <p float=left align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/ready2dryrun.png align=top width=30% height=30%> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/dryrun.png width=30% height=30%> </p> <p>If there is no error in the <code>Dry Run</code>, please close the dry-run popup box and proceed to the next step.</p> <h3 id=step-9-run-the-pipeline>Step 9. Run the pipeline<a class=headerlink href=#step-9-run-the-pipeline title="Permanent link">&para;</a></h3> <p>You are now ready to run the Quantification and Quality-control pipeline! Please select the <strong><code>Run</code></strong> button. In a few moments, a pop-up box will appear stating that the job is starting. Please select <strong><code>OK</code></strong>. Once <code>OK</code> is selected from this popup box, the job will be submitted to the SLURM queue on Biowulf.</p> <p align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/starting_job.png width=50% height=50%> </p> <p>That's it! You will receive an automated email when the pipeline starts and ends.</p> <p align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/RNA_email_pipeline_began.png width=50% height=50%> </p> <p>You can now close out of the GUI. It should be noted that this pipeline takes around 3-4 hours to run.</p> <h2 id=check-progress>Check Progress<a class=headerlink href=#check-progress title="Permanent link">&para;</a></h2> <p>If you are curious to see what jobs are running, you can run <code>sjobs</code> (or <code>squeue -u $USER</code>). All of Pipeliner's jobs begin with the prefix <code>pl:</code>.</p> <p align=center> <img src=https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/sjobs.png width=50% height=50%> </p> <p>Before running the second pipeline, it is important to ensure the pipeline has run to completion. There are a lot of factors that can cause a job to fail, some of which are out of our hands. To see if all the jobs have finished successfully, navigate to the Reports directory within your working directory, and examine the last few lines of the snakemake.log file. This file tracks the progress of all the jobs that get submitted to the cluster. If the pipeline has run to completion, it will look like this:</p> <p align=center> <img src="https://github.com/CCBR/Pipeliner/wiki/Images/RNA-Seq_Doc/Fig. SnakemakeLog.png" width=50% height=50%> </p> <p>If the pipeline reports that it's <strong><code>(100%) done</code></strong>, you are ready to run the Differential Expression Pipeline.</p> <p>You can also verify that all your jobs have completed by checking out the following file in your pipeline's working directory: <em>HPC_usage_table.txt</em>. This file doesn't appear until all jobs are finished. It contains information about each job submitted to the cluster. Here are columns in this tab-delimited file: <div class=highlight><pre><span></span><code><span class=m>1</span> JobName
<span class=m>2</span> Jobid
<span class=m>3</span> Partition
<span class=m>4</span> State
<span class=m>5</span> Nodes
<span class=m>6</span> CPUs
<span class=m>7</span> Walltime
<span class=m>8</span> Runtime
<span class=m>9</span> MemReq
<span class=m>10</span> MemUsed
<span class=m>11</span> Nodelist
<span class=m>12</span> MaxCPUUsed
<span class=m>13</span> Queuetime
<span class=m>14</span> CPUHours
<span class=m>15</span> Account
<span class=m>16</span> Username
</code></pre></div> This file can be opened in Microsoft Excel for easy viewing, or it can be viewed from the command-line. Please examine the values under the column <strong>State</strong>. The values can be <strong>COMPLETED</strong>, <strong>RUNNING</strong>, or <strong>FAILED</strong>. If there are no <strong>FAILED</strong> jobs and all jobs are <strong>COMPLETED</strong>, then the Quantification and Quality-control pipeline has finished running successfully. You are now ready to run to the Differential Expression pipeline.</p> <hr> <div class=md-source-file> <small> Last update: 2022-11-04 </small> </div> </article> </div> </div> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../TLDR-RNA-seq/ class="md-footer__link md-footer__link--prev" aria-label="Previous: Quick Start" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> Quick Start </div> </div> </a> <a href=../Differential-expression-pipeline-tools-and-versions/ class="md-footer__link md-footer__link--next" aria-label="Next: Resources" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Resources </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2020 CCBR </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=http://bioinformatics.cancer.gov target=_blank rel=noopener title=bioinformatics.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M319.9 320c57.41 0 103.1-46.56 103.1-104s-46.54-104-103.1-104c-57.41 0-103.1 46.56-103.1 104-.9 57.4 45.7 104 103.1 104zm50 32h-99.8C191.6 352 128 411.7 128 485.3c0 14.8 12.7 26.7 28.4 26.7h327.2c15.7 0 28.4-11.9 28.4-26.7 0-73.6-63.6-133.3-142.1-133.3zM512 160c44.18 0 80-35.82 80-80S556.2 0 512 0c-44.18 0-80 35.82-80 80s35.8 80 80 80zm-328.1 56c0-5.449.982-10.63 1.609-15.91C174.6 194.1 162.6 192 149.9 192H88.08C39.44 192 0 233.8 0 285.3 0 295.6 7.887 304 17.62 304h199.5c-20.42-23.8-33.22-54.3-33.22-88zM128 160c44.18 0 80-35.82 80-80S172.2 0 128 0C83.82 0 48 35.82 48 80s35.82 80 80 80zm423.9 32h-61.84c-12.8 0-24.88 3.037-35.86 8.24.6 5.26 1.6 10.36 1.6 15.76 0 33.71-12.78 64.21-33.16 88h199.7c9.76 0 17.66-8.4 17.66-18.7 0-51.5-39.4-93.3-88.1-93.3z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://hub.docker.com/orgs/nciccbr/repositories target=_blank rel=noopener title=hub.docker.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.b028fd86.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../assets/javascripts/bundle.0f659f14.min.js></script> </body> </html>