## v4.0.2  

* [Increasing picard memory allocation and num of threads for garbage collection](https://github.com/CCBR/Pipeliner/commit/c69ab80a2794114b7b3fd1508e24960aea93e15b)
* [ChIP-seq updates: estimated fragment length and more (#450)](https://github.com/CCBR/Pipeliner/commit/2082431d04ffd3b17eedd8f727680b4eb801c0cd)
* [Merge pull request #449 from wong-nw/activeDev](https://github.com/CCBR/Pipeliner/commit/8701b49d5bcdb64726ededb66353899a2d97acab)
* [Bug Fix in integrateBatches.R URD call](https://github.com/CCBR/Pipeliner/commit/9810ee09ea2c1ef2342845ffcfcbba0946c58a72)

## v4.0.1  

* [removed AVIA](https://github.com/CCBR/Pipeliner/commit/63a5ce9ac8992b5160a2fa266462a86c5dbbe1b8)
* [Increasing resources for fusion inspector](https://github.com/CCBR/Pipeliner/commit/e6ac6c176c7855feef35380fce3d19abc8a46623)
* [Switch to python script](https://github.com/CCBR/Pipeliner/commit/6becb18e6da109428253b6688b33b0981eba13a9)
* [Python implementation of 'reformat_bed.pl'](https://github.com/CCBR/Pipeliner/commit/f82a692aef74e29b0078bb54504f101ad0691701)
* [Adding memory for fusion inspector rules](https://github.com/CCBR/Pipeliner/commit/95c32405aa1266939e1b13857d0f13523b2c075c)
* [mm10 error](https://github.com/CCBR/Pipeliner/commit/d9378e8cfd8907614cab07752b259c06160f11b6)
* [Run only for 'hg19' or 'hg38'](https://github.com/CCBR/Pipeliner/commit/4142003616e706729177208c16f9b123aae8b8bd)
* [Remove ancient tag](https://github.com/CCBR/Pipeliner/commit/ab608406aa2743d84bce68fbc3339c8c551b1566)
* [Fixing STAR fusion perl issue](https://github.com/CCBR/Pipeliner/commit/693d84a776848316d06aaee122499621e18c143e)
* [Explicitly load Perl](https://github.com/CCBR/Pipeliner/commit/ead71c061550f08b7b794d0f42326786162a0792)
* [scRNASeq changes Feb 2020 (#438)](https://github.com/CCBR/Pipeliner/commit/baf82250a44a790a66c4abe5b88d0ebbf024b8be)
* [Changing git version: module git/2.15.1 is no longer available](https://github.com/CCBR/Pipeliner/commit/749f808efee6aef5479e6e47d8b2d1c13035ca75)

## v4.0.0  

* [Merging activeDev for v4.0.1 release (#444)](https://github.com/CCBR/Pipeliner/commit/57ca868f7cbb96eb805d37afbd07e56659958e8e)

## v3.0.6  

* [Setting activeDev branch to master](https://github.com/CCBR/Pipeliner/commit/e92a6a357403789fae073a5c27253da46ffd18dc)
* [Adding updated viral integrated genome hg38_30+HPV16](https://github.com/CCBR/Pipeliner/commit/d7ef5c52cb5646820e13a28fe67c0bf49b77d316)
* [Archiving older viral integrated hg38_28+HPV16 references json](https://github.com/CCBR/Pipeliner/commit/e03e5eb17f822dd233910710d40ae9ec54411f04)
* [Explicitly loading perl to aviod: 'Perl module GD::Graph::bars not installed, skipping charts' error](https://github.com/CCBR/Pipeliner/commit/2ab3f8378c678e6c618f4325b3a6412e9f98cee1)
* [Replacing hg19_KSHV with hg38_30_KSHV](https://github.com/CCBR/Pipeliner/commit/223f71749c75237c8bca39ff82f2ea2e83a54083)
* [Adding "References/" as a cache for old reference files](https://github.com/CCBR/Pipeliner/commit/e0a28aad0858898203d5f191c5accf5f4c4e2e5f)
* [Updating targets file text, now changes based on reference genome](https://github.com/CCBR/Pipeliner/commit/d2d175b92905f224989a716caabd93b31379bd50)
* [Merge pull request #432 from mtandon09/activeDev](https://github.com/CCBR/Pipeliner/commit/13178d6504fb8925f90478093533720d1bc052cc)
* [Specify R version (3.5) because default (3.6) does not have 'magick' installed](https://github.com/CCBR/Pipeliner/commit/f473d5864fed36c2d6565eff60fe78b8f16758f6)
* [Merge pull request #431 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/9b17bdaf757e0510c55b1cdc1cd878a2273e4e50)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/a5bdb62813411f2e1269d11c02159ca41c6a06a4)
* [fixed gzipped VCF issue with germline calling](https://github.com/CCBR/Pipeliner/commit/115a593f66964894c8fe60287493914dcea63f56)
* [Merge pull request #430 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/6dccc34be8ebe8d9bc95c6878ec2a52fb2078d0e)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/b922e977a7fa9a61d37008c38152da416f89bc36)
* [updated hg38 PON](https://github.com/CCBR/Pipeliner/commit/8ea10fddba86f82418d977bb62f5467a7bee50cd)
* [Updating rule pca: prevents race condition due to severe filesystem latency](https://github.com/CCBR/Pipeliner/commit/963b73f1e3798440f507eaa1ccb4597305654fb1)
* [Updating rule pca: severe filesystem latency can create race condition when copying over files](https://github.com/CCBR/Pipeliner/commit/4dcf8a59a24a1617c89894feb43f90c9355ca689)
* [Updating pcacall usage to aviod filesystem latency issues](https://github.com/CCBR/Pipeliner/commit/0925e5e4d538a83339d4e9faff87a0294d804fc6)
* [RNA-seq Initial QC: Contrasts file not longer required](https://github.com/CCBR/Pipeliner/commit/ba16a03a67866fefdba601e1c4c4c46e0f0b4145)
* [ChIP-seq GUI: Adding "contrasts" button to the GUI](https://github.com/CCBR/Pipeliner/commit/1931a807af9c69040e82574b527b72f5ca271a39)
* [Updating Initial QC RNA-seq Interface: only ask for group information](https://github.com/CCBR/Pipeliner/commit/c65b2e79fc7eba8a0cefd1b9428cd8f9b4886aa0)
* [Update dryrun file-checking for DEG pipelines only](https://github.com/CCBR/Pipeliner/commit/77acdb2ee8ed4891e52990810fdba22c810bbd99)
* [Updating DEG Reports for mm10_M21](https://github.com/CCBR/Pipeliner/commit/392c2663583b3a03f8759bf52569e68f25252ff0)
* [Updating karyploter's usage to include mm10_M21](https://github.com/CCBR/Pipeliner/commit/734b16eec9c544be587d59dd0ca164931d1803e2)
* [Adding "mm10_M21" to drop down reference genome](https://github.com/CCBR/Pipeliner/commit/ef24dd86adb7e9ed544b607b3716dc7ba6c9d824)
* [Adding "mm10_M21" Reference Genome: Built using version M21 (Ensembl 96) annotation](https://github.com/CCBR/Pipeliner/commit/1077ced925beb59abf18580c07dcb723e5bbcd69)
* [Adding __pycache__ to gitignore](https://github.com/CCBR/Pipeliner/commit/45841878a9b7222952f4738caf7fe0ed62306491)
* [Merge pull request #429 from kopardev/activeDev](https://github.com/CCBR/Pipeliner/commit/dc0d10e503fceea160c98639e711abf12de4fc45)
* [tidyverse added to genecounts.R along with writetogz function](https://github.com/CCBR/Pipeliner/commit/7849530e04a23b2265461421dd35563ae46c7469)
* [trim_se is now temp too](https://github.com/CCBR/Pipeliner/commit/090c8f02985f605ebdeabf0bfe12bffb6806ef81)
* [tmp is actually temp](https://github.com/CCBR/Pipeliner/commit/2d8f342de64484bc6066ed5030cb70d9228332f5)
* [tmp is actually temp](https://github.com/CCBR/Pipeliner/commit/1fd2ddca3d4db26e56e332148148956cf361dde4)
* [make trim files temp](https://github.com/CCBR/Pipeliner/commit/6bd98fcd0dbc4768fb78ddf077d304fb5df9657a)
* [not usin g deeptools anymore for bam2bw ... not doing any smoothing for bigwig files](https://github.com/CCBR/Pipeliner/commit/1ae01455a2391f1b3a6cbd1072decc1bc2755fc3)
* [bringing back bw creation](https://github.com/CCBR/Pipeliner/commit/cc68b8c5393471e922706ffa4690c122ee129478)
* [Merge remote-tracking branch 'upstream/activeDev' into activeDev](https://github.com/CCBR/Pipeliner/commit/e69aec24f837ba1eaa7ec17e2a0ddaf85d5a400c)
* [Updating STAR from 2.5.2b to 2.7.0f for all RNA-seq references genomes](https://github.com/CCBR/Pipeliner/commit/d4159d6e7748b364d78222d9f1b44e1489613743)
* [changing PcaReport.Rmd genecounts.R rsemcounts.R to account for the scenario that we have only 1 group](https://github.com/CCBR/Pipeliner/commit/e3b3407af7e32271f41cd6c66ab748b090aeafa1)
* [Adding compatibility for "ref_vAnnotation" formatted reference genomes](https://github.com/CCBR/Pipeliner/commit/37c34bc4279572af183dbaa16e1306d7866f311f)
* [Removing Strand-specific BigWigs as a target file](https://github.com/CCBR/Pipeliner/commit/c974e2a33005b677654d4d2a930268c6aad2d97b)
* [tabs converted to 4 spaces](https://github.com/CCBR/Pipeliner/commit/f26a19941ac251bae08235322c03678ac107d751)
* [bam2bw bc fix](https://github.com/CCBR/Pipeliner/commit/a6d8bda3be02d5d13dbf4f526d177c3504f75ee1)
* [renaming bam2bw in rnaseq to bam2bw_rnaseq... and increasing its ram to 120g and cpus to 56](https://github.com/CCBR/Pipeliner/commit/43b1cec7cc7d486206dde3036744db2ff6c69dc3)
* [Adding hg38_30 to pipeliner's user interface](https://github.com/CCBR/Pipeliner/commit/2b367ba59cefd0cba9b988acd7ff7c7b50f1f87f)
* [Adding support for hg38_30 to karyoploter.R](https://github.com/CCBR/Pipeliner/commit/10edf55929f51cf5239938368220268eeea37fdc)
* [Updating DEG Reports for hg38_30](https://github.com/CCBR/Pipeliner/commit/9b348301c105f360ea8151365f541a1814a36b6d)
* [Updating PATH to KARYOPLOTER](https://github.com/CCBR/Pipeliner/commit/14c5cb078f5c9817f36494e771643b006c0edec2)
* [Updating the version of STAR from 2.5.2b to 2.7.0f](https://github.com/CCBR/Pipeliner/commit/5f11b91f5e614d3c1f234da885e2144e54e942c9)
* [Adding hg38_30 references using GENCODE version 30 annotation of the human genome (GRCh38)](https://github.com/CCBR/Pipeliner/commit/22d67c65e020fa870c0cab98114dcd33d2cf5d4b)
* [Merge pull request #428 from kopardev/activeDev](https://github.com/CCBR/Pipeliner/commit/6ac7383d3b3add034484794cd9c156b15d07a08a)
* [added missing tail -n1 bam2bw rule](https://github.com/CCBR/Pipeliner/commit/cbc0fec56d829a3ed223016eb70f33b5732c0176)
* [Merge pull request #427 from kopardev/activeDev](https://github.com/CCBR/Pipeliner/commit/4173ccedf9adeebda1c8be7f2047588eebf9e690)
* [D1.0 type fixed to 1.0 in bam2bw rule](https://github.com/CCBR/Pipeliner/commit/81836701167bf417cf30a8266855f1912b612051)
* [Merge pull request #426 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/eb80395dee354d01c3e1255668d9bfbfb60815e1)
* [updated STARfusion and FusionInspector, added merged final fusion outputs across 2 callers](https://github.com/CCBR/Pipeliner/commit/e656192da80a1f0f15ff3b87a88cadbcb4c6ba48)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/3a53d0f2169de7af6b054a4b2acc5ef9f00ff5b6)
* [updated STARfusion, fusioncatcher, and FusionInspector versions](https://github.com/CCBR/Pipeliner/commit/0d4e54513ceb4e0f9824b54ba81c59c809ca741e)
* [cluster.json ... partition fixed back to norm only](https://github.com/CCBR/Pipeliner/commit/fa276d46a87c23994e30ace1d1558703144c1093)
* [Merge pull request #425 from kopardev/activeDev](https://github.com/CCBR/Pipeliner/commit/a3340aed78b478d6c9eb8f51db51a799a2d85f03)
* [Deeptools rule now outputs bigwig files to bams folder](https://github.com/CCBR/Pipeliner/commit/a48f458766c455e5707a480b3fefae141d3c266e)
* [Deeptools rule all wildcard fix](https://github.com/CCBR/Pipeliner/commit/4161c6e165c6f9cef8f007c43a4676305c3d00f0)
* [Deeptools rule all indentation error fix](https://github.com/CCBR/Pipeliner/commit/a1283f5eb745b2cb31c8a5f8e8c258685641d4c5)
* [rule all edited to add bam2bw outputs](https://github.com/CCBR/Pipeliner/commit/d667ab51d54b4e8eaaa2fd53ca7daed85fd2dec1)
* [Merge pull request #424 from kopardev/activeDev](https://github.com/CCBR/Pipeliner/commit/7234355c5da237a6daed4175013de28a7fd8a231)
* [rpsd fix](https://github.com/CCBR/Pipeliner/commit/7337e36025d2df0fea5ce442228c916c42c90f1d)
* [forward-prob fix for rsem](https://github.com/CCBR/Pipeliner/commit/3d077d47a6bd2a5342077d696ad22f48ec793333)
* [file2 defined for se rsem rule](https://github.com/CCBR/Pipeliner/commit/2cf3eaa799bd3f0ef7bda37f1d3e8bc3ab5f94e0)
* [adding strand specificity to rsem counts command; adding rule to convert bams to forward and reverse bigwigs for easy visualization](https://github.com/CCBR/Pipeliner/commit/c6472ed1fad86ba9f56da18c771188709ca8c32a)
* [Merge pull request #423 from tovahmarkowitz/activeDev](https://github.com/CCBR/Pipeliner/commit/4334dd59b04fb2743a39e6ea913168ee875eb994)
* [fixed bug in gem rule](https://github.com/CCBR/Pipeliner/commit/71a2562f5c2ad24224e2f22a391f44479bbef9db)
* [DiffBind,uropa,homer,manorm](https://github.com/CCBR/Pipeliner/commit/5f327211324f3052b23b3d3c8d5653040aa6fc68)
* [homer,uropa,manorm,diffbind,jaccard](https://github.com/CCBR/Pipeliner/commit/4e0fabbed9ff550e0dd3f6b422077cb2674f400d)
* [added BCFtools variant QC](https://github.com/CCBR/Pipeliner/commit/525dc930eb6fa1a9ca31765cfb140efb02e8e6ae)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/a2ce46bed52f584bc1e2f2950d2a038e12c1ff2e)
* [homer,manorm,uropa,protcoding](https://github.com/CCBR/Pipeliner/commit/45ede8c129d5db282ee53a29336375b981cfa911)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/2075e2c87f4f5743d7711a250009abe56afe7ecb)
* [added new fusion rules](https://github.com/CCBR/Pipeliner/commit/acc9d4d7c157fb2ae1b9062086dd91e262acf2ea)
* [removed target selection from polished germline WES variants; added BCFtools stats QC; substituted Agilent SS v7 target BED file for default targets](https://github.com/CCBR/Pipeliner/commit/90b1534dc3947b2fde7d3bad666ea283975eb7ea)
* [update to mm9/mm10 json files... blacklist bwa index now includes chr_rDNA as well](https://github.com/CCBR/Pipeliner/commit/dfb9016188c95b5bb5d07113d704a6d952dbef39)
* [merging with main branch nggerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/bd051b1c36b59aea203941ba78cba5c151598e5b)
* [Changing effective genome size and adding differential binding analysis](https://github.com/CCBR/Pipeliner/commit/38d74eee130ebc0bfd7f43ee9b9a8ab4c0520ac8)
* [Added warning message to "karyoploter.R"](https://github.com/CCBR/Pipeliner/commit/7600f40cb5f5879770b537e208d7c1d0f1393bb9)
* [Adding DEG Type prefix to output karyplot to prevent race conditions](https://github.com/CCBR/Pipeliner/commit/7b33da1cf71ff51ec3d0a6b4857d6094c81dc749)
* [Adding last reference files for "karoploter.R"](https://github.com/CCBR/Pipeliner/commit/4090cbae8b05a38f918356f7dfbd84d674e58703)
* [Adding gene expression karyoploter script](https://github.com/CCBR/Pipeliner/commit/88a7e65ace5f7f92b97a33ec4e582105f806472c)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/4b01ce4ca9c4ab0373f1e75619740f6645f213ce)
* [Marked Qualimap Counts to be moved to DEG](https://github.com/CCBR/Pipeliner/commit/e2bb8bf8190f62d4166bc38fa098e347c364a443)
* [Updating GUI title](https://github.com/CCBR/Pipeliner/commit/eb766fcb8192549001bd0637db1c2a8771e1c6fd)
* [Updating version to "4.0" for activeDev pipelines](https://github.com/CCBR/Pipeliner/commit/7d5c0778ea2b288de457ab3786efa0b20a04d7da)
* [Updating "activeDev" Pipeliner branch version to 4.0](https://github.com/CCBR/Pipeliner/commit/23340faed56873eaaed6413367443af24d4cfa76)
* [Merge pull request #422 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/e730d1d84ec3b66e2c6b23e728a293ab8f7661e4)
* [Adding "karyotxt" reference file to DESeq2, edgeR, limma calls](https://github.com/CCBR/Pipeliner/commit/75f154fef4145d4bbe57af452b7546fc5e4fcecb)
* [Updating LimmaReport Rmarkdown](https://github.com/CCBR/Pipeliner/commit/38a834184f968020d4d422a20c79d0ba05f4964b)
* [Updating EdgerReport Rmarkdown](https://github.com/CCBR/Pipeliner/commit/f85070fb6a0a1798dc574b50956adaeeedf01439)
* [Updating DESeq2 Rmarkdown](https://github.com/CCBR/Pipeliner/commit/1ffa5b36fd4c15657905a41b4c166a2575d12f7f)
* [Adding karyoploter reference file to organism json files](https://github.com/CCBR/Pipeliner/commit/20e76b0184e740fcf6d9c7beffa8c84990898a61)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/16cb9170f4372924b725cdf0cf1b54dcf1ac7e0d)
* [expanded MAF annotation to merged somatic variants](https://github.com/CCBR/Pipeliner/commit/8df9f5da77f6416729cdfecb2a6fc98d9b5ebff7)
* [Updating EBSeq.py Script](https://github.com/CCBR/Pipeliner/commit/7336554cd30aeae834d00b5ae053765d5c6b45cd)
* [Updating Ebseq rules](https://github.com/CCBR/Pipeliner/commit/012965f38de51287bed206f370801470dfa5f275)
* [Adding 'hg38_HPV16' to GUI](https://github.com/CCBR/Pipeliner/commit/16ecfca6c1a9fdf09ae48cca8c697326fd9baeec)
* [Adding resources for hg38_HPV16: hg38_HPV16.json](https://github.com/CCBR/Pipeliner/commit/7584175ac3bd26d2e273996b132792369e73ea04)
* [Merge pull request #421 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/baf646f5c928c1c857320978f448510af611566c)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/9cc7fae271e12d9920ff908835bcdae8b4ad7986)
* [reduced verbosity of VarDictt to decrease slurm log file size](https://github.com/CCBR/Pipeliner/commit/1cdea7d0cf45c472578825e0e42ad4128e2ec6d1)
* [Adding hg19_KSHV reference files and updating GUI](https://github.com/CCBR/Pipeliner/commit/e68d788799aa008709578d7268b88012cf37744f)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/7058e92867f12dcbf497fd182a1cb459eab796c0)
* [Updating QualiMap commands](https://github.com/CCBR/Pipeliner/commit/1a42a00bb4cc380667d78d263de36bb2bf1a3969)
* [Merge pull request #420 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/5babfabbbf2317add2bac77aecf02e650a4f3bc4)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/21d48ac7518dc87e1b8434768d4fe52bb825ee17)
* [added multi-caller somatic variant detection to tumor-normal pipelines](https://github.com/CCBR/Pipeliner/commit/0cb02628f9a008ca8d700325ba68df2c2f21ffd6)
* [Adding QualiMap bamQC and QualiMap counts](https://github.com/CCBR/Pipeliner/commit/0abb32e102745e7125a10b239b81cedfabc32aba)
* [Updating path to RNA-seq MultiQC config file](https://github.com/CCBR/Pipeliner/commit/eef42b4f4d6f68f8efccbf2de1428c47f39df967)
* [Adding adding qualimap sources to json files](https://github.com/CCBR/Pipeliner/commit/a6355f9d1adc664ccbb5124a30067a59f4b5c9b1)
* [Adding support for edge-case: group CPM and MINSAMPLE filtering 0 0 (i.e. no filtering)](https://github.com/CCBR/Pipeliner/commit/5679c242cc7039337d25a35daa5460d7f195a8be)
* [completed multi-caller additions to tumor-only somatic pipeline](https://github.com/CCBR/Pipeliner/commit/dbf4d753725290e3d42dc76db492bb2ca515c0fe)
* [Adding rule 'pca' to cluster.json](https://github.com/CCBR/Pipeliner/commit/4c2ee26aa2afa598e48edec7d97bb10698d15e36)
* [fixed bugs in expanded WES tumor only pipeline; changed admixture outputs to all be printed in admixture_out; incorporated vardict somatic calling](https://github.com/CCBR/Pipeliner/commit/f00d7cbc6a0f54ae49f67fd53fa7916c4138e3a8)
* [added Vardict tumor only somatic caller, added joint calling with 3 callers for tumor only calling](https://github.com/CCBR/Pipeliner/commit/9b15248742981172b2efd37511ff5e3c6c038fc5)
* [improving HOMER/UROPA speeds](https://github.com/CCBR/Pipeliner/commit/c52fc9527d530328cda4d682d75ebfd53e817643)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/db164af37cb9be8de0c9516087a7ceba6b2fa687)
* [Merge pull request #417 from tovahmarkowitz/activeDev](https://github.com/CCBR/Pipeliner/commit/6349c7ae176349f271e0a1fc89b52d13f9da6f64)
* [adding spearman correlation heatmap to multiQC](https://github.com/CCBR/Pipeliner/commit/66d2cb021876430f8c79da58cbafb2256d28461e)
* [adding to the multiQC report](https://github.com/CCBR/Pipeliner/commit/cfd6a329cc6271d5c756884c0fc3d5036b9c4b05)
* [removed mtDNA from germline genotyping](https://github.com/CCBR/Pipeliner/commit/2a32b661418a812d98aead9855c974638e25c71e)
* [Updating 'rule QCstats' to grab the fragment Length](https://github.com/CCBR/Pipeliner/commit/b5c7acb406c2b331b5cc99de3fcabc6c5922f0f5)
* [Updating to 'filterMetrics' to accept fragLen as an argument](https://github.com/CCBR/Pipeliner/commit/6df704d62cb46b7180f32579c12105589384ddaf)
* [improving multiqc report, deeptools runtime, and other details](https://github.com/CCBR/Pipeliner/commit/51e94d63fc22cf268b9ded8db205e63d4262435b)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/4863a7b8db6b46b308f8c5f67e0a10cdce258d1f)
* [fixed bug in admixture plot generation](https://github.com/CCBR/Pipeliner/commit/50fed72562e1fead35a95ac3b47afbd2e6a1fb24)
* [Resolves #416: use the project directory's 'cluster.json' file](https://github.com/CCBR/Pipeliner/commit/282b2580693946a2169f767cc01be883f820fad5)
* [Merge pull request #408 from tovahmarkowitz/activeDev](https://github.com/CCBR/Pipeliner/commit/290b7c7a3b717343034ef13ce00b45765a10e418)
* [begin tracking R scripts for generating sample network and admixture PNGs](https://github.com/CCBR/Pipeliner/commit/647a0879bae97bab4a025dd9532dfda2ecfd6845)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/be01e5ad46f687f946b3e97f48cf150625f6703f)
* [added conversion of sample network to PNG and embedding it in multiQC report](https://github.com/CCBR/Pipeliner/commit/5198cc89540602bf5e63d750db76dd5990d66018)
* [updates to gui by Vishal](https://github.com/CCBR/Pipeliner/commit/f6b1ec16299ea00986f7fca23192f881c8ba21bb)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/7a8cec3bfab0bf69bc3539bec7a6903275f692f8)
* [Resolves #415, added canFam3 and Mmul_8.0.1 to gui](https://github.com/CCBR/Pipeliner/commit/f3deb2fd1893482b70d01c2fc03e6dbc0c6db08c)
* [added summary figure generation and P-value annotation for FREEC CNVs](https://github.com/CCBR/Pipeliner/commit/3de4652a64310d8867e3629017e63a987b7b9f83)
* [Adding Macaca mulatta (Mmul_8.0.1) and Canis_lupus_familiaris (canFam3) reference files](https://github.com/CCBR/Pipeliner/commit/224aeb34466454e13c387d78a4a79b601c46dafe)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/02191eabadbd266ea907b92a7fcbaeaa8796ba5d)
* [adding vardict calling and multi-callier merging](https://github.com/CCBR/Pipeliner/commit/6673df4a85ca412cf9343dafb5e486036beec5e2)
* [e for your changes. Lines starting](https://github.com/CCBR/Pipeliner/commit/23604f65f40208f0036ff1a28dd65eba24f68f3b)
* [rsemcount.R but fix... columns were getting duplicate names](https://github.com/CCBR/Pipeliner/commit/5ebe1264651920aa768df855304d6e2718c8cb6b)
* [Updating STAR-p1 arguement: "clip3pAdapterSeq" to "--clip3pAdapterSeq"](https://github.com/CCBR/Pipeliner/commit/0b8070775e141ebeeaddcfdf2bfc280decf2507b)
* [contrasts.tab now accepts 2, 3 or 4 columns, accepts numbers or fraction as column 4, fraction as to be >=0.5 and <1, numbers have to be integers; both numbers and fractions now denote number of samples or fraction of samples PER GROUP where GROUP is one of the groups in the contrast. Filtering of genes in DEG_ALL is changed from 0.5 CPM cutoff of atleast 2 samples out of all the samples to 0.5,0.5 setting ie., 50% of the samples PER GROUP should have atleast 0.5 CPM.](https://github.com/CCBR/Pipeliner/commit/9afe4bb5189267a874c02120cef10976b47e3374)
* [more memory for ngsqc](https://github.com/CCBR/Pipeliner/commit/2b56fcfa6ea173bd26600c893d6aaba95c508087)
* [multiqc update and uropa update](https://github.com/CCBR/Pipeliner/commit/e4158dc4ab073f8c7f777ff2658ec86e5dac1f42)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/d0575c89982732fff473b10f3516ee4ad19d567f)
* [Merge pull request #406 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/74827afc65d0108a4e131adda1828a9510919658)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/a21564907418c681c8b639f67bc2538aa2b7dcdc)
* [fixed bug in prep_mafs.pl script](https://github.com/CCBR/Pipeliner/commit/7129d464a120b69b3e25579424f0afb9744d2cdb)
* [Updating ambiguous Subread genes output filename](https://github.com/CCBR/Pipeliner/commit/9530af736039744d145ed78f40a9b77e0b727c13)
* [uropa fixes and cluster.json update](https://github.com/CCBR/Pipeliner/commit/e7a7772ed9c5b67751c809cf5d17a8cedcfc768c)
* [filtersamples.R: changed .. 4th column in contrasts.tab now denotes the number of samples PER GROUP that are required to meet the CPM threshold denoted in column 3 of the same file. If number in column 4 is greater than number of samples in the group, then MINCOUNT is reset to the number of samples in the group.](https://github.com/CCBR/Pipeliner/commit/603bf223532cc60cc38b57f2f468711022ca7c2c)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/4c5bb9858152b173504183cea8d609e79a3a018b)
* [Resolves #405, 'preseq lc_extrap' approximation](https://github.com/CCBR/Pipeliner/commit/b0d993306b0e5ccff00287d299c9f8e9c9f2ebaf)
* [uropa temporary workaround](https://github.com/CCBR/Pipeliner/commit/35a7acf743eb015af1f6c6bc8d85f22d1b764c75)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/347cb5697def53d35036795cd20a0825594f35b6)
* [Merge pull request #404 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/df316c998d788a8711ef2c907a32924b1dd5c617)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/8a9a669d46c6d74f2db7e893ae07f4193b4a1383)
* [improved somatic variant fitlering and maftools summaries](https://github.com/CCBR/Pipeliner/commit/e534349afbe4fbc660304d082bed4597d50fd77d)
* [merging branches](https://github.com/CCBR/Pipeliner/commit/e6dc1422b8d8822ea4e6e9d820d9bd4cb652968d)
* [adding uropa and other changes to chip-seq pipeline](https://github.com/CCBR/Pipeliner/commit/bb74f19297498dc0e382c65f627348ae8be72e8d)
* [PhantomPeakQualTool: Bigwig Generation from Estimated Fragment Length, resolves #403](https://github.com/CCBR/Pipeliner/commit/1039f14977df2a4b8cb516b2be80cb8723a1f3dc)
* [Updated Heatmap Height and PCA report generation (when > 100 samples)](https://github.com/CCBR/Pipeliner/commit/b1589c51bc9c8d97c94a6db254534a482ab64cb6)
* [fixing issues caused by lack of input or PE sequencing data](https://github.com/CCBR/Pipeliner/commit/80788a3195e0727d3d311044ef604ba9a9d97a0f)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/712a68035e1876288e5aede53ed17e63e4ca691b)
* [Merge pull request #401 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/1d20ea1f73e818903ea2024772745b3f22083120)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/ae0d41fc5b5f2124322dd6af428c6b6b63dbb104)
* [fixed bug in WES BED file generation where empty files are written when an incorrect path to target bed is given](https://github.com/CCBR/Pipeliner/commit/2b698e6f7c1a23612912797880252306cbbbcf33)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/cd72f4a3a9b25e41157abc928398a277604d0678)
* [Merge pull request #400 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/cb2fedc760e7e35efc01974b65f1584f201c799d)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/c3f95fdae0fb33a82c5a4287aa1625b1c85dccfb)
* [moved sort out of STAR and into samtools rule to deal with memory bug](https://github.com/CCBR/Pipeliner/commit/a8e9aa83c088456309e45b523902433e01847930)
* [PcaReport Rmarkdown Fix: Before Normalization/DESeq2 #399](https://github.com/CCBR/Pipeliner/commit/dbd5052c02fe061076995cdaf8517860cd866770)
* [reworked ChIPseq.snakefile with improvements](https://github.com/CCBR/Pipeliner/commit/0e56f041dd3a4613dc19f1c8a499e7d880b98642)
* [Merge pull request #394 from tovahmarkowitz/activeDev](https://github.com/CCBR/Pipeliner/commit/fc7345e95b9a14cc607f1d31b62493c4e90a54ce)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/cba940382b4856594f3936d7e19d260a148418f1)
* [Merge pull request #398 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/48e868587ec2726672b0c4441f97a76806794264)
* [added memory limit to STAR 2-pass in rnaseqvarcalling](https://github.com/CCBR/Pipeliner/commit/8ca55b3ad564c5c88f511ac443c6b7c93c99f985)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/a5e7c58d5bd4ab1ad8401f793834c91baa13719e)
* [fixed bug in 2nd pass of freec](https://github.com/CCBR/Pipeliner/commit/effa650c5cf51405dfe8da46d071d5ef5ba7dea7)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/8b837a25da3580b1b016205d5be69420eccff263)
* [Merge pull request #397 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/ae68b5f1a1a928c686d12fd7150cf4e938b14d5c)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/138c52f91818154b4934f06002388d4f25684d34)
* [fixed bug tracking FREEC 2nd pass output](https://github.com/CCBR/Pipeliner/commit/e26e107bdeddeb978f935c8785e1fd61cb122fff)
* [gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/6de40056823fb1733793fa8aa5604c439763ba76)
* [Merge pull request #396 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/00d2f3cb32b6c646cd5f47f8d51fbe2830bc956a)
* [completed new CNV workflow for WES and WGS](https://github.com/CCBR/Pipeliner/commit/9683091965c0a9c757fb76fcdef75ca436c7201f)
* [implementing 2-pass FREEC and Sequenza to WES and WGS pipelines](https://github.com/CCBR/Pipeliner/commit/8ce26e8a2c65c35b6edeac5ea23cf97c37796293)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/d3ecf3c198af8ce7b74218aa7601ff1c30543241)
* [completed sequenza rule for WES and WGS, and completed second-pass rules for 2-pass freec](https://github.com/CCBR/Pipeliner/commit/d01d82ce8b64119f6b03220fe8a6b1225a364834)
* [added multipass freec CNV analysis to use ploidy and purity prior from sequenza](https://github.com/CCBR/Pipeliner/commit/a5135e4949034d415e33ae9baa339186cc01fc0b)
* [tracking new files and scripts to run FREEC and sequenza](https://github.com/CCBR/Pipeliner/commit/da54b3c49f2e85288e6b8ecf44e77b95f7871a6e)
* [added FREEC to tumor only pipeline, and sequenza and freec to tumor-normal pipeline](https://github.com/CCBR/Pipeliner/commit/f95d3f3966f6e1c85dc160a785028ef7884fcd95)
* [updated Canvas germline to most recent version, and fixed bugs](https://github.com/CCBR/Pipeliner/commit/56eb3a22ec4254f558e2308a213fa33fc3616869)
* [updated Canvas and fixed bugs in Canvas somatic CNV calling](https://github.com/CCBR/Pipeliner/commit/cc0950bdc6ab164bbb80765965e2498bce5f08ef)
* [fixed error with ngsqc_plot rule](https://github.com/CCBR/Pipeliner/commit/abb6466c0155829d5ef51dea90079e2008af4336)
* [adding ngsqc plot feature](https://github.com/CCBR/Pipeliner/commit/097280f65cbb8ff3adf6405e60932ff862d939be)
* [adding macs2 dedup functionality](https://github.com/CCBR/Pipeliner/commit/f76552758659b5932d8c49994fb4956f97549e8b)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/f27d54d61a70acf89cc7f5787397957d221e2bfe)
* [added full germline calling to wgs tumor-only pipeline](https://github.com/CCBR/Pipeliner/commit/de7b4d2ab0347f33b7ce8ac438edf5cdd796cb9d)
* [added full germline genotyping to all somatic pipelines](https://github.com/CCBR/Pipeliner/commit/bedbd58bd3a8f88cde594bde92b8be94ad465bc4)
* [Merge pull request #389 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/063d63828e3b3e88fc5d88dd5132def2ac4ad6f8)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/a18b3d566586ef9d4631b76e45acbe0d9916fc0c)
* [updated GATK version to 3.8 in combine somatic GVCFs, and switched to filtering all MAF files with perl script rather than in maftools](https://github.com/CCBR/Pipeliner/commit/287069e9067fa40445a85a89465c9ddadf6e79a4)
* [Merge pull request #388 from tovahmarkowitz/activeDev](https://github.com/CCBR/Pipeliner/commit/65cdb13ffdce911f0f52965a162be2b1f6657814)
* [fixed fingerprint plot and multiQC](https://github.com/CCBR/Pipeliner/commit/81b654d61614be1ada9148f961b180ab9aa77b82)
* [Fix: WES Slurm Master Job Failure](https://github.com/CCBR/Pipeliner/commit/e1795c0010ecf0b0424b2ab87001811f0729e95f)
* [WGS/WES Reference Genome Selection Warning Message #377](https://github.com/CCBR/Pipeliner/commit/a4e2cb1f64b889e81198d77f381f5c0f143cd043)
* [Merge pull request #376 from tovahmarkowitz/activeDev](https://github.com/CCBR/Pipeliner/commit/59531998a77c483159eb98c8b219a4398fe16d19)
* [Update cluster.json](https://github.com/CCBR/Pipeliner/commit/7666d4247cf18ed25b8a6bac069c99ebfe2afce4)
* [Initial ChIP-seq QC DAG SVG Generation, Updated Peaking Calling Rules, Generated MEMEchip Reference Files #375](https://github.com/CCBR/Pipeliner/commit/9309efae58873bb259f711a65fcc2ce9c26634a0)
* [fixed ppqt](https://github.com/CCBR/Pipeliner/commit/4a33bd74406bf1e7a846cddc037c68f4c03abaf2)
* [bak files removed; cutadapt updated to 1.18 on rnaseq](https://github.com/CCBR/Pipeliner/commit/4c6c5fb844a29ecbbb87feb93c70be212cd5093f)
* [fixing .gitignore](https://github.com/CCBR/Pipeliner/commit/3ab0d0e8be537628e214c51d485ab5256e943468)
* [Fixed ChIP-seq Python Package ImportError #374](https://github.com/CCBR/Pipeliner/commit/85da38684a879382f8794a23670c5dc1e4b67b5e)
* [Fixed ChIP-seq Python Package ImportError #374](https://github.com/CCBR/Pipeliner/commit/477aecebac322aae9ed8fabee361baa2961a183f)
* [Merge pull request #1 from vnkoparde/activeDev](https://github.com/CCBR/Pipeliner/commit/21d2437a10f87bc68d3272d7e074a5e53c4b8b3b)
* [changing cutadapt in initial chipseq qc snakefile from 1.16 to 1.18 and adding a ppqt_pe rule](https://github.com/CCBR/Pipeliner/commit/1e4ead78963514acd67bc5a4475cbcf74ac6d04a)
* [actual purge and adding .DS_Store to the list](https://github.com/CCBR/Pipeliner/commit/63d4f46f09c32a8e0197b784545c43901f0c18b2)
* [adding .gitignore to ignore .* *.bak and *~ files](https://github.com/CCBR/Pipeliner/commit/601e537fe96068d3f6884aa028342effcdf24636)
* [standard-bin.json changed to change version of cutadapt from 1.16 to 1.18](https://github.com/CCBR/Pipeliner/commit/2d6487a2deb77a8defd1ce24cc0b594af2accfaa)
* [Rolling back InitialChIP Seq snakefile to commit ac47e92a20af8de3fe1ddd9344c613dc38f9cf63](https://github.com/CCBR/Pipeliner/commit/ece9a57576710bcb6f628cde0cc1b1d977aa057c)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/ef995f94a14b712c8c0d8f30f97fb6f99e8f0f30)
* [In PE mode readlength based filters to be applied to any (not both) of the the reads](https://github.com/CCBR/Pipeliner/commit/461027f4cdb0d88194b26e94d0b51b8d6600ae8d)
* [Merge pull request #373 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/ea998d9f4c53c9a4a3a443df76be5e47997c2484)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/44e3d03c7b505c96aa834051b95395339c56c395)
* [added automated filtering to prep_mafs.pl script](https://github.com/CCBR/Pipeliner/commit/b31dc57bc811ec0d650169087732f10828512777)
* [Python Package Import Incomaptiblity Fix: pysam](https://github.com/CCBR/Pipeliner/commit/e9e1d6780ae14ff29fb66ff7ba64c3a618b0207c)
* [chipseq:cutadapt to v 1.18, remove files ending in tilda, added .gitignore, added separate ppqt rules for se and pe](https://github.com/CCBR/Pipeliner/commit/62f008ad32ea8bb5ab1493845233bc812d00ee39)
* [Labels Testing: unbound variable fix](https://github.com/CCBR/Pipeliner/commit/362e3e8d3abcfec55e358ef0b56deda3a767cfbc)
* [updated genome fasta for rnaseqvarcalling for mm10](https://github.com/CCBR/Pipeliner/commit/ac47e92a20af8de3fe1ddd9344c613dc38f9cf63)
* [Labels file edge case: peak call pop-up box fix](https://github.com/CCBR/Pipeliner/commit/0ba2b674f05608e82f710a087e0bc4fbfba3ece6)
* [Copy peakcall.tab if labels.txt in rawdata directory](https://github.com/CCBR/Pipeliner/commit/efa6a25ba16431fb9b6d9195b8798dc6fd85c84a)
* [merge with main pipeline gerge th otherbranch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/5549023baf23f5250cc6d00d19d4c4bce91bd2a6)
* [ltered somatic variant filtering to be based on alt and ref counts in tumor sample, instead of allele frequency](https://github.com/CCBR/Pipeliner/commit/c394c4edca32714dcf599391a8df3b24231f6300)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/dc977d6ceb9395498ade7b88cb4551eea60e8d28)
* [Explicitly Loading Snakemake Module #371](https://github.com/CCBR/Pipeliner/commit/606964e162874ef7ada04b36fec25160a28444d8)
* [Explicitly Loading Snakemake Module #371](https://github.com/CCBR/Pipeliner/commit/34064433a4903f8ffdc55e26b4f037b97b94ae09)
* [fixed somatic directory creation bug due to snakemake version change](https://github.com/CCBR/Pipeliner/commit/29b8db54469f4aa770335131364063b6bc36f0ed)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/be1bbe28f97573c990368edc5da03ef72be85268)
* [fixed bug in loading VEP and vcf2maf for MAF generation rules](https://github.com/CCBR/Pipeliner/commit/0039502a264de3c2cbfdf0eacd1fc661ad285485)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev combining pipelines :gibincomninge enter a commit message to explain why this merge is necessary,](https://github.com/CCBR/Pipeliner/commit/51b1e823bcf13e3d403262c31113b6d445c93104)
* [separating fingerprint plots; merging inputnorm heatmaps](https://github.com/CCBR/Pipeliner/commit/dff8bae3f44b179f68664a31145b94d3834d2d1d)
* [Update README.md](https://github.com/CCBR/Pipeliner/commit/2f266a670c4ebb1524d45570a18eb6a7fbcf5d4e)
* [Update README.md](https://github.com/CCBR/Pipeliner/commit/33be5dbf0e4ec9cc99b64c38e4562d6f956c7246)
* [readding geneinfo to chipseq](https://github.com/CCBR/Pipeliner/commit/def151e7c92b284a6694899905d19f1c9a2a7118)
* [fixing merge issue](https://github.com/CCBR/Pipeliner/commit/e51d90ac647119adb7d7a47b4fc78ac037c62393)
* [merging with CCBR/activeDev gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/16219046e827ba998ab6beba3e015c483a269716)
* [Dynamically Resolve CCR BUY-IN NODES ISSUE #358](https://github.com/CCBR/Pipeliner/commit/7e072714e59e5607d2d2987d660e393346ce079d)
* [Dynamically Resolve BUY-IN NODE Issue #358](https://github.com/CCBR/Pipeliner/commit/b1953058609d3758162916c036dc780891a1f8bb)
* [RSEM CPM filtering Issue #357](https://github.com/CCBR/Pipeliner/commit/576bb9a442ea36374beeaec0872011ea1dd441dd)
* [json merge issue](https://github.com/CCBR/Pipeliner/commit/f2a9b3ae891659e3fd51177385c96f1c737ca211)
* [merging json files](https://github.com/CCBR/Pipeliner/commit/68d45f104192a96196db1e1b39671d899b926642)
* [Removed an include statement](https://github.com/CCBR/Pipeliner/commit/687ceba6b378603cbee39728d5897b024a45f9b7)
* [Updated GEM GEMONECHRS](https://github.com/CCBR/Pipeliner/commit/872c2a92f39c5c748999bba9b793ba96743bf462)
* [Merge pull request #356 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/d3291219d322487ed5f53ffb9c5c0e8f0bd6c9f0)
* [updated splitNcigar GATK version to 3.8](https://github.com/CCBR/Pipeliner/commit/076782d08c6c052b860c16d8dc3ece9820d74aa8)
* [Re-validated cluster.json file](https://github.com/CCBR/Pipeliner/commit/9cbdb92f6e024864fe67e54be095d8da5f0d3da8)
* [fix error with read extension](https://github.com/CCBR/Pipeliner/commit/e14d71a7c76bdd672f771648ed6b3f77f6aad6c1)
* [gerge with Justin's changes branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/2d83f280c2ab038938a1a8d0f51f793f8aa65186)
* [Changing Partition from 'ccr,norm' to 'norm'](https://github.com/CCBR/Pipeliner/commit/86941fc5dff331b18b8c7ac2800219c44675b307)
* [Merge pull request #355 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/22b95b3b14bb4e1a3c931e36f7b3a3aae86e8a97)
* [reduced wall times for rules that do not need max wall time](https://github.com/CCBR/Pipeliner/commit/df80c7381117fe05a6eb23d685771172e60f095e)
* [changed default partition to norm](https://github.com/CCBR/Pipeliner/commit/effb13bfc3331e534b6979b9a9f8d193097b33f9)
* [renaming bws](https://github.com/CCBR/Pipeliner/commit/c75005ea5ca75694bc636a8ca6ec8169d7afe475)
* [changing bw naming](https://github.com/CCBR/Pipeliner/commit/e4bc04612dc0078dc72c1ef3b21cc3e73a531927)
* [Update Deeptools PE Target Rule](https://github.com/CCBR/Pipeliner/commit/411a99e8d7ca60f0274d8f25e244d05d24024b18)
* [Resolving Deeptools Filenames](https://github.com/CCBR/Pipeliner/commit/4e30326b1b7b9b1e1a2348bf5414f8ca7313aafe)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/75ee4a1bc08f2d3a1f1799ed697c8f4a5871382a)
* [added indexing rule to all WGS and WES pipelines](https://github.com/CCBR/Pipeliner/commit/8eeb32685dacf72839cd05424c83e62301025b54)
* [playing with dependencies](https://github.com/CCBR/Pipeliner/commit/0f845a9efe010a0f8e8d8230d8ab60848ce486b7)
* [merge with other pipelines branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/add0d9c24db8481734d130cd7bb55ea7c958af65)
* [Merge pull request #351 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/a508b77f8fa8d23b0af5b86b90afb47d4f384dda)
* [repaired wgs_tumor_only bug in maftools](https://github.com/CCBR/Pipeliner/commit/94fa796e8b7cf398a23f413d70126788482b8a2c)
* [merging ChIP-seq with other pipelines gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/4bb47b673cf187dbe5b8b6c4eef343accb6204a4)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/fa07df2342641518e268e1c56286b12279ebb0e1)
* [CPM Filtering Fix](https://github.com/CCBR/Pipeliner/commit/24e800643530139f2d915d2304dbc0d414f4bb14)
* [Use Updated Sampletable for PCA](https://github.com/CCBR/Pipeliner/commit/8f0f3cc5cff240260eb01184a4a0706adbcf951a)
* [Export RSTUDIO_PANDOC](https://github.com/CCBR/Pipeliner/commit/62e3420d7caf3545b4cd65dd43d766962a22cd29)
* [merging with Paul's function gerge branch 'activeDev' of https://github.com/tovahmarkowitz/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/1072a13ff9a11087f59e2702f9155be0110bcaa3)
* [merging with Justin's changes gerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/267383c87718b62c3b3bbafec4ce03b4891e2501)
* [Inputnorm rule integration](https://github.com/CCBR/Pipeliner/commit/3819b9e1ff8b098dfc2d29ab06672f74174ece83)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/3878a8a170a4ab8a8d8e5862b3d0e0dfd2039064)
* [increased master job wall time to 10 days](https://github.com/CCBR/Pipeliner/commit/bdf2115b917fbac160d2cebdf84bfed9b9ee41b7)
* [Merge pull request #350 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/c22bf7c0c28e43f243331172c16b03fc34327a7a)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/fe124f6216d1113b4db33b95536f22dc4232a905)
* [made theta2 outputs dynamic to allow pipeline to complete under conditions where the theta2 rule fails](https://github.com/CCBR/Pipeliner/commit/ea70d094b86d27a50186312ae688438c6b8bf711)
* [read extension for bigwigs and dynamic deeptools rules](https://github.com/CCBR/Pipeliner/commit/3c5c76609c738466830f4cd3168567fc10706037)
* [merge with Justin's changes branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/f94657f4dfff19344f816122eedb02e57d4d7812)
* [Merge pull request #349 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/cc04a483a8af4fd711d19be735c8e8da5eb326f7)
* [added chr prefix to hg38 chromosome names in haplotype caller rules split by chromosome](https://github.com/CCBR/Pipeliner/commit/682850331d0d66164b898a7388c3c1b88c53ecde)
* [merging with contrasts.tab changes Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/0c5b04437ec286481a1fc864eb175738cab00666)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/8a9987e7cd72637e7bb8fe9326f85ed8b4fac96e)
* [added filtering for MAF files, and added additional summary plot and analysis of MAF files](https://github.com/CCBR/Pipeliner/commit/c96c899585e17a725b52614613fb86142f4d9581)
* [Safe-guards to ensure referential integrity during differential expression analysis](https://github.com/CCBR/Pipeliner/commit/419bc3364f902ef9ad05e484fced5a1ae4648a83)
* [Updating rule pca](https://github.com/CCBR/Pipeliner/commit/11e049bd5fb037c28740ffe60f1aa0595d3ae178)
* [adding heatmaps and profile plots](https://github.com/CCBR/Pipeliner/commit/1bfadf14bb213f3c2406154889f72b0a2639344f)
* [merging changes with ChIP-seq and RNA-seq pipelinesMerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/c05f54c39fd023ecc482681e2c4547349d987c35)
* [Pop-up box warning if 'groups.tab' or 'contrast.tab' are not in RNA-seq working directory before dry-running](https://github.com/CCBR/Pipeliner/commit/b97bf163dbb65306b536ed83f7651bda5a193416)
* [pcacall.R revert](https://github.com/CCBR/Pipeliner/commit/d908ec2ade22ffb9ae87fcc62dd6c84814b12fe7)
* [removed some pointers](https://github.com/CCBR/Pipeliner/commit/1a6c66e27018f7193e190f2c6a80d5bd183108df)
* [changed conditions for trim_pe and trim_se](https://github.com/CCBR/Pipeliner/commit/bc7de0614a3bc0ad0816947841acaa3dd68757b2)
* [combining structural annotation information and chip-seq file pointers Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/6967f367dd7689f79c0ed152a2dac0e892018fd9)
* [Merge pull request #348 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/f47775321a5672d58ec15f95e0e12c56ab55d595)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/be80367e63d96d36a88efbe031a54532c409924f)
* [split rnaseqvarcalling by chromosome](https://github.com/CCBR/Pipeliner/commit/50fe1e87836a07110a8afbd0f4d6b567a040f1e0)
* [Adjusting different rules in IniitialChIPseq.snakefile merge branch 'activeDev' of https://github.com/tovahmarkowitz/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/d3a1e5c44a34970ed9e7a441b8f8f7aad4c61797)
* [changes made to both exome-seq and chip-seq portions of hg19.json Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/0b029e8d67dafe96d0f1cf3d316d0dfb8dfd6ae0)
* [Merge pull request #347 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/cb3b65a13d6b3aab09461955260539c07c3a46a0)
* [fixed bug in location of fusion STAR libraries](https://github.com/CCBR/Pipeliner/commit/ce7629176ef5f981a583944be80a7773a4100325)
* [add peakcall.tab parsing and rule inputnorm (not implemented)](https://github.com/CCBR/Pipeliner/commit/fb15dd2423b4ad059204633ef9dcf3d895735691)
* [new heatmap rule](https://github.com/CCBR/Pipeliner/commit/825ac64a2ffe0a4ff3dc38773429e573c6f2727c)
* [adding GENEINFO to ChIP-seq](https://github.com/CCBR/Pipeliner/commit/120ab94bc8f7ec1d5227df346d70c75310065cd2)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/17243949e4ad167fcd735f5a586829fcca27aee0)
* [fixing thread usage](https://github.com/CCBR/Pipeliner/commit/bdb8a22396742481b96345416a6a9a7719056246)
* [Merge pull request #345 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/fa5b8b50ad581df0232b506e51ca144676337d1d)
* ['peakcall.tab' is copied over if it's in the data directory](https://github.com/CCBR/Pipeliner/commit/a4a0f20d24eea0858cddda6c8b3d2a00ad811eba)
* [bug fixes with deeptools_QC](https://github.com/CCBR/Pipeliner/commit/1fd121f67aad4e8a4b2fdee5ece6dde882bffea0)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/56aeedd76daff7500849d3c75e8596551caf423f)
* [increased the number of jobs submitted at a time to 500](https://github.com/CCBR/Pipeliner/commit/9c9b9e7d4d909bd16fb723c92fd0589310340b05)
* [bug fixes](https://github.com/CCBR/Pipeliner/commit/7c6c0f2a2832d26acc9f74c1b48ea489e9f2a0b5)
* [adding fingerprint plot](https://github.com/CCBR/Pipeliner/commit/0afe04bc1d30b908dfd70b0767863d319e470611)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev changes to RNA-seq merged with changes to ChIP-seq:](https://github.com/CCBR/Pipeliner/commit/4d8dd35dc9daec3ef7fd9bd9ac12b9c76d40f9df)
* [Merge branch 'kopardev-activeDev' into activeDev 344 pr updates](https://github.com/CCBR/Pipeliner/commit/3dfbefb8a76ce2f9888f27fd0678930a38126d79)
* [344 pr updates](https://github.com/CCBR/Pipeliner/commit/33fc78ad5705cef0e27e8b3fbb707cf4cf606102)
* [split haplotype caller by chromosome for all germline pipelines](https://github.com/CCBR/Pipeliner/commit/39f6ca37203c01be0f9d50006981da57ad6157c7)
* [removed Delly from all pipelines](https://github.com/CCBR/Pipeliner/commit/3ac58f5c781686c27d5f6b85625142b9d0ee4b0c)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/ebde723c4359cac3461b180aa962901bc4a30ce7)
* [added knwn ploidy config file to mm10.json for running Canvas](https://github.com/CCBR/Pipeliner/commit/350550c915300901e89d9159d22d9567e7bbc20f)
* [Rename symlinked bams to .recal.bam](https://github.com/CCBR/Pipeliner/commit/cee89f8014fd96eadb0e07a36d0128377ee2e64d)
* [Genome-seq/Exome-seq symlink update to work with labels.txt file](https://github.com/CCBR/Pipeliner/commit/21273c1057449b45c0cdab3c6169013595956cf2)
* [fixed bug](https://github.com/CCBR/Pipeliner/commit/02d518d38a92ae5a6e3c89339162a9dc0503ad54)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/7e6e9778e6b31660b000ffa3f0dd1d6291bd70e6)
* [GUI Changes: peakcall.tab now required for InitialChIPseqQC](https://github.com/CCBR/Pipeliner/commit/f0b040278bce738c6bdce5fd38637fd39bb4268a)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/816c2d558321821a5e17ff23d091e6da79aa7183)
* [Symlink BAM and VCF files for Exome-seq/Genome-seq](https://github.com/CCBR/Pipeliner/commit/b5e81e38df6c273ab76b9b6acff63aa3ec4f4574)
* [fixed rule connections](https://github.com/CCBR/Pipeliner/commit/d81826d78cc707a5488307dd1fa3e41aac6d1314)
* [fixed indentation error](https://github.com/CCBR/Pipeliner/commit/b53e16ca36bb016b407c39d4d2e2ce9a082eb767)
* [organized rules](https://github.com/CCBR/Pipeliner/commit/895e0001b2d1fc1449cd77e4e5b3f874d1c42210)
* [cleaning script for clarity](https://github.com/CCBR/Pipeliner/commit/82af20667f56d5d262e2d8a27d9c2f399aca9108)
* [consolidating rules](https://github.com/CCBR/Pipeliner/commit/d96a6524883deb3e9edc62b44754f0bf31f4c718)
* [updated MallDoneerge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/d9d07c5c883c5a66ab0f3c2061f644a74ee4d51f)
* [repaired starfusion bug and hg38 exome somatic only bug](https://github.com/CCBR/Pipeliner/commit/bf2950de67f085c99ac011aee006d93d4dfbb704)
* [cleaned up code](https://github.com/CCBR/Pipeliner/commit/385073abfd085ef76e01db6858ef894839aa877c)
* [Merge pull request #343 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/2e38f57d96a07d53b0d4150c33ad95252c7cd526)
* [updated versions for STARfusion and Canvas, and updated annotation resources for STARfusion to the most recent. Also, added multithreading to STARfusion](https://github.com/CCBR/Pipeliner/commit/5dfa18233205ce3d716efebe93b7506e704f174f)
* [fixed bug in all rule for wgs somatic tumor-only](https://github.com/CCBR/Pipeliner/commit/267e4a4d2126ac84095cc48b77a749e22a70284e)
* [stepped back to v0.9.1 for cnvkit due to python bug](https://github.com/CCBR/Pipeliner/commit/aba6142612056b7c7a31429feaaca9a0abeb6cb5)
* [RNA-seq DEG changes](https://github.com/CCBR/Pipeliner/commit/fe333e21ed99400792da35e5cee976fe101bbe9a)
* [Syntax Error: removing extra parentheses](https://github.com/CCBR/Pipeliner/commit/20bf985143d5028ec019f8e17abd754201da74a3)
* [Undo reading in files in binary mode](https://github.com/CCBR/Pipeliner/commit/41304a9a036abc47bbba777c7fb3741f3212dfff)
* [Low Abundance Gene Thresholds option removed from the DEG gui frame ... this data is now entered in the contrasts.tab](https://github.com/CCBR/Pipeliner/commit/1bf7a8ccf4bdfcc5886693f37b89aad5e8f20b3b)
* [rRNA list fixed for mm9](https://github.com/CCBR/Pipeliner/commit/e6653a4f8a299abd70f1edf387e60101d8db61f9)
* [Merge pull request #342 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/c4211e897e62c5f1e8f607a5e65a252ec5a1dd10)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/2a7e1cbdcfa0cd3e20693ac2a9b77f2f1b1337c8)
* [updated snpeff genome version for mouse](https://github.com/CCBR/Pipeliner/commit/037939d6f5f5032dc383df44e78b1f596a5e5a35)
* [Merge pull request #341 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/1b52c7dcc7dcf7b39cb1eee11abbb8435a9d14d6)
* [changed hg38 filter ExAC VCF](https://github.com/CCBR/Pipeliner/commit/1e341699edebcb0bc3d8d9365d1c1ebeb43af1a9)
* [adding warnings for selecting the wrong genome ... for eg. GRCh38 will only work with scRNASeq pipeline ... pop up warning if it is selected for other pipeline](https://github.com/CCBR/Pipeliner/commit/1a13002ef70a5c7df881a78c370ad56f87636328)
* [adding decoy genomes for RNASeq/ChIPSeq pipelines ... updating annotations for other genomes](https://github.com/CCBR/Pipeliner/commit/a7d19c09413ca94ca0a4196de39729b73f29a3e2)
* [new mouse RNASeq and ChIPSeq indexes included in json files](https://github.com/CCBR/Pipeliner/commit/888b362cd3aa429375fce135fd225e03d5d3406e)
* [Fixed indention inconsistency](https://github.com/CCBR/Pipeliner/commit/4f7740403c4797636cc5fc40e373a91252920f23)
* [QCTable NReads Fix](https://github.com/CCBR/Pipeliner/commit/b468c3546e325a909a683a413b518029a6f477cf)
* [add sample info to initialrnaseqqc gui; added mincount column to groups.tab](https://github.com/CCBR/Pipeliner/commit/1f3f8b38b688bb58c8d479b9b66169d3fd529e5d)
* [Merge pull request #339 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/9f8f08ed481f8463ad57564ca1877460d2b621cb)
* [updated hatplotype caller steps to GATK 3.8-0, fixed jdk inflater bug, and added index BAMs step to Dragen entry point](https://github.com/CCBR/Pipeliner/commit/977123b0f6998f7a50c5045cbd320b28f68deb1b)
* [Adding peakcaller GEM](https://github.com/CCBR/Pipeliner/commit/091b13f96f2feeee6e8d0e7f8deacc75a18a5f8c)
* [ChIP-seq QC Table](https://github.com/CCBR/Pipeliner/commit/477715accec22eb3398a80bdc69da8a780693e22)
* [Merge pull request #338 from jlac/activeDev](https://github.com/CCBR/Pipeliner/commit/8711aeb536f667377d5f2b7618742a1e3179261d)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev updating](https://github.com/CCBR/Pipeliner/commit/1da27b402e2b9166470879647fac08562becfb18)
* [updated Canvas version from 1.31 to 1.35](https://github.com/CCBR/Pipeliner/commit/98b2e8b755e90b7a20d2e448aa3df7a97f3f840e)
* [ChIP-seq Multiqc: Samtools Flagstat and Fastq Screen](https://github.com/CCBR/Pipeliner/commit/dc3340fc10f7c278367ec908980740a6c2596099)
* [Delete PipelinerVer1.0_documentation.pdf](https://github.com/CCBR/Pipeliner/commit/328b92c9d68e3383755a23e8e2379101e7c524c4)
* [Delete PipelinerVer1.0_documentation.docx](https://github.com/CCBR/Pipeliner/commit/23524d2e3872e65a5b2b6f6ad9a93ba1dd9e1024)
* [Delete PipelinerVer1.0_documentation.pdf](https://github.com/CCBR/Pipeliner/commit/2bf2ab1142f29ff2ada7eec701b80d6e7c3d6825)
* [Delete PipelinerVer1.0_documentation.docx](https://github.com/CCBR/Pipeliner/commit/d73e9c66f8d778e6d2cfa73a2971c75b23592280)
* [Filename bug fix](https://github.com/CCBR/Pipeliner/commit/16423fb408024a300649a77144a9ae2059981a84)
* [Snakemake DAG generation fix](https://github.com/CCBR/Pipeliner/commit/929c83c8218e651b19a54ea675d652136de283f5)
* [filtersample scripts](https://github.com/CCBR/Pipeliner/commit/865540fd1d5deb029d74db0794d780d5ce649323)
* [DEG_ALL folder ... extra column for contrasts.tab to have different cutoffs](https://github.com/CCBR/Pipeliner/commit/cc758c1b3dfdf7c2191131f7629191ef4e83e496)
* [increased memory for qualimap rules](https://github.com/CCBR/Pipeliner/commit/2ae45460ea2d74c90a0cf38c65300cf274409417)
* [Label Name Fix](https://github.com/CCBR/Pipeliner/commit/4672f7027fe0c7f2dc7760999ed57db6173341f5)
* [Merge pull request #332 from CCBR/revert-331-activeDev](https://github.com/CCBR/Pipeliner/commit/d4a80ff4f438561f57860b8c20c05680e0a22c3b)
* [Revert "test commit"](https://github.com/CCBR/Pipeliner/commit/1195565af206a7a3c6ce400c615490b6a38fd4fa)
* [Merge pull request #331 from pajailwala/activeDev](https://github.com/CCBR/Pipeliner/commit/879b5b4a525d4464d4bebc41c4729432e03b67d3)
* [test commit](https://github.com/CCBR/Pipeliner/commit/6ce187a3e17b3044e179d47f7b7dc26e99c6681b)
* [Venn Diagram Additions](https://github.com/CCBR/Pipeliner/commit/882aa1c0634f53e65307fc8428edf41037dbaaf3)
* [deleting test](https://github.com/CCBR/Pipeliner/commit/0067f30df7c99b9edc01a568f77c35cdd2948fd2)
* [Merge branch 'activeDev' of https://github.com/CCBR/Pipeliner into activeDev](https://github.com/CCBR/Pipeliner/commit/963bc331ac818776dca6fd8ed05e6ebfbff9f131)
* [Update README.md](https://github.com/CCBR/Pipeliner/commit/1572a568139da88a7701f13a0fb5aaca726f4e3f)
* [avia curl request fix](https://github.com/CCBR/Pipeliner/commit/311f923777f75ceba8d2f2d06370732037ab02fe)
* [test2](https://github.com/CCBR/Pipeliner/commit/995d727f6522206d367634aa5ef345d3b5d7cebf)
* [testing](https://github.com/CCBR/Pipeliner/commit/50741c9078b46128dff383517b8f6ad9e518638a)
* [Venn Diagram showing overlap of DE genes](https://github.com/CCBR/Pipeliner/commit/fc7b6f548b2a5dc75a612c47246d42890a0b18e4)
* [Reformatted Limma, DESeq2, EdgeR Output: RNASeq](https://github.com/CCBR/Pipeliner/commit/7bce1390aa0733ff1db53c5351bd74be8ef580dc)

## v3.0.5  

* [ChIP-seq bug fix and gui additions](https://github.com/CCBR/Pipeliner/commit/bfa24e22f518cdba5c527783ebb9476e2e284893)
* [Merge pull request #322 from nikhilbg/centosOS7](https://github.com/CCBR/Pipeliner/commit/a68a919703fb547be3fcf0a92bcc3443dd65675f)
* [minor bugfix for annotationdbi usage in scrnaseq](https://github.com/CCBR/Pipeliner/commit/f6290f124eefcb99d980b712e554d1b431c81328)
* [cell cycle regression option in scrnaseq](https://github.com/CCBR/Pipeliner/commit/a05b313e0e5fd86a3180af4d73d589cda2cc4e3f)
* [Nihkil changes for Single Cell](https://github.com/CCBR/Pipeliner/commit/5bacec9ffacbf2624d539f1c327e4fb7f12e72c6)

## v3.0.4  

* [centOS7 ChIP-seq compatibility and requested GUI changes](https://github.com/CCBR/Pipeliner/commit/9bea53ed92bc793d65afb2c7cad3b2b47d8caf6f)

## v3.0.3  

* [WGS and WES changes for centos7](https://github.com/CCBR/Pipeliner/commit/6516d590f56807e9be011a7792e7057fc7ad03c4)

## v3.0.2  

* [Skylers changes for centos7 / RNASeq](https://github.com/CCBR/Pipeliner/commit/24784bef0b5eeb3eeed9ea72426312de825298b9)

## v3.0.1  

* [bringing branch to the version on ccbrpipeliner (v3.0.2)](https://github.com/CCBR/Pipeliner/commit/3fa25a97851510c7c174b5bde3743824e140bc9c)
* [bringing upto v3.0.2](https://github.com/CCBR/Pipeliner/commit/6a081055bf028d3845a6ae34d83feb7878071a31)
* [v3.0.2](https://github.com/CCBR/Pipeliner/commit/b2863296fb2c4eb2e897125d08bcc147e4c91196)

## v3.0.0  

* [multiqc is now not running in DEG part](https://github.com/CCBR/Pipeliner/commit/997b26d11be674d3f7c3d173fc2606d215dcfa49)
* [Merge pull request #316 from jlac/master](https://github.com/CCBR/Pipeliner/commit/61972bafcc918c8259af83e4f3a18436267c6b1c)
* [updated versions for cnvkit and manta](https://github.com/CCBR/Pipeliner/commit/c58223a3eb123bdbaf4f0e737e54775b13ca059d)
* [Merge pull request #315 from abdallahamr/master](https://github.com/CCBR/Pipeliner/commit/f31de25c8785f6c1516e56b708e0d001008754d6)
* [This is for the cpm filtering as it was not doing cpm twice, I only commented out one line per file](https://github.com/CCBR/Pipeliner/commit/cf80d51fffa5022c9cedebf8092ede7b1d0c1b18)
* [Merge pull request #314 from jlac/master](https://github.com/CCBR/Pipeliner/commit/796b75389a6efee6cf0e7202f7c33299194d84e8)
* [removed realign and recal from WES somatic and germline calling stages of WES pipelines](https://github.com/CCBR/Pipeliner/commit/227067f6f2051079c279aca9695776f475a45578)
* [Merge pull request #313 from CCBR/v3.0branch](https://github.com/CCBR/Pipeliner/commit/cc75e682b610d99c4042cd0655d008341443f0ec)
* [data/work folder is copy paste on gui... workflow error fix](https://github.com/CCBR/Pipeliner/commit/ba1622cc66aa305f9a3bbb9d5f95c5356f3a7b92)
* [Merge pull request #312 from jlac/master](https://github.com/CCBR/Pipeliner/commit/c22456bd304e298ebb6d8c79039ba52378da92e8)
* [Merge remote branch 'upstream/master'](https://github.com/CCBR/Pipeliner/commit/a2556171ea5dd438f394446d248b05b4f35a85e8)
* [updates version of multiqc to v1.3](https://github.com/CCBR/Pipeliner/commit/7214b76ad4ccf70f6459d81bff4a2064eb404905)
* [Merge pull request #311 from jlac/master](https://github.com/CCBR/Pipeliner/commit/0234876f789e9cd214b9504fd34f7a2d35075584)
* [fixed bugs in fusion expression cleanup](https://github.com/CCBR/Pipeliner/commit/49d4c11e11a6c2be969839f2162d5bbd9d08d5c7)
* [Merge pull request #310 from jlac/master](https://github.com/CCBR/Pipeliner/commit/f8889fef7023427e0bcae50ed990d26f4d227432)
* [Merge remote branch 'upstream/master'](https://github.com/CCBR/Pipeliner/commit/4694b4e3bb9f1cc4dea3a09f5cb69c915602139b)
* [changed path to Oncofuse jar file](https://github.com/CCBR/Pipeliner/commit/9f080d856f31e4b449262fbdbd391c110f69dcd4)
* [Merge pull request #309 from jlac/master](https://github.com/CCBR/Pipeliner/commit/01a8fe866279d1bc7fadf9a2b5ba3e8d3783f3ff)
* [removed subread components from gene expression rules for fusion pipeline](https://github.com/CCBR/Pipeliner/commit/86b20b4056851ef76ddd3f77a3ed3ad5e86a6020)
* [Merge remote branch 'upstream/master'](https://github.com/CCBR/Pipeliner/commit/a7ac920c79a0e6acf10ffc7ae3af11e950c71643)
* [changed filtering criteria for MAF filtering](https://github.com/CCBR/Pipeliner/commit/47319f6f12935dcb0c3e7fe8ac7f361cf18c71d4)
* [fixed bug in processing tumor-only maftools rule](https://github.com/CCBR/Pipeliner/commit/ef31f08b9176d8c5d77b1692af3c7a0a15756c6a)
* [Merge pull request #308 from jlac/master](https://github.com/CCBR/Pipeliner/commit/1706cd0d1ea9aafa641172d88b5407f3306a7efb)
* [Merge remote branch 'upstream/master'](https://github.com/CCBR/Pipeliner/commit/99045db3f413b9f1286cfe2e6f823c1e12662973)
* [added filtering to somatic mutation calls; fixed jsons to repair missing resources in fusion and rnaseqvar pipelines](https://github.com/CCBR/Pipeliner/commit/f6614e8c3a6599f4083f80ca36699437703ecd55)
* [Merge pull request #307 from jlac/master](https://github.com/CCBR/Pipeliner/commit/fc70f228034685fc8d9345729cd3f811fe21eec4)
* [fixed Canvas version module](https://github.com/CCBR/Pipeliner/commit/7df59ea61cce96dedcd89e4839aa870c212437ad)
* [Merge pull request #306 from jlac/master](https://github.com/CCBR/Pipeliner/commit/b13808eb1a8436bf5e39f0e90f7df60b4d727d79)
* [repaired missing reference resources for fusion and ranseqvar pipelines](https://github.com/CCBR/Pipeliner/commit/c25067a04304f21c66913bcb404a23effcbf034e)
* [fixed bug in counting sample for mutsigCV](https://github.com/CCBR/Pipeliner/commit/b93793dc84b38fdc0afd2fd58ce2ff9c3dc10638)
* [Merge pull request #305 from jlac/master](https://github.com/CCBR/Pipeliner/commit/6b1c9888433b2c6dd1c304d7b89f37a36837a453)
* [Merge remote branch 'upstream/master'](https://github.com/CCBR/Pipeliner/commit/c2f0cd759a64251e807c2c48bbab770b91a70771)
* [added population allele frequencies to MAF filtering, and set mutsig a threshold of 10 samples or tumor/normal pairs for mutsig to run](https://github.com/CCBR/Pipeliner/commit/ebf060eaf7ad065f1bcf1c99e4c883940aedec3b)
* [Merge pull request #304 from jlac/master](https://github.com/CCBR/Pipeliner/commit/35e747f773fd4fa2e52f7a6e374854b6b506d052)
* [Merge remote branch 'upstream/master'](https://github.com/CCBR/Pipeliner/commit/6492220d870c29fc978c500f747eb5fefaec88a7)
* [fixed X11 graphics bug in qualimap](https://github.com/CCBR/Pipeliner/commit/b45c833d909be723ba7ce379bf8ce56684ce994a)
* [Merge remote branch 'upstream/master'](https://github.com/CCBR/Pipeliner/commit/8818d0e58ec1c9b068d90dfc38eb7acb0a01ea80)
* [removed extraneous file](https://github.com/CCBR/Pipeliner/commit/bf6ceb7742650662d128db7928c339ea1fb0a7d5)
