<!doctype html><html lang=en class=no-js> <head><meta charset=utf-8><meta name=viewport content="width=device-width,initial-scale=1"><meta name=description content="Pipeliner is an open-source and scalable solution for analyzing  next-generation sequencing data. Pipeliner provides access to  the same best-practices NGS pipelines developed and benchmarked  by experts at CCBR and NCBR. This repository is the main source  of documentation for users and developers working with or  contributing to Pipeliner."><meta name=author content="Skyler Kuhn"><link rel=icon href=../../assets/favicon/favicon.ico><meta name=generator content="mkdocs-1.4.2, mkdocs-material-8.3.0"><title>Resources - Pipeliner Documentation</title><link rel=stylesheet href=../../assets/stylesheets/main.fd896c8a.min.css><link rel=stylesheet href=../../assets/stylesheets/palette.cbb835fc.min.css><link rel=preconnect href=https://fonts.gstatic.com crossorigin><link rel=stylesheet href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback"><style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style><script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script></head> <body dir=ltr data-md-color-scheme=default data-md-color-primary data-md-color-accent> <input class=md-toggle data-md-toggle=drawer type=checkbox id=__drawer autocomplete=off> <input class=md-toggle data-md-toggle=search type=checkbox id=__search autocomplete=off> <label class=md-overlay for=__drawer></label> <div data-md-component=skip> <a href=#reference-genomes class=md-skip> Skip to content </a> </div> <div data-md-component=announce> </div> <header class=md-header data-md-component=header> <nav class="md-header__inner md-grid" aria-label=Header> <a href=../.. title="Pipeliner Documentation" class="md-header__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> <label class="md-header__button md-icon" for=__drawer> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg> </label> <div class=md-header__title data-md-component=header-title> <div class=md-header__ellipsis> <div class=md-header__topic> <span class=md-ellipsis> Pipeliner Documentation </span> </div> <div class=md-header__topic data-md-component=header-topic> <span class=md-ellipsis> Resources </span> </div> </div> </div> <label class="md-header__button md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> </label> <div class=md-search data-md-component=search role=dialog> <label class=md-search__overlay for=__search></label> <div class=md-search__inner role=search> <form class=md-search__form name=search> <input type=text class=md-search__input name=query aria-label=Search placeholder=Search autocapitalize=off autocorrect=off autocomplete=off spellcheck=false data-md-component=search-query required> <label class="md-search__icon md-icon" for=__search> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </label> <nav class=md-search__options aria-label=Search> <button type=reset class="md-search__icon md-icon" aria-label=Clear tabindex=-1> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg> </button> </nav> </form> <div class=md-search__output> <div class=md-search__scrollwrap data-md-scrollfix> <div class=md-search-result data-md-component=search-result> <div class=md-search-result__meta> Initializing search </div> <ol class=md-search-result__list></ol> </div> </div> </div> </div> </div> <div class=md-header__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> </nav> </header> <div class=md-container data-md-component=container> <main class=md-main data-md-component=main> <div class="md-main__inner md-grid"> <div class="md-sidebar md-sidebar--primary" data-md-component=sidebar data-md-type=navigation> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--primary" aria-label=Navigation data-md-level=0> <label class=md-nav__title for=__drawer> <a href=../.. title="Pipeliner Documentation" class="md-nav__button md-logo" aria-label="Pipeliner Documentation" data-md-component=logo> <img src=../../assets/icons/doc-book.svg alt=logo> </a> Pipeliner Documentation </label> <div class=md-nav__source> <a href=https://github.com/CCBR/Pipeliner title="Go to repository" class=md-source data-md-component=source> <div class="md-source__icon md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 448 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg> </div> <div class=md-source__repository> CCBR/Pipeliner </div> </a> </div> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../.. class=md-nav__link> Home </a> </li> <li class=md-nav__item> <a href=../../about/ class=md-nav__link> About </a> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3 type=checkbox id=__nav_3> <label class=md-nav__link for=__nav_3> Getting started <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label="Getting started" data-md-level=1> <label class=md-nav__title for=__nav_3> <span class="md-nav__icon md-icon"></span> Getting started </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_1 type=checkbox id=__nav_3_1> <label class=md-nav__link for=__nav_3_1> Users <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Users data-md-level=2> <label class=md-nav__title for=__nav_3_1> <span class="md-nav__icon md-icon"></span> Users </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Prerequisites </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Installation </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_3_2 type=checkbox id=__nav_3_2> <label class=md-nav__link for=__nav_3_2> Developers <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Developers data-md-level=2> <label class=md-nav__title for=__nav_3_2> <span class="md-nav__icon md-icon"></span> Developers </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Installation </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Customization </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4 type=checkbox id=__nav_4 checked> <label class=md-nav__link for=__nav_4> Pipelines <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=Pipelines data-md-level=1> <label class=md-nav__title for=__nav_4> <span class="md-nav__icon md-icon"></span> Pipelines </label> <ul class=md-nav__list data-md-scrollfix> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_1 type=checkbox id=__nav_4_1> <label class=md-nav__link for=__nav_4_1> ChIP-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=ChIP-seq data-md-level=2> <label class=md-nav__title for=__nav_4_1> <span class="md-nav__icon md-icon"></span> ChIP-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../ChIP-seq/ChIP-seq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../ChIP-seq/ChIP-seq-output-files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../ChIP-seq/ChIP-seq-tools/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--active md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_2 type=checkbox id=__nav_4_2 checked> <label class=md-nav__link for=__nav_4_2> RNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=RNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_2> <span class="md-nav__icon md-icon"></span> RNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../Theory-and-practical-guide-for-RNA-seq/ class=md-nav__link> Theory </a> </li> <li class=md-nav__item> <a href=../Gene-and-isoform-expression-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../TLDR-RNA-seq/ class=md-nav__link> Quick Start </a> </li> <li class=md-nav__item> <a href=../Gene-and-isoform-expression-guided-tutorial/ class=md-nav__link> User Tutorial </a> </li> <li class="md-nav__item md-nav__item--active"> <input class="md-nav__toggle md-toggle" data-md-toggle=toc type=checkbox id=__toc> <label class="md-nav__link md-nav__link--active" for=__toc> Resources <span class="md-nav__icon md-icon"></span> </label> <a href=./ class="md-nav__link md-nav__link--active"> Resources </a> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#reference-genomes class=md-nav__link> Reference genomes </a> </li> <li class=md-nav__item> <a href=#tools-and-versions class=md-nav__link> Tools and versions </a> <nav class=md-nav aria-label="Tools and versions"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#quality-control-pipeline class=md-nav__link> Quality-control pipeline </a> </li> <li class=md-nav__item> <a href=#differential-expression-pipeline class=md-nav__link> Differential expression pipeline </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#references class=md-nav__link> References </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_3 type=checkbox id=__nav_4_3> <label class=md-nav__link for=__nav_4_3> miRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=miRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_3> <span class="md-nav__icon md-icon"></span> miRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../miRNA-seq/overview-miRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/TLDR-miRNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/miRSeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../miRNA-seq/miRSeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_4 type=checkbox id=__nav_4_4> <label class=md-nav__link for=__nav_4_4> scRNA-seq <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=scRNA-seq data-md-level=2> <label class=md-nav__title for=__nav_4_4> <span class="md-nav__icon md-icon"></span> scRNA-seq </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../scRNA-seq/overview-scRNA-seq/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/TLDR-Single-Cell-RNASeq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/scRNASeq-Output-Files/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../scRNA-seq/scRNASeq-Tools-and-Versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_4_5 type=checkbox id=__nav_4_5> <label class=md-nav__link for=__nav_4_5> WES/WGS <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=WES/WGS data-md-level=2> <label class=md-nav__title for=__nav_4_5> <span class="md-nav__icon md-icon"></span> WES/WGS </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-overview/ class=md-nav__link> Overview </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/TLDR-WEG-seq/ class=md-nav__link> User Tutorial </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-output/ class=md-nav__link> Output Files </a> </li> <li class=md-nav__item> <a href=../../WEG-seq/Whole-exome-genome-tools-and-versions/ class=md-nav__link> Resources </a> </li> </ul> </nav> </li> </ul> </nav> </li> <li class="md-nav__item md-nav__item--nested"> <input class="md-nav__toggle md-toggle" data-md-toggle=__nav_5 type=checkbox id=__nav_5> <label class=md-nav__link for=__nav_5> FAQ <span class="md-nav__icon md-icon"></span> </label> <nav class=md-nav aria-label=FAQ data-md-level=1> <label class=md-nav__title for=__nav_5> <span class="md-nav__icon md-icon"></span> FAQ </label> <ul class=md-nav__list data-md-scrollfix> <li class=md-nav__item> <a href=../../troubleshooting/ class=md-nav__link> Troubleshooting </a> </li> <li class=md-nav__item> <a href=../../dev/coming-soon/ class=md-nav__link> Data Privacy </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=../../community-and-contribution/ class=md-nav__link> Contribute </a> </li> <li class=md-nav__item> <a href=../../changelog/ class=md-nav__link> Changelog </a> </li> <li class=md-nav__item> <a href=../../contact-us/ class=md-nav__link> Contact </a> </li> <li class=md-nav__item> <a href=../../license/ class=md-nav__link> License </a> </li> </ul> </nav> </div> </div> </div> <div class="md-sidebar md-sidebar--secondary" data-md-component=sidebar data-md-type=toc> <div class=md-sidebar__scrollwrap> <div class=md-sidebar__inner> <nav class="md-nav md-nav--secondary" aria-label="Table of contents"> <label class=md-nav__title for=__toc> <span class="md-nav__icon md-icon"></span> Table of contents </label> <ul class=md-nav__list data-md-component=toc data-md-scrollfix> <li class=md-nav__item> <a href=#reference-genomes class=md-nav__link> Reference genomes </a> </li> <li class=md-nav__item> <a href=#tools-and-versions class=md-nav__link> Tools and versions </a> <nav class=md-nav aria-label="Tools and versions"> <ul class=md-nav__list> <li class=md-nav__item> <a href=#quality-control-pipeline class=md-nav__link> Quality-control pipeline </a> </li> <li class=md-nav__item> <a href=#differential-expression-pipeline class=md-nav__link> Differential expression pipeline </a> </li> </ul> </nav> </li> <li class=md-nav__item> <a href=#references class=md-nav__link> References </a> </li> </ul> </nav> </div> </div> </div> <div class=md-content data-md-component=content> <article class="md-content__inner md-typeset"> <a href=https://github.com/CCBR/pipeliner-docs/edit/master/docs/RNA-seq/Differential-expression-pipeline-tools-and-versions.md title="Edit this page" class="md-content__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg> </a> <h1>Resources</h1> <h2 id=reference-genomes>Reference genomes<a class=headerlink href=#reference-genomes title="Permanent link">&para;</a></h2> <blockquote> <p><em><strong>Warning:</strong></em> This section contains FTP links for downloading each reference file. </p> </blockquote> <p>The quality-control and differential expression pipeline support the following genomes: <br> <!---
**Human** <code>hg19</code> <code>hg38</code> <code>hg38_30</code> <code>hs37d5</code> <code>hs38d1</code> <code>hg38_30</code><br />**Human + _Integrated Virus_** <code>hg38_30_KSHV</code> <code>hg38_HPV16</code><br />**Mouse** <code>mm10</code> <code>mm9</code> <code>mm10_M21</code><br />**Dog** <code>canFam3</code><br />**Rhesus macaque** <code>Mmul_8.0.1</code>
---></p> <table> <thead> <tr> <th>GenomeName</th> <th>Species</th> <th>Annotation Version</th> <th>Comments</th> </tr> </thead> <tbody> <tr> <td>hg19</td> <td>Homo sapiens (human)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_19/gencode.v19.annotation.gtf.gz>Gencode Release 19</a></td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_19/GRCh37.p13.genome.fa.gz>GRCh37</a>, Release date: 07/2013</td> </tr> <tr> <td>hg38</td> <td>Homo sapiens (human)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_28/gencode.v28.annotation.gtf.gz>Gencode Release 28</a></td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_28/GRCh38.primary_assembly.genome.fa.gz>GRCh38</a>, Annotation Release date: 11/2017</td> </tr> <tr> <td>hg38_30</td> <td>Homo sapiens (human)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/gencode.v30.annotation.gtf.gz>Gencode Release 30</a></td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_28/GRCh38.primary_assembly.genome.fa.gz>GRCh38</a>, Annotation Release date: 11/2018</td> </tr> <tr> <td>hs37d5</td> <td>Homo sapiens (human)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_19/gencode.v19.annotation.gtf.gz>Gencode Release 19</a></td> <td><a href=ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence>hg19 + decoy sequences</a></td> </tr> <tr> <td>hs38d1</td> <td>Homo sapiens (human)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_28/gencode.v28.annotation.gtf.gz>Gencode Release 28</a></td> <td>hg38 + <a href=https://www.ncbi.nlm.nih.gov/assembly/GCA_000786075.2/ >decoy sequences</a></td> </tr> <tr> <td>hg38_30_KSHV</td> <td>Homo sapiens + KSHV</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/gencode.v30.annotation.gtf.gz>Gencode Release 30</a> (hg38) + <a href=https://www.ncbi.nlm.nih.gov/nuccore/NC_009333.1>06/2019</a> (KSHV)</td> <td>hg38 + <a href=https://www.ncbi.nlm.nih.gov/nuccore/NC_009333.1>NC_009333.1</a>. Annotation Release dates: 11/2018(human) + 06/2019(KSHV)</td> </tr> <tr> <td>hg38_HPV16</td> <td>Homo sapiens + HPV16</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_28/gencode.v28.annotation.gtf.gz>Gencode Release 28</a> (hg38) + 03/2019 (HPV16 custom annotation from Zheng lab)</td> <td>hg38 + HPV16 custom sequence based off of KU298885.1 with custom annotation</td> </tr> <tr> <td>mm9</td> <td>Mus musculus (house mouse)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M1/gencode.vM1.annotation.gtf.gz>M1</a></td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M1/NCBIM37.genome.fa.gz>NCBIM37</a>, Annotation Release date: 12/2011</td> </tr> <tr> <td>mm10</td> <td>Mus musculus (house mouse)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M18/gencode.vM18.annotation.gtf.gz>M18</a></td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M18/GRCm38.primary_assembly.genome.fa.gz>GRCm38</a>, Annotation Release date: 07/2018</td> </tr> <tr> <td>mm10_M21</td> <td>Mus musculus (house mouse)</td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M21/gencode.vM21.annotation.gtf.gz>M21</a></td> <td><a href=ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M18/GRCm38.primary_assembly.genome.fa.gz>GRCm38</a>, Annotation Release date: 04/2019</td> </tr> <tr> <td>canFam3</td> <td>Canis lupus familiaris (dog)</td> <td><a href=ftp://ftp.ensembl.org/pub/release-94/gtf/canis_familiaris/Canis_familiaris.CanFam3.1.94.gtf.gz>Ensembl Release 94</a></td> <td><a href=https://www.ncbi.nlm.nih.gov/assembly/GCF_000002285.3/ >CanFam3.1</a></td> </tr> <tr> <td>Mmul_8.0.1</td> <td>Macaca mulatta (Rhesus monkey or macaque)</td> <td><a href=ftp://ftp.ensembl.org/pub/release-97/gtf/macaca_mulatta/Macaca_mulatta.Mmul_8.0.1.97.gtf.gz>Ensembl Release 97</a></td> <td><a href=ftp://ftp.ensembl.org/pub/release-95/fasta/macaca_mulatta/dna/Macaca_mulatta.Mmul_8.0.1.dna.toplevel.fa.gz>Mmul_8.0.1 (rheMac8)</a></td> </tr> </tbody> </table> <blockquote> <p><strong>Please note</strong>: If you are looking for a reference genome and/or annotation <em><strong>that is currently not available</strong></em>, it can be generated using <a href=https://github.com/CCBR/PIM>Pipeliner Index Maker (PIM) </a>. Given the reference's FASTA file <code>ref.fa</code> and a GTF file <code>genes.gtf</code>, PIM will create all of the required reference files to run RNA-seq pipeline on Biowulf.</p> </blockquote> <h2 id=tools-and-versions>Tools and versions<a class=headerlink href=#tools-and-versions title="Permanent link">&para;</a></h2> <h3 id=quality-control-pipeline>Quality-control pipeline<a class=headerlink href=#quality-control-pipeline title="Permanent link">&para;</a></h3> <blockquote> <p><em>Raw data &gt; Adapter Trimming &gt; Alignment &gt; Quantification (genes and isoforms)</em></p> </blockquote> <table> <thead> <tr> <th>Tool</th> <th align=center>Version</th> <th>Notes</th> </tr> </thead> <tbody> <tr> <td>FastQC<sup>2</sup></td> <td align=center>0.11.5</td> <td><strong>Quality-control step</strong> to assess sequencing quality, run before and after adapter trimming</td> </tr> <tr> <td>Cutadapt<sup>3</sup></td> <td align=center>1.18</td> <td><strong>Data processing step</strong> to remove adapter sequences and perform quality trimming</td> </tr> <tr> <td>Kraken<sup>18</sup></td> <td align=center>1.1</td> <td><strong>Quality-control step</strong> to assess microbial taxonomic composition</td> </tr> <tr> <td>KronaTools<sup>19</sup></td> <td align=center>2.7</td> <td><strong>Quality-control step</strong> to visualize kraken output</td> </tr> <tr> <td>FastQ Screen<sup>21</sup></td> <td align=center>0.9.3</td> <td><strong>Quality-control step</strong> to assess contamination; additional dependencies: <code>bowtie2/2.3.4</code>, <code>perl/5.24.3</code></td> </tr> <tr> <td>STAR<sup>4</sup></td> <td align=center>2.7.0f</td> <td><strong>Data processing step</strong> to align reads against reference genome (using its two-pass mode)</td> </tr> <tr> <td>QualiMap<sup>20</sup></td> <td align=center>2.2.1</td> <td><strong>Quality-control step</strong> to assess various alignment metrics, also calculates insert_size</td> </tr> <tr> <td>Picard<sup>12</sup></td> <td align=center>2.17.11</td> <td><strong>Quality-control step</strong> to run <code>MarkDuplicates</code>, <code>CollectRnaSeqMetrics</code> and <code>AddOrReplaceReadGroups</code></td> </tr> <tr> <td>Preseq<sup>1</sup></td> <td align=center>2.0.3</td> <td><strong>Quality-control step</strong> to estimate library complexity</td> </tr> <tr> <td>SAMtools<sup>17</sup></td> <td align=center>1.6</td> <td><strong>Quality-control step</strong> to run <code>flagstat</code> to calculate alignment statistics</td> </tr> <tr> <td>bam2strandedbw</td> <td align=center><a href=https://github.com/CCBR/Pipeliner/blob/master/Results-template/Scripts/bam2strandedbw.pe.sh>custom</a></td> <td><strong>Summarization step</strong> to convert STAR aligned PE bam file into forward and reverse strand bigwigs suitable for a genomic track viewer like IGV</td> </tr> <tr> <td>RSeQC<sup>11</sup></td> <td align=center>2.6.4</td> <td><strong>Quality-control step</strong> to infer stranded-ness and read distributions over specific genomic features</td> </tr> <tr> <td>RSEM<sup>5</sup></td> <td align=center>1.3.0</td> <td><strong>Data processing step</strong> to quantify gene and isoform counts</td> </tr> <tr> <td>Subread<sup>14</sup></td> <td align=center>1.5.2</td> <td><strong>Data processing step</strong> to run <code>featureCounts</code>, an alternative quantification method to RSEM</td> </tr> <tr> <td>PCA Report<sup>16</sup></td> <td align=center><a href=https://github.com/CCBR/Pipeliner/blob/master/Results-template/Scripts/PcaReport.Rmd>custom</a></td> <td><strong>Summarization step</strong> to identify outliers prior to DE, contains pre- and post- normalization plots</td> </tr> <tr> <td>MultiQC<sup>15</sup></td> <td align=center>1.4</td> <td><strong>Reporting step</strong> to aggregate sample statistics and quality-control information across all sample</td> </tr> </tbody> </table> <h3 id=differential-expression-pipeline>Differential expression pipeline<a class=headerlink href=#differential-expression-pipeline title="Permanent link">&para;</a></h3> <blockquote> <p><em>Raw counts matrix &gt; Normalization &gt; Differential Expression Analysis &gt; Fuctional Impact</em></p> </blockquote> <table> <thead> <tr> <th>Tool</th> <th align=center>Version</th> <th>Notes</th> </tr> </thead> <tbody> <tr> <td>filtersamples<sup>16</sup></td> <td align=center><a href=https://github.com/CCBR/Pipeliner/blob/master/Results-template/Scripts/filtersamples.R>custom</a></td> <td><strong>Data processing step</strong> to remove low CPM genes prior to differential expression analysis</td> </tr> <tr> <td>PCAReport<sup>16</sup></td> <td align=center><a href=https://github.com/CCBR/Pipeliner/blob/master/Results-template/Scripts/PcaReport.Rmd>custom</a></td> <td><strong>Summarization step</strong> to identify outliers prior to DE, contains pre- and post- normalization plots</td> </tr> <tr> <td>EBSeq<sup>22</sup></td> <td align=center>1.2.0</td> <td><strong>Data processing step</strong> to find differentially expressed isoforms, additional dependencies: <code>rsem/1.3.0</code></td> </tr> <tr> <td>edgeR<sup>23</sup></td> <td align=center>3.24.3</td> <td><strong>Data processing step</strong> to find differentially expressed genes. Counts are modeled using a negative binomial distribution with mean equal to the multiplication of library size and relative abundance while a quasi-likelihood F-test is used for testing gene differential expression</td> </tr> <tr> <td>DESeq2<sup>13</sup></td> <td align=center>1.22.2</td> <td><strong>Data processing step</strong> to find differentially expressed genes. Counts are modeled using a negative binomial distribution similar to edgeR while a wald-test is implemented to test for differential expression</td> </tr> <tr> <td>limma<sup>7,8</sup></td> <td align=center>3.38.3</td> <td><strong>Data processing step</strong> to find differentially expressed genes. Log-transformed counts are modeled using a method analogous to a t-distribution while a moderated t-statistics is used to test for differential expression</td> </tr> <tr> <td>l2p<sup>16</sup></td> <td align=center><a href=https://github.com/CCBR/l2p>custom</a></td> <td><strong>Summarization step</strong> for gene set enrichment analysis</td> </tr> </tbody> </table> <h2 id=references>References<a class=headerlink href=#references title="Permanent link">&para;</a></h2> <p><sup><strong>1.</strong> Daley, T. and A.D. Smith, Predicting the molecular complexity of sequencing libraries. Nat Methods, 2013. 10(4): p. 325-7.</sup><br> <sup><strong>2.</strong> Andrews, S. (2010). FastQC: a quality control tool for high throughput sequence data.</sup><br> <sup><strong>3.</strong> Martin, M. (2011). "Cutadapt removes adapter sequences from high-throughput sequencing reads." EMBnet 17(1): 10-12.</sup><br> <sup><strong>4.</strong> Dobin, A., et al., STAR: ultrafast universal RNA-seq aligner. Bioinformatics, 2013. 29(1): p. 15-21.</sup><br> <sup><strong>5.</strong> Li, B. and C.N. Dewey, RSEM: accurate transcript quantification from RNA-Seq data with or without a reference genome. BMC Bioinformatics, 2011. 12: p. 323.</sup><br> <sup><strong>6.</strong> Harrow, J., et al., GENCODE: the reference human genome annotation for The ENCODE Project. Genome Res, 2012. 22(9): p. 1760-74.</sup><br> <sup><strong>7.</strong> Law, C.W., et al., voom: Precision weights unlock linear model analysis tools for RNA-seq read counts. Genome Biol, 2014. 15(2): p. R29.</sup><br> <sup><strong>8.</strong> Smyth, G.K., Linear models and empirical bayes methods for assessing differential expression in microarray experiments. Stat Appl Genet Mol Biol, 2004. 3: p. Article3.</sup><br> <sup><strong>9.</strong> Fabregat, A., et al., The Reactome Pathway Knowledgebase. Nucleic Acids Res, 2018. 46(D1): p. D649-D655.</sup><br> <sup><strong>10.</strong> Liberzon, A., et al., Molecular signatures database (MSigDB) 3.0. Bioinformatics, 2011. 27(12): p. 1739-40.</sup><br> <sup><strong>11.</strong> Wang, L., et al. (2012). "RSeQC: quality control of RNA-seq experiments." Bioinformatics 28(16): 2184-2185.</sup><br> <sup><strong>12.</strong> The Picard toolkit. <a href=https://broadinstitute.github.io/picard/ >https://broadinstitute.github.io/picard/</a>.</sup><br> <sup><strong>13.</strong> Love, M. I., et al. (2014). "Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2." Genome Biol 15(12): 550.</sup><br> <sup><strong>14.</strong> Liao, Y., et al. (2013). "The Subread aligner: fast, accurate and scalable read mapping by seed-and-vote." Nucleic Acids Research 41(10): e108-e108.</sup><br> <sup><strong>15.</strong> Ewels, P., et al. (2016). "MultiQC: summarize analysis results for multiple tools and samples in a single report." Bioinformatics 32(19): 3047-3048.</sup><br> <sup><strong>16.</strong> R Core Team (2018). R: A Language and Environment for Statistical Computing. Vienna, Austria, R Foundation for Statistical Computing.</sup><br> <sup><strong>17.</strong> Li, H., et al. (2009). "The Sequence Alignment/Map format and SAMtools." Bioinformatics 25(16): 2078-2079.</sup><br> <sup><strong>18.</strong> Wood, D. E. and S. L. Salzberg (2014). "Kraken: ultrafast metagenomic sequence classification using exact alignments." Genome Biol 15(3): R46.</sup><br> <sup><strong>19.</strong> Ondov, B. D., et al. (2011). "Interactive metagenomic visualization in a Web browser." BMC Bioinformatics 12(1): 385.</sup><br> <sup><strong>20.</strong> Okonechnikov, K., et al. (2015). "Qualimap 2: advanced multi-sample quality control for high-throughput sequencing data." Bioinformatics 32(2): 292-294.</sup><br> <sup><strong>21.</strong> Wingett, S. and S. Andrews (2018). "FastQ Screen: A tool for multi-genome mapping and quality control." F1000Research 7(2): 1338.</sup><br> <sup><strong>22.</strong> Leng, N., et al. (2013). "EBSeq: an empirical Bayes hierarchical model for inference in RNA-seq experiments." Bioinformatics 29(8): 1035-1043.</sup><br> <sup><strong>23.</strong> Robinson, M. D., et al. (2009). "edgeR: a Bioconductor package for differential expression analysis of digital gene expression data." Bioinformatics 26(1): 139-140.</sup> </p> <hr> <div class=md-source-file> <small> Last update: 2022-11-04 </small> </div> </article> </div> </div> </main> <footer class=md-footer> <nav class="md-footer__inner md-grid" aria-label=Footer> <a href=../Gene-and-isoform-expression-guided-tutorial/ class="md-footer__link md-footer__link--prev" aria-label="Previous: User Tutorial" rel=prev> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg> </div> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Previous </span> User Tutorial </div> </div> </a> <a href=../../miRNA-seq/overview-miRNA-seq/ class="md-footer__link md-footer__link--next" aria-label="Next: Overview" rel=next> <div class=md-footer__title> <div class=md-ellipsis> <span class=md-footer__direction> Next </span> Overview </div> </div> <div class="md-footer__button md-icon"> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg> </div> </a> </nav> <div class="md-footer-meta md-typeset"> <div class="md-footer-meta__inner md-grid"> <div class=md-copyright> <div class=md-copyright__highlight> Copyright &copy; 2020 CCBR </div> Made with <a href=https://squidfunk.github.io/mkdocs-material/ target=_blank rel=noopener> Material for MkDocs </a> </div> <div class=md-social> <a href=http://bioinformatics.cancer.gov target=_blank rel=noopener title=bioinformatics.cancer.gov class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M319.9 320c57.41 0 103.1-46.56 103.1-104s-46.54-104-103.1-104c-57.41 0-103.1 46.56-103.1 104-.9 57.4 45.7 104 103.1 104zm50 32h-99.8C191.6 352 128 411.7 128 485.3c0 14.8 12.7 26.7 28.4 26.7h327.2c15.7 0 28.4-11.9 28.4-26.7 0-73.6-63.6-133.3-142.1-133.3zM512 160c44.18 0 80-35.82 80-80S556.2 0 512 0c-44.18 0-80 35.82-80 80s35.8 80 80 80zm-328.1 56c0-5.449.982-10.63 1.609-15.91C174.6 194.1 162.6 192 149.9 192H88.08C39.44 192 0 233.8 0 285.3 0 295.6 7.887 304 17.62 304h199.5c-20.42-23.8-33.22-54.3-33.22-88zM128 160c44.18 0 80-35.82 80-80S172.2 0 128 0C83.82 0 48 35.82 48 80s35.82 80 80 80zm423.9 32h-61.84c-12.8 0-24.88 3.037-35.86 8.24.6 5.26 1.6 10.36 1.6 15.76 0 33.71-12.78 64.21-33.16 88h199.7c9.76 0 17.66-8.4 17.66-18.7 0-51.5-39.4-93.3-88.1-93.3z"/></svg> </a> <a href=https://github.com/CCBR target=_blank rel=noopener title=github.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 496 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg> </a> <a href=https://hub.docker.com/orgs/nciccbr/repositories target=_blank rel=noopener title=hub.docker.com class=md-social__link> <svg xmlns=http://www.w3.org/2000/svg viewbox="0 0 640 512"><!-- Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M349.9 236.3h-66.1v-59.4h66.1v59.4zm0-204.3h-66.1v60.7h66.1V32zm78.2 144.8H362v59.4h66.1v-59.4zm-156.3-72.1h-66.1v60.1h66.1v-60.1zm78.1 0h-66.1v60.1h66.1v-60.1zm276.8 100c-14.4-9.7-47.6-13.2-73.1-8.4-3.3-24-16.7-44.9-41.1-63.7l-14-9.3-9.3 14c-18.4 27.8-23.4 73.6-3.7 103.8-8.7 4.7-25.8 11.1-48.4 10.7H2.4c-8.7 50.8 5.8 116.8 44 162.1 37.1 43.9 92.7 66.2 165.4 66.2 157.4 0 273.9-72.5 328.4-204.2 21.4.4 67.6.1 91.3-45.2 1.5-2.5 6.6-13.2 8.5-17.1l-13.3-8.9zm-511.1-27.9h-66v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm78.1 0h-66.1v59.4h66.1v-59.4zm-78.1-72.1h-66.1v60.1h66.1v-60.1z"/></svg> </a> </div> </div> </div> </footer> </div> <div class=md-dialog data-md-component=dialog> <div class="md-dialog__inner md-typeset"></div> </div> <script id=__config type=application/json>{"base": "../..", "features": [], "search": "../../assets/javascripts/workers/search.b028fd86.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script> <script src=../../assets/javascripts/bundle.0f659f14.min.js></script> </body> </html>